Thermal reactionome of the common ant species *Aphaenogaster*
================================================================
   
**Author:** [John Stanton-Geddes](john.stantongeddes.research@gmail.com)

**April 29, 2014**

**Technical Report No. 3**

**Department of Biology**

**University of Vermont**

```{r setup, echo=FALSE, results='hide', message = FALSE}
# Global settings
options(stringsAsFactors=FALSE, digits = 3, scipen = 1)

# Load libraries
library(R.utils)
library(ggplot2)
library(knitr)
library(knitcitations)
library(pander)
library(stringr)
library(data.table)
library(RCurl)
library(plyr)
library(MASS)

# Load Bioconductor libraries
#source("http://bioconductor.org/biocLite.R")
#biocLite("topGO")
#biocLite("Rgraphviz") # INSTALL BROKEN IN R 3.1.0
library(topGO) 
#suppressMessages(library(Rgraphviz))

# load custom functions
source("scripts/RxNseq.R")

# knitr options
opts_chunk$set(cache=TRUE)

# directory to save results
resultsdir <- "results/"
```


## Summary ##
  
In this technical report, which accompanies the manuscript **Thermal reactionome of a common ant species** (Stanton-Geddes et al., in press), we:

1. describe the *de novo* assembly of the transcriptome for two ant colonies with in the *Aphaenogaster rudis-picea-fulva* species complex `r citep("10.1155/2012/752815")`
2. identify thermally-responsive genes
3. perform gene set enrichment analysis

This script is completely reproducible assuming that R, `knitr` and the other required libraries (listed within the source document) are installed on a standard linux system using the following:
    
    Rscript -e "library(knitr); knit('ApTranscriptome_TR.Rmd')"

The assembled transcriptome, annotation and expression values are downloaded rather than re-run due to the computational demands, but the exact commands for each of these steps are documented below.



## Data ##

The Illumina fastq files are available from [https://minilims1.uvm.edu/BCProject-26-Cahan/downloads.html]. The Trimmomatic filtered fastq files should be downloaded, uncompressed and moved to the appropriate directory using the following code.

~~~
# download filtered fastq files, uncompress and move
wget --no-check-certificate https://minilims1.uvm.edu/BCProject-26-Cahan/_downloads/trimmomatic_output.tar.gz
tar -zxvf trimmomatic_output.tar.gz
mkdir -p data/
mv ind_files data/.
~~~



## Sample description ##

Two ant colonies were used for the transcriptome sequencing. The first, designated *A22*, was collected at Molly Bog, Vermont in August 2012 by Nick Gotelli and Andrew Nguyen. The second colony, designated *Ar*, was collected by Lauren Nichols in Raleigh, North Carolina. These colonies were identified as *A. picea* and *A. carolinensis* by Bernice Bacon de Marco. The colonies were maintained in the lab for 6 months prior to sample collection. 

For each colony, three ants were exposed to one of 12 temperature treatments, every 3.5C ranging from 0C to 38.5C, for one hour in glass tubes in a water bath. The ants were flash frozen and stored at -80C until RNA was extracted using a two step extraction; [RNAzol RT](http://www.mrcgene.com/rnazol.htm) (Molecular Research Center, Inc) followed by an [RNeasy Micro](http://www.qiagen.com/products/catalog/sample-technologies/rna-sample-technologies/total-rna/rneasy-micro-kit) column (Qiagen). Samples from each colony were pooled and sequenced in separate lanes on a 100bp paired-end run of an Illumina HiSeq at the University of Minnesota Genomics Center, yielding 20^6^ and 16^6^ reads for the *A22* and *Ar* samples, respectively.



## Transcriptome assembly ##

The Illumina reads were filtered using the program [Trimmomatic](http://www.usadellab.org/cms/?page=trimmomatic) `r citep("10.1093/nar/gks540")` to remove Ilumina adapter sequences and filter out bases with quality scores less than ??. 

~~~
TRIMMOMATIC CODE
~~~

This filtering yielded...

These reads were combined and used in *de novo* transcriptome assembly using the program [Trinity](http://trinityrnaseq.sourceforge.net/) `r citep("10.1038/nbt.1883")`. Note that this required ??? GB RAM and ??? hours run-time and was run on the [Vermont Genetics Network](http://vgn.uvm.edu/) computing cluster. 

~~~
TRINITY CODE
~~~

This assembly contained 100,381 unique components (roughly genes) in 126,172 total transcripts (Table 1). 

As we were assembling two divergent colonies into a single transcriptome, we suspected that this assembly would be susceptible to known problems of errors during assembly (e.g. chimeric transcripts that are fusions of two transcripts) and redundancy `r citep("10.1186/1471-2164-14-328")`. To account for this, we performed two post-assembly processing steps.

First, we ran the program [cap3](http://seq.cs.iastate.edu/) `r citep("10.1101/gr.9.9.868")` setting the maximum gap length and band expansion size to 50 `-f 50 -a 50`, no end clipping as the reads were already filtered `k 0`, requiring 90% identity for assembly, and a minimum overlap length of 100 bp `-o 100`. The percent identity threshold of 90% was chosen to liberally collapse orthologous contigs from the two colonies that may have been assembled separately. 

    cap3 Trinity.fasta -f 50 -a 50 -k 0 -p 90 -o 100 > Trinity_cap3.out

The output of `cap3` gives assembled "contigs" and unassembled "singlets" that were concatenated into a single file.

    # check the number of contigs clustered
    grep -c "Contig" Trinity.fasta.cap.contigs
    grep -c "comp" Trinity.fasta.cap.singlets
    # compare to contigs from Trinity output
    grep -c "comp" Trinity.fasta

    # Combine contigs and singlets from CAP3
    cat Trinity.fasta.cap.contigs Trinity.fasta.cap.singlets > Trinity_cap3.fasta

The output file "Trinity.fasta.cap.info" gives specific information on which contigs were collapsed.

Subsequent to running `cap3`, we ran [uclust](http://drive5.com/usearch/manual/uclust_algo.html) to cluster sequences completely contained within longer sequences, again specificing a 90% identity cutoff for clustering. 

    # sort
    uclust --sort Trinity_cap3.fasta --output Trinity_cap3_sorted.fasta
    # cluster by 90% similarity threshold
    uclust --input Trinity_cap3_sorted.fasta --uc Trinity_cap3_uclust.out --id 0.90
    # convert uclust to fasta format
    uclust --uc2fasta Trinity_cap3_uclust.out --input Trinity_cap3_uclust.fasta

These post-processing step removed `r round((126172-105536)/126172 * 100, 0)`% of the initial reads (Table 1).

```{r assemstats, results = "asis", echo=FALSE}
# make table
trinity <- c("126,172", "100,389,539", "358", "795", "16,201", "1,631")
reduced <- c("105,536", "62,648,997", "320", "593", "15,491", "895")

assemstats <- rbind(trinity, reduced)
colnames(assemstats) <- c("Total contigs", "Total length", "Median contig size", "Mean contig size", "N50 contig", "N50 Length")

pandoc.table(assemstats, style="rmarkdown", caption = "Statistics for Trinity and cap3+uclust reduced transcriptome assemblies")
```

To remove contigs that are likely contaminants from bacterial, archael, virus or human sources, we used the program [DeconSeq](http://deconseq.sourceforge.net/) `r citep("10.1371/journal.pone.0017288")`. We downloaded the bacteria, virus, archae and human [databases of contaminants](ftp://edwards.sdsu.edu:7009/deconseq/db), modified the `DeconSeqConfig.pm` file as described [here](http://www.vcru.wisc.edu/simonlab/bioinformatics/programs/install/deconseq.htm) to point to the databases, and ran DeconSeq specifiying 95% identity over 50% the length of contig

    deconseq.pl -c 50 -i 95 -f Trinity_cap3_uclust.fasta -d Trinity_cap3_uclust -dbs hsref,bast,vir,arch
	
This resulted in removing 5,675 contigs as contaminants, leaving 99,861 "clean" contigs. We spot-checked the contaminants by BLAST and confirmed that they matched bacteria, human or viral sources by greater than 95%. 


Running Trinity and subsequent programs is time and memory-intensive, so the final filtered clean assembly can be downloaded from [...] 

~~~
# download filtered Trinity assembly, uncompress and move
wget http://johnstantongeddes.org/assets/files/Aphaenogaster_transcriptome.tar
tar -xvf Aphaenogaster_transcriptome.tar
# check md5sum
md5sum Trinity_cap3_uclust_clean.fa
# c604159ee944dddda7b4e2a556bf2f53
mkdir -p results/
mkdir -p results/trinity-full/
mv Trinity_cap3_uclust_clean.fasta results/trinity-full/.
~~~



## Transcriptome annotation ##

Annotation was performed by uploading the reduced assembly "Trinity_cap3_uclust.fasta" to the web-based annotation program [FastAnnotator](http://fastannotator.cgu.edu.tw/index.php) `r citep("10.1186/1471-2164-13-S7-S9")`.

Results are available as job ID [13894410176993](http://fastannotator.cgu.edu.tw/job.php?jobid=13894410176993#page=basicinfo).

This annotation file can be read directly to R:

```{r download, echo = TRUE}
### Annotation file

# from either AWS or GoogleDrive
annotationURL <- getURL("http://johnstantongeddes.org/assets/files/ApTranscriptome_AnnotationTable_20140113.txt")
#a2 <- getURL("https://googledrive.com/host/0B75IymziRJ_9Tlg1U1Vxbjk1bzg") # GoogleDrive link

annotationfile <- read.csv(textConnection(annotationURL), header = TRUE, sep = "\t", stringsAsFactors = FALSE)
str(annotationfile)

# Convert to data.table
annotationtable <- data.table(annotationfile)
head(annotationtable)
```

Note that I used the reduced assembly, prior to cleaning out contaminants. As I use the "clean" assembly for read mapping and identification of responsive genes, the contaminants are dis-regarded in downstream analyses. 


## Identification of thermally-responsive genes ##

### Quantify gene expression ###

I quantified gene expression using [sailfish](http://www.cs.cmu.edu/~ckingsf/software/sailfish/index.html). To run this program, first make sure that PATHs to the software libraries are set up correctly: 
                                                 
    export LD_LIBRARY_PATH=/opt/software/Sailfish-0.6.2-Linux_x86-64/lib:$LD_LIBRARY_PATH
    export PATH=/opt/software/Sailfish-0.6.2-Linux_x86-64/bin:$PATH

Then build the index of the assembly:

    sailfish index -t Trinity_cap3_uclust_clean.fasta -o sailfish-index -k 20 -p 4

Once this is done, quantify expression for the Trimmomatic filtered reads from each colony-treatment sample separately. Note that for each sample, there are three four filtered read files:

- paired.left.fastq
- paired.right.fastq
- unpaired.left.fastq
- unpaired.right.fastq
                                                 
Make a directory for the expression values

    mkdir sailfish-expression                                                 

Then, for each sample, run the following command:
                                                 
    sailfish -i sailfish-index -o sailfish-expression/A22-0 --reads A22-0_ATCACG.paired.left.fastq A22-0_ATCACG.paired.right.fastq A22-0_ATCACG.unpaired.left.fastq A22-0_ATCACG.unpaired.right.fastq -p 4

Or, using a loop in R:                                                 
                                                 
```{r sailfish, eval=TRUE, echo=TRUE, cache=TRUE}
# directory containing trimmed reads
readdir <- "data/ind_files/" 
# sailfish index directory
sfindex <- "results/trinity-full/sailfish-index"
# sailfish expression directory
sfexpressionroot <- "results/trinity-full/sailfish-expression/"

# list of reads in each of four trimmed classes
readlist <- list.files(readdir)
(paired.left <- readlist[grep(".\\.paired.left.fastq$", readlist)])
(paired.right <- readlist[grep("\\.paired.right.fastq$", readlist)])
(unpaired.left <- readlist[grep("unpaired.left.fastq$", readlist)])
(unpaired.right <- readlist[grep("unpaired.right.fastq$", readlist)])

# Loop across each sample and quantify expression

# NOTE - samples listed in same order as given by the above lists
samples <- c("A22-0", "A22-10", "A22-14", "A22-17", "A22-21", "A22-24", "A22-28", "A22-31", "A22-35", "A22-38", "A22-3", "A22-7", "Ar-0", "Ar-10", "Ar-14", "Ar-17", "Ar-21", "Ar-24", "Ar-28", "Ar-31", "Ar-35", "Ar-38", "Ar-3", "Ar-7")

for (j in 1:length(samples)) {
    message("Start expression quantification for sample ", samples[j], ": ", Sys.time())
    quantdir <- paste(sfexpressionroot, samples[j], "_quant", sep="")
    samp.pos <- grep(paste(paste(samples[j], "_", sep="")), paired.left)
    samp.paired.l <- paste(readdir, paired.left[samp.pos], sep="")
    samp.paired.r <- paste(readdir, paired.right[samp.pos], sep="")
    samp.unpaired.l <- paste(readdir, unpaired.left[samp.pos], sep="")
    samp.unpaired.r <- paste(readdir, unpaired.right[samp.pos], sep="")
    sailfishcmd <- paste("sailfish quant -i ", sfindex, " -o ", quantdir, " --reads ", samp.paired.l, " ", samp.paired.r, " ", samp.unpaired.l, " ", samp.unpaired.r, " -p 4", sep="")
#   print(sailfishcmd)
    system(sailfishcmd)
    message("Done with expression quantification for sample ", samples[j], ": ", Sys.time(), "\n")
}
```

This generated a directory for each sample

`r list.files(sfexpressionroot)`

and within each directory there are the following r:

`r list.files(quantdir)`

The file *quant_bias_corrected.sf* contains the following columns, following a number of header lines:

1. Transcript ID
2. Transcript Length
3. Transcripts per Million (TPM): computed as described in `r citep("10.1093/bioinformatics/btp692")`, and is meant as an estimate of the number of transcripts, per million observed transcripts, originating from each isoform.
4. Reads Per Kilobase per Million mapped reads (RPKM): classic measure of relative transcript abundance, and is an estimate of the number of reads per kilobase of transcript (per million mapped reads) originating from each transcript.

The TPM column for each sample was extracted and combined into a matrix for each colony.

```{r load_expression_data, eval=TRUE, echo=FALSE, results='hide'}
# read in each file using loop
samples <- c("A22-0", "A22-3", "A22-7", "A22-10", "A22-14", "A22-17", "A22-21", "A22-24", "A22-28", "A22-31", "A22-35", "A22-38", "Ar-0", "Ar-3", "Ar-7", "Ar-10", "Ar-14", "Ar-17", "Ar-21", "Ar-24", "Ar-28", "Ar-31", "Ar-35", "Ar-38")

for (j in 1:length(samples)) {
    samp <- samples[j]
    trtval <- as.numeric(str_split_fixed(samp, "-", 2)[2])
    outpre <- gsub("-", "_", samp)
    outname <- paste(outpre, "_quant", sep="")
    read.sailfish.quant(filein=paste(sfexpressionroot, samp, "_quant/quant_bias_corrected.sf", sep=""), outname=outname, samp = samp, trtval = trtval)
}

# combine into long format
A22.TPM <- rbind(A22_0_quant, A22_3_quant, A22_7_quant, A22_10_quant, A22_14_quant, A22_17_quant, A22_21_quant, A22_24_quant, A22_28_quant, A22_31_quant, A22_35_quant, A22_38_quant)
str(A22.TPM)

# convert to data.table
A22.TPM <- data.table(A22.TPM)
setkey(A22.TPM, Transcript, val)

# set "trt" to true values - truncated in file names for convenience

A22.TPM[val==3,val:=3.5]
A22.TPM[val==10,val:=10.5]
A22.TPM[val==17,val:=17.5]
A22.TPM[val==24,val:=24.5]
A22.TPM[val==31,val:=31.5]
A22.TPM[val==38,val:=38.5]
head(A22.TPM)
str(A22.TPM)


# Repeat for Ar
Ar.TPM <- rbind(Ar_0_quant, Ar_3_quant, Ar_7_quant, Ar_10_quant, Ar_14_quant, Ar_17_quant, Ar_21_quant, Ar_24_quant, Ar_28_quant, Ar_31_quant, Ar_35_quant, Ar_38_quant)
Ar.TPM <- data.table(Ar.TPM)
setkey(Ar.TPM, Transcript, val)
Ar.TPM[val==3,val:=3.5]
Ar.TPM[val==10,val:=10.5]
Ar.TPM[val==17,val:=17.5]
Ar.TPM[val==24,val:=24.5]
Ar.TPM[val==31,val:=31.5]
Ar.TPM[val==38,val:=38.5]
head(Ar.TPM)
str(Ar.TPM)

tables()
```

Note that expression levels at each temperature treatment are highly correlated between the two colonies.

```{r exp_correlations, echo=FALSE, results='asis'}
# Make Rmarkdown pandoc table of correlations among expression levels between colonies at each temp

temp <- c(0, 3.5, 7, 10.5, 14, 17.5, 21, 24.5, 28, 31.5, 35, 38.5)
cors <- c(round(cor(Ar_0_quant$TPM, A22_0_quant$TPM), 2), round(cor(Ar_3_quant$TPM, A22_3_quant$TPM), 2), round(cor(Ar_7_quant$TPM, A22_7_quant$TPM), 2), round(cor(Ar_10_quant$TPM, A22_10_quant$TPM), 2), round(cor(Ar_14_quant$TPM, A22_14_quant$TPM), 2), round(cor(Ar_17_quant$TPM, A22_17_quant$TPM), 2), round(cor(Ar_21_quant$TPM, A22_21_quant$TPM), 2), round(cor(Ar_24_quant$TPM, A22_24_quant$TPM), 2), round(cor(Ar_28_quant$TPM, A22_28_quant$TPM), 2), round(cor(Ar_31_quant$TPM, A22_31_quant$TPM), 2), round(cor(Ar_35_quant$TPM, A22_35_quant$TPM), 2), round(cor(Ar_38_quant$TPM, A22_38_quant$TPM), 2))

cortable <- cbind(temp, cors)

pandoc.table(cortable, style="rmarkdown", caption = "correlations between colonies at each temperature treatment")
```

## Identification of thermally-responsive genes

To identify transcripts (roughly equivalent to genes) that show thermal responsiveness, I fit the following linear model to each transcript:

$$ log(TPM + 1) = \beta_0 + \beta_1(colony) + \beta_2(temp) + \beta_3(temp^2) + \beta_4(colony * temp) + \beta_5(colony * temp^2) + \epsilon $$

where TPM is transcripts per million. 



**Preliminary [examination](https://minilims1.uvm.edu/BCProject-26-Cahan/methods.html#clustering-of-samples) of the data indicated that the A22_7 and Ar_7 samples may have been switched, so I remove these values from the analysis.** 

```{r RxN, echo=TRUE, eval=TRUE, cache=TRUE}
A22.TPM[,colony:="A22"]
Ar.TPM[,colony:="Ar"]
TPM.dt <- rbind(A22.TPM, Ar.TPM)
TPM.dt$colony <- as.factor(TPM.dt$colony)
str(TPM.dt)

setkey(TPM.dt, val)
TPM.dt.sub <- TPM.dt[val != 7] 
unique(TPM.dt.sub$val)
```

(1) Identify transcripts with overall significant model fit. Adjust *P* values for multiple testing using FDR and retain transcripts with FDR < 0.05. Use log-transformed response to account for outliers.


```{r modp}
# define model for RxN function
model <-  "log(TPM+1) ~ colony + val + I(val^2) + colony:val + colony:I(val^2)"

# define function for `ddply` to calculate overall model P value
modpFunc <- function(foo) {
  # fit lm
  lmout <- eval(parse(text = paste("lm(", model, ", data = foo)", sep = "")))
  
  # calculate overall model significance
  f <- summary(lmout)$fstatistic
  pval <- unname(pf(f[1],f[2],f[3],lower.tail=F))
  attributes(pval) <- NULL
  
  # return pvalue
  c(pval = pval) 
}

RxNpval <- ddply(TPM.dt.sub, .(Transcript), .progress="text", .inform="TRUE", modpFunc)
```

Of the `r nrow(RxNpval)` transcripts, `r length(which(RxNpval$pval < 0.05))` have models with P < 0.05.

Many of these are likely false positives, so I adjust P-values using false discovery rate (FDR). Only those transcripts with less than 5% FDR are retained as significant. 


```{r padjust}
RxNpval$padj <- p.adjust(RxNpval$pval, method = "fdr")
# Plot FDR values against initial pvalues
par(mfrow = c(2,1))
hist(RxNpval$pval)
hist(RxNpval$padj)

# subset to significant transcripts
signif.transcripts <- RxNpval[which(RxNpval$padj < 0.05), ]

# extract significant transcripts
setkey(TPM.dt.sub, Transcript)
sig.TPM.dt.sub <- TPM.dt.sub[signif.transcripts$Transcript]
```

At the 5% FDR significance threshold, there are `r nrow(signif.transcripts)` transcripts with an overall significant model.


(2) Fit linear model to overall significant transcripts; perform stepAIC to retain only significant terms, and save `lm` output to list

```{r lmFunc}
# first, define function to use
lmFunc <- function(bar) {
  # fit lm
  lmout <- eval(parse(text = paste("lm(", model, ", data = bar)", sep = "")))  
  # stepAIC to drop non-significant terms
  f.lmout <- stepAIC(lmout)
  # return final model
  return(f.lmout) 
}

# apply function to all responsive transcripts
RxNlmAIC <- dlply(sig.TPM.dt.sub, .(Transcript), .progress="text", lmFunc)
```


## Thermally-responsive transcripts

The set of transcripts with significant expression patterns include those with expression that differs by colony, temperature and the interaction of colony and temperature. In this section, I am specifically interested in the thermally-responsive transcripts (temperature and colony x temperature) so I subset the significant transcripts to examine these. 

```{r responsive_transcripts}
grepFunc <- function(lmitem, term = NA) {
  coefs <- coefficients(lmitem)
  coef.names <- names(coefs)
  keep <- if(length(grep(term, coef.names)) > 0) TRUE else FALSE
}

interaction.lms <- RxNlmAIC[which(Map(grepFunc, RxNlmAIC, term = "colonyAr:") == TRUE)]
other.lms <- RxNlmAIC[setdiff(names(RxNlmAIC), names(interaction.lms))]
temperature.lms <- other.lms[which(Map(grepFunc, other.lms, term = "val") == TRUE)]
colony.lms <- other.lms[setdiff(names(other.lms), names(temperature.lms))]
responsive.lms <- c(temperature.lms, interaction.lms)
rm(other.lms)
```

```{r table_results, echo=FALSE, results='asis'}
clist <- c('Total', 'Colony', 'Temperature', 'Temperature:Colony')
siglist <- c(length(RxNlmAIC),
             length(colony.lms),
             length(temperature.lms),
             length(interaction.lms)
             )

sigtable <- data.frame(Coefficient = clist, "Number significant" = siglist)

pandoc.table(sigtable, style="rmarkdown", caption = "Number of transcripts (of 105,536 total) at 5% FDR with expression that depends on colony, temperature or their interaction.")
```


### Thermal-response functional types ###

The previous section identified the transcripts with thermally-responsive expression. In this section, I determine the shape of the expression response to temperature for each transcript. Catego
ries of expression response are:

* High - increase expression with temperature
* Low - decrease expression with temperature
* Intermediate - maximum expression at intermediate temperatures (14 - 28C)
* Bimodal - expressed greater than two standard deviations of expression at both low and high temperatures

I do this first for the thermally-responsive transcripts where there is no interaction with colony. For the transcripts where thermal-responsive expression depends on colony, I determine the functional type of the expression response separately for each colony. 


```{r expression_shape, eval=TRUE, echo=TRUE}
# calculate response type for responsive transcripts
interaction.response.type <- ldply(interaction.lms, .progress = "text", .inform = TRUE, RxNtype)
stopifnot(nrow(interaction.response.type) == length(interaction.lms))

# calculate response types for transcripts without interactions
temperature.response.type <- ldply(temperature.lms, .progress = "text", .inform = TRUE, RxNtype)
stopifnot(nrow(temperature.response.type) == length(temperature.lms))

# merge results
Ap.response.type <- rbind(interaction.response.type, temperature.response.type)
colnames(Ap.response.type)[which(colnames(Ap.response.type) == ".id")] <- "Transcript"
str(Ap.response.type)

# save results to file
write.table(file = paste(resultsdir, "Ap_responsive_transcripts.txt", sep = ""), Ap.response.type, row.names = FALSE, col.names = TRUE, quote = FALSE, sep = "\t")
```

Examine transcripts by temperature response. Compare between colonies.

```{r temperature_response, eval=TRUE, echo=FALSE, results='asis'}
A22.type.table <- table(Ap.response.type[ , 'A22.type'])
Ar.type.table <- table(Ap.response.type[ , 'Ar.type'])

type.table <- table(Ap.response.type[ , 'A22.type'], Ap.response.type[ , 'Ar.type'])
# reorder for plotting
tt2 <- type.table[c(2,4,1,3,5), c(2,4,1,3,5)]

png(file = paste(resultsdir, "responsive_transcripts_mosaicplot.png", sep = ""), width = 600, height = 480)
mosaicplot(tt2, xlab = "ApVT", ylab = "AcNC", cex = 0.7, las = 1, main = "")
dev.off()

pandoc.table(tt2, style="rmarkdown", caption = "Number of transcripts with maximum expression at high, low, intermediate, both high and low (bimodal) temperatures or are not thermally-responsivefor each colony and their overlap.")

# Pearson Chi-square test
chisq.test(Ap.type.table)
```


Table 4 shows the number of transcripts that fall into each expression type for each each colony. The totals for each colony include the `r nrow(temperature.lms)` transcripts that have consistent temperature responses between the two colonies. 



## Colony-level comparison ##

In this section, I compare the thermal reactionome between the *Ar* and *A22* colonies. 

### Plasticity versus constitutive expression

Selection may have acted in response to thermal stress such that some genes are constitutively activated in one colony while plastically-expressed in the other. I evaluate this hypothesis by comparing expression levels at the at optimum (19.5C) temperature between the two colonies for genes in that are either in the 'High' or 'Low' expression group in the other colony. Specifically, I expect that genes upregulated at high temperatures in *A22* are more highly expressed at 19.5C in *Ar*? Conversely, I expect genes that are upregulated at low temps in *Ar* to have greater constitutive expression in *A22*? 

```{r optimum_expression_comparison, cache=TRUE, eval=FALSE}
# list of transcripts that are 'high' expressed in A22
A22.high.transcripts <- Ap.response.type[which(Ap.response.type$A22.type == "High"), ]

# Compare expression at optimum temp (A22.opt) between colonies using t-test
t.test(A22.high.transcripts$A22.opt, A22.high.transcripts$Ar.opt)
boxplot(A22.high.transcripts$A22.opt, A22.high.transcripts$Ar.opt)

# T test on log-transformed values to control for outliers
t.test(log(A22.high.transcripts$A22.opt+1), log(A22.high.transcripts$Ar.opt+1))
boxplot(log(A22.high.transcripts$A22.opt+1), log(A22.high.transcripts$Ar.opt+1))
```

The `t.test` fails to account for the many orders of magnitude difference in expression among transcripts, e.g. non-equal variances. This problem is the key issue in the analysis of differential expression `r citep(c("10.1186/1471-2105-11-94", "10.1038/nprot.2013.099")`. As my goal is simply to determine if expression is typically greater at 19.25C in *Ar* than *A22* for genes that are up-regulated at high temperatures in *A22*, I use a non-parametric Wilcoxon signed rank-test

```{r A22_high_wilcoxon}
(w1 <- wilcox.test(A22.high.transcripts$A22.opt, A22.high.transcripts$Ar.opt, alternative = "two.sided", paired = TRUE, conf.int = TRUE))
```

Yes - transcripts that are up-regulated at high temperatures in *A22* are more highly-expressed at 19.25°C in *Ar*. Note that A22 had the larger library size so if this was due to TPM not correctly accounting for differences in reads between samples, we would expect to see a positive instead of negative value here.

Next I test the converse, that transcripts that are up-regulated at low temperatures in *Ar* are more highly-expressed at 19.25C temperatures in *A22*. 

```{r Ar_low_wilcoxon}
# list of transcripts that are 'high' expressed in Ar
Ar.low.transcripts <- Ap.response.type[which(Ap.response.type$Ar.type == "Low"), ]

# compare expression at optimum temp (Ar.opt) between colonies using t-test
t.test(Ar.low.transcripts$A22.opt, Ar.low.transcripts$Ar.opt)
boxplot(Ar.low.transcripts$A22.opt, Ar.low.transcripts$Ar.opt)
# t-test with log values
t.test(log(Ar.low.transcripts$A22.opt+1), log(Ar.low.transcripts$Ar.opt+1))
boxplot(log(Ar.low.transcripts$A22.opt+1), log(Ar.low.transcripts$Ar.opt+1))

# Wilcoxon signed rank-test
(w2 <- wilcox.test(Ar.low.transcripts$A22.opt, Ar.low.transcripts$Ar.opt, alternative = "two.sided", paired = TRUE, conf.int = TRUE))
```

Counter to expectations, expression at 19.5C is also greater in *Ar* than *A22* for transcripts upregulated at low temperatures in *Ar*. 

To confirm that there are not sample-level issues, I performed the same comparison using transcripts where I do *not* expect to see a difference in expression.

```{r intermediate_transcripts}
# list of transcripts that are 'Intermediate' expressed in Ar
Ar.int.transcripts <- Ap.response.type[which(Ap.response.type$Ar.type == "Intermediate"), ]
# T test
t.test(Ar.int.transcripts$A22.opt, Ar.int.transcripts$Ar.opt)
t.test(log(Ar.int.transcripts$A22.opt+1), log(Ar.int.transcripts$Ar.opt+1))
# Wilcoxon signed rank-test
wilcox.test(Ar.int.transcripts$A22.opt, Ar.int.transcripts$Ar.opt, alternative = "two.sided", paired = TRUE, conf.int = TRUE)

# list of transcripts that are 'Intermediate' expressed in A22
A22.int.transcripts <- Ap.response.type[which(Ap.response.type$A22.type == "Intermediate"), ]
# T test
t.test(A22.int.transcripts$A22.opt, A22.int.transcripts$Ar.opt)
t.test(log(A22.int.transcripts$A22.opt+1), log(A22.int.transcripts$Ar.opt+1))
# Wilcoxon signed rank-test
wilcox.test(A22.int.transcripts$A22.opt, A22.int.transcripts$Ar.opt, alternative = "two.sided", paired = TRUE, conf.int = TRUE)
```

The non-parametric test for both comparisions also finds greater constitutive expression in *Ar* than *A22*.

The overall pattern that emerges from these 4 tests is that constitutive expression at optimal conditions of 19.5°C is overall greater for *Ar* than *A22*.


### Thermal tolerance indicated by region of constant gene expression ###

The 'Intermediate' expressed transcripts are core molecular processes that are expressed at non-stressful temperatures, and shut-off when the organism experiences thermall stress. We hypothesized that if the more southern *Ar* colony was more thermally-tolerant than *A22*, transcripts with 'Intermediate' expression (10-30°C) would be active across a wider range of temperatures. To test this with our data, we calculated the standard deviation of the expression function for each temperature transcript that was 'Intermediate' expressed in each colony.

```{r thermal_breadth}
# extract 'Intermediate' expressed transcripts for A22 colony
A22.int.lm <- RxNlmAIC[A22.int.transcripts$Transcript]
length(A22.int.lm)

# apply `transcriptSD` function to all transcripts
A22.int.sd <- unlist(Map(transcriptSD, A22.int.lm, colony = "A22"))
A22.int.sd <- data.frame(colony = "A22", exp_sd = A22.int.sd)

# repeat for Ar
Ar.int.lm <- RxNlmAIC[Ar.int.transcripts$Transcript]
Ar.int.sd <- unlist(Map(transcriptSD, Ar.int.lm, colony = "Ar"))
Ar.int.sd <- data.frame(colony = "Ar", exp_sd = Ar.int.sd)

# T-test comparing standard deviation of expression between colonies
(t1 <- t.test(Ar.int.sd$exp_sd, A22.int.sd$exp_sd, alternative = "two.sided"))
```

Consistent with our hypothesis, 'Intermediate' transcripts in *Ar* are expressed over a significantly wider range of temperatures than in *A22*. 

```{r plotthermalbreadth, echo=FALSE, eval=TRUE}
# prep data for plotting
Ap.int.sd <- rbind(A22.int.sd, Ar.int.sd)

# ggplot
g1 <- ggplot(Ap.int.sd, aes(x=exp_sd, fill=colony)) + geom_density(alpha=0.2, position="identity")
g1 + scale_y_continuous(name="Density") +
  scale_x_continuous(name=expression("Standard deviation of expression"))
```


```{r plot_thermal_breadth2, echo = FALSE, eval = TRUE}
# predict expression for these transcripts
int.predicted <- matrix(nrow = 0, ncol = 5)

# for A22
colony = "A22"

for(i in 1:length(A22.int.lm)) {
    lmitem <- A22.int.lm[[i]]
    pred.vals <- seq(from = 0, to = 38.5, by = 2.5)
    newdf <- data.frame(Transcript = names(A22.int.lm[i]), colony = rep(colony, length = length(pred.vals)), val = pred.vals)
    pout <- predict(lmitem, newdata=newdf)
    pout <- cbind(newdf, logTPM = pout) # note this is on log scale
    # back-transform from log-scale
    pout$pTPM <- exp(pout$logTPM)
    # change any negative values to zero
    pout$pTPM <- ifelse(pout$pTPM < 0, 0, round(pout$pTPM, 5))
    # concatenate
    int.predicted <- rbind(int.predicted, pout)
}

# for Ar
colony = "Ar"

for(i in 1:length(Ar.int.lm)) {
    lmitem <- Ar.int.lm[[i]]
    pred.vals <- seq(from = 0, to = 38.5, by = 2.5)
    newdf <- data.frame(Transcript = names(Ar.int.lm[i]), colony = rep(colony, length = length(pred.vals)), val = pred.vals)
    pout <- predict(lmitem, newdata=newdf)
    pout <- cbind(newdf, logTPM = pout) # note this is on log scale
    # back-transform from log-scale
    pout$pTPM <- exp(pout$logTPM)
    # change any negative values to zero
    pout$pTPM <- ifelse(pout$pTPM < 0, 0, round(pout$pTPM, 5))
    # concatenate
    int.predicted <- rbind(int.predicted, pout)
}

# change to data.table
int.predicted <- data.table(int.predicted)
# scale transcripts so can compare
int.predicted[,pTPM.scaled:=scale(pTPM), by = Transcript]
str(int.predicted)

# plot
intplot <- ggplot(int.predicted, aes(x=val, y=pTPM.scaled, group=Transcript)) + 
  geom_line() + 
  facet_grid(. ~ colony)
intplot
```

Plot shows all the intermediate transcripts, though not possible to tell that *Ar* transcripts have greater variance than *A22* transcripts.


### Thermal sensitivity indicated by response of bimodally-expressed transcripts ###

As the converse of the above hypothesis, a colony that is especially thermally-sensitive is likely to activate expression of molecular processes more quickly. We tested this using the same approach as for the 'Intermediate' transcripts, but using the inverse of the 'Bimodal' expressed transcripts. 


```{r calc_Bimodal_variance}
# extract 'Bimodal' expressed transcripts for A22 colony
A22.bim.lm <- RxNlmAIC[Ap.response.type[which(Ap.response.type$A22.type == "Bimodal"), "Transcript"]]
length(A22.bim.lm)

# apply `transcriptSD` function to all transcripts
A22.bim.sd <- unlist(Map(transcriptSD, A22.bim.lm, colony = "A22"))

# repeat for Ar 
Ar.bim.lm <- RxNlmAIC[Ap.response.type[which(Ap.response.type$Ar.type == "Bimodal"), "Transcript"]]
length(Ar.bim.lm)
Ar.bim.sd <- unlist(Map(transcriptSD, Ar.bim.lm, colony = "Ar"))

# t-test to compare standard deviation of 'Bimodal' transcripts between colonies
(t2 <- t.test(Ar.bim.sd, A22.bim.sd))
```

No difference in the standard deviation of expression for bimodally-expressed transcripts between colonies.


### Compare peak expression among colonies ###

Probability density function of peak expression for transcripts that differ in expression between *A22* and *Ar*

```{r max_exp_PDF}
# reshape data
Ap.df <- data.frame(Transcript = rep(Ap.response.type$Transcript, times = 2), colony = rep(c("ApVT", "ApNC"), each = length(Ap.response.type$Transcript)), max.val = c(Ap.response.type$A22.max, Ap.response.type$Ar.max))

png(paste(resultsdir, "PDF_expression_all.png", sep=""))
g3 <- ggplot(Ap.df, aes(x=max.val, fill=colony)) + 
  geom_density(alpha=0.2, position="identity") + 
  #scale_fill_manual(name = "Colony", values = c("white", "black")) + 
  scale_y_continuous(name="Density") +
  scale_x_continuous(name="Temperature of maximum expression")
suppressWarnings(print(g3))
dev.off()
```

For the transcripts that differed in thermal responsiveness due to temperature, was the difference primarily due to differences in the mean value of expression, slope, curvature of a higher order effect? To test this, I use the framework of `r citet("10.1086/675302")`:

- *Offset, O*: overall difference in the mean expression value across all temperatures
- *Slope, S*: difference in overall slope
- *Curvature, C*: average difference in curvature of the reaction norm
- *Wiggle, W*: variability in shape not captured by the previous three measures

```{r offset, echo=TRUE, eval=TRUE}
varshape.out <- ldply(interaction.lms, .progress = "text", RxNvarshape)

# plot
boxplot(varshape.out$A22.mean, varshape.out$Ar.mean)
boxplot(log(varshape.out$A22.mean), log(varshape.out$Ar.mean))

# calculate differences in mean, slope, curvature and wiggle between colonies for each transcript
# offset = mean1 - mean2
varshape.out$offset <- varshape.out$A22.mean - varshape.out$Ar.mean
# to compare among transcripts that differ in scale of expression, standardize by overall mean for each transcript
varshape.out$s.offset <- NA
for(i in 1:nrow(varshape.out)) {
  varshape.out[i, "s.offset"] <- (varshape.out[i, "offset"] / mean(c(varshape.out[i, "A22.mean"], varshape.out[i, "Ar.mean"])))
}

# t-test
(t3 <- t.test(varshape.out$s.offset, alternative = "two.sided"))

# slope = slope1 - slope2
varshape.out$slope <- varshape.out$A22.slope - varshape.out$Ar.slope
# to compare among transcripts that differ in scale of expression, standardize by overall mean for each transcript
varshape.out$s.slope <- NA
for(i in 1:nrow(varshape.out)) {
  varshape.out[i, "s.slope"] <- (varshape.out[i, "slope"] / mean(c(varshape.out[i, "A22.slope"], varshape.out[i, "Ar.slope"])))
}
# for transcripts where both samples had slope=0, change s.slope from Inf to 0 as there is truly no difference in slope
varshape.out[which(varshape.out$Ar.slope == 0 & varshape.out$A22.slope == 0), "s.slope"] <- 0

# t-test
(t4 <- t.test(varshape.out$s.slope, alternative = "two.sided"))


# curvature = curvature1 - curvature2
varshape.out$curvature <- varshape.out$A22.curve - varshape.out$Ar.curve
# to compare among transcripts that differ in scale of expression, standardize by overall mean for each transcript
varshape.out$s.curvature <- NA
for(i in 1:nrow(varshape.out)) {
  varshape.out[i, "s.curvature"] <- (varshape.out[i, "curvature"] / mean(c(varshape.out[i, "A22.curve"], varshape.out[i, "Ar.curve"])))
}
# change 'Inf' values to 0
varshape.out[which(varshape.out$Ar.curve == 0 & varshape.out$A22.curve == 0), "s.curvature"] <- 0
# t-test
(t5 <- t.test(varshape.out$s.curvature, alternative = "two.sided"))


# wiggle
varshape.out$wiggle <- varshape.out$A22.wiggle - varshape.out$Ar.wiggle
varshape.out$s.wiggle <- NA
for(i in 1:nrow(varshape.out)) {
  varshape.out[i, "s.wiggle"] <- (varshape.out[i, "wiggle"] / mean(c(varshape.out[i, "A22.wiggle"], varshape.out[i, "Ar.wiggle"])))
}
# change 'Inf' values to 0
varshape.out[which(varshape.out$Ar.wiggle == 0 & varshape.out$A22.wiggle == 0), "s.wiggle"] <- 0
# note that only 407 transcripts have any 'wiggle'
length(which(varshape.out$s.wiggle != 0))

# t-test
(t6 <- t.test(varshape.out$s.wiggle, alternative = "two.sided"))
```

Next, I partition the differences in the reaction norms into the variation explained by differences in the trait means (offset), slope, curvature and wiggle.


```{r partition_reaction_norm_variation}
# weighted mean-standardized values of each measure
mean(abs((varshape.out$s.offset)))
mean(abs((varshape.out$s.slope)))
mean(abs((varshape.out$s.curvature)))
mean(abs((varshape.out$s.wiggle)))

# take absolute value of each value and sum to get total differences between reaction norms
varshape.out$s.total <- abs(varshape.out$s.offset) + abs(varshape.out$s.slope) + abs(varshape.out$s.curvature) + abs(varshape.out$s.wiggle)

t.test(varshape.out$s.total, alternative = "two.sided")

# variation in reaction norms due to differences in mean
varshape.out$prop.mean <- abs(varshape.out$s.offset) / varshape.out$s.total
varshape.out$prop.slope <- abs(varshape.out$s.slope) / varshape.out$s.total
varshape.out$prop.curve <- abs(varshape.out$s.curvature) / varshape.out$s.total
varshape.out$prop.wiggle <- abs(varshape.out$s.wiggle) / varshape.out$s.total

# Mean proportion and 95% CI of total variation of each measure
mean(varshape.out$prop.mean)
quantile(varshape.out$prop.mean, probs = c(0.05, 0.5, 0.95))

mean(varshape.out$prop.slope)
quantile(varshape.out$prop.slope, probs = c(0.05, 0.5, 0.95))

mean(varshape.out$prop.curve)
quantile(varshape.out$prop.curve, probs = c(0.05, 0.5, 0.95))

mean(varshape.out$prop.wiggle)
quantile(varshape.out$prop.wiggle, probs = c(0.05, 0.5, 0.95))
```



## Functional annotation ##

In the previous section, I identified transcripts that show significant responses in expression. Next, I add gene annotation and ontology information to these transcripts.  

```{r annotation}
setkey(annotationtable, Sequence.Name)
signif.transcripts <- data.table(signif.transcripts)
setkey(signif.transcripts, Transcript)
signif.transcripts <- annotationtable[signif.transcripts]
setnames(signif.transcripts, "Sequence.Name", "Transcript")
```

```{r echo=FALSE, results='hide'}
responsive.lms.ann.type <- signif.transcripts[names(responsive.lms)]
# combine responsive.lms.ann with "type" information
responsive.lms.ann.type <- merge(responsive.lms.ann.type, Ap.response.type[,c("Transcript", "A22.type", "Ar.type")], by = "Transcript", all = TRUE)
str(responsive.lms.ann.type)
write.csv(responsive.lms.ann.type, file = "results/Ap_responsive_genes_20140409.csv", row.names = FALSE)
```


## Gene set enrichment analysis ##

I perform gene set enrichment analysis below, but a quick `grep` shows that there are 26 transcripts with GO term "response to stress", though this is not enriched compared to the frequency of this term in the full dataset.
  
```{r GO_contingency_test}
# GO 'response to stress' hits in responsive transcripts
GO0006950.responsive <- responsive.lms.ann.type[grep("GO:0006950", responsive.lms.ann.type$GO.Biological.Process), list(Transcript, best.hit.to.nr, A22.type, Ar.type)]

# in high category
GO0006950.responsive[union(with(GO0006950.responsive, grep("High", Ar.type)), with(GO0006950.responsive, grep("High", A22.type))), ]
# in low category
GO0006950.responsive[union(with(GO0006950.responsive, grep("Low", Ar.type)), with(GO0006950.responsive, grep("Low", A22.type))), ]
# in bimodal category
GO0006950.responsive[union(with(GO0006950.responsive, grep("Bimodal", Ar.type)), with(GO0006950.responsive, grep("Bimodal", A22.type))), ]
# in intermediate category
GO0006950.responsive[union(with(GO0006950.responsive, grep("Intermediate", Ar.type)), with(GO0006950.responsive, grep("Intermediate", A22.type))), ]

# Chi-square test to see if 'response to stress' related genes overrepresented in responsive.lms compared to full list
resp.stress.responsive.count <- nrow(responsive.lms.ann.type[grep("GO:0006950", responsive.lms.ann.type$GO.Biological.Process), list(Transcript, best.hit.to.nr)])
# GO 'response to stress' hits in all transcripts
resp.stress.all.count <- nrow(annotationtable[grep("GO:0006950", annotationtable$GO.Biological.Process), list(Sequence.Name, best.hit.to.nr)])

GO.stress.table <- matrix(rbind(resp.stress.responsive.count, nrow(responsive.lms.ann.type) - resp.stress.responsive.count, resp.stress.all.count, nrow(annotationtable) - resp.stress.all.count), nrow = 2)

GO.stress.Xsq <- chisq.test(GO.stress.table)
GO.stress.Xsq
```




There are also 6 heat shock related genes in the responsive transcripts, out of 130 total.

```{r grep_shock}
hsp_responsive <- responsive.lms.ann.type[grep("shock", responsive.lms.ann.type$best.hit.to.nr), list(Transcript, best.hit.to.nr, A22.type, Ar.type)]
hsp_responsive

hsp_all <- annotationtable[grep("shock", annotationtable$best.hit.to.nr), list(Sequence.Name, best.hit.to.nr)]
nrow(hsp_all)
```



I use [topGO](http://www.bioconductor.org/packages/2.12/bioc/html/topGO.html) to perform gene set enrichment analysis (GSEA) seperately for each expression type (bimodal, intermediate, high, low).

First need to create gene ID to GO term map file

```{r geneid2go, echo=TRUE, eval=TRUE, cache=TRUE}
# create geneid2go.map file from FastAnnotator AnnotationTable.txt
geneid2GOmap(annotationfile)
```

then read map file.

```{r readmap, echo=TRUE, eval=TRUE}
# read mappings file
geneID2GO <- readMappings(file = "geneid2go.map")
str(head(geneID2GO))
```

### GSEA for thermally-responsive transcripts ###

Using this gene2GO map file, perform GSEA for:

**1) all responsive transcripts**

Use `selectFDR` function to select transcripts with adjusted P < 0.05.

```{r gsea_all}
# create geneList. note that NA values cause problems with topGO
# so set any NA to 1 as need to retain for GO analysis
Ap.geneList <- RxNpval$padj
Ap.geneList[which(is.na(Ap.geneList))] <- 1
stopifnot(length(which(is.na(Ap.geneList))) == 0)
names(Ap.geneList) <- RxNpval$Transcript
str(Ap.geneList)

# Function to select top genes (defined above)
selectFDR <- function(padj) {
    return(padj < 0.05)
}

# create topGOdata object
Ap.BP.GOdata <- new("topGOdata",
                 description = "BP gene set analysis", ontology = "BP",
                 allGenes = Ap.geneList, geneSel = selectFDR,
                 nodeSize = 10,
                 annot = annFUN.gene2GO, gene2GO = geneID2GO)

Ap.BP.GOdata

# perform enrichment analysis using parentchild method
Ap.BP.resultParentChild <- runTest(Ap.BP.GOdata, statistic = 'fisher', algorithm = 'parentchild')
Ap.BP.resultParentChild

# table results
Ap.BP.ResTable <- GenTable(Ap.BP.GOdata, parentchild = Ap.BP.resultParentChild, topNodes = 113)
dim(Ap.BP.ResTable)
#pandoc.table(Ap.BP.ResTable)

# graph significant nodes

#pdf(paste(resultsdir, "Ap.BP_topGO_sig_nodes.pdf", sep=""))
#showSigOfNodes(Ap.BP.GOdata, score(Ap.BP.resultParentChild), firstSigNodes = 10, useInfo = 'all')
#dev.off()
```

**2) High**

Genes with *High* expression in both colonies

Use `selectTranscript` function to select transcripts from 'Ap.response.type' for GSEA.

```{r gsea_high}
# Select transcripts
Ap.high <- Ap.response.type[which(Ap.response.type$Ar.type == "High" & Ap.response.type$A22.type == "High"), "Transcript"]
Ap.geneList.high <- rep(0, length = length(Ap.geneList))
names(Ap.geneList.high) <- names(Ap.geneList)
Ap.geneList.high[(which(names(Ap.geneList.high) %in% Ap.high))] <- 1
# check correct number of values set to 1
table(Ap.geneList.high)

# Run GSEA
Ap.high.gsea <- gsea(genelist = Ap.geneList.high, geneID2GO = geneID2GO, plotpath = NA)
#pandoc.table(Ap.high.gsea)
# output for Revigo
write.table(Ap.high.gsea[,"GO.ID"], file = "results/Ap_high_gsea.txt", row.names = FALSE, sep = "\t", quote = FALSE)
```


**3) Low**

Genes with *Low* expression in both colonies

```{r gsea_low}
# Select transcripts
Ap.low <- Ap.response.type[which(Ap.response.type$Ar.type == "Low" & Ap.response.type$A22.type == "Low"), "Transcript"]
Ap.geneList.low <- rep(0, length = length(Ap.geneList))
names(Ap.geneList.low) <- names(Ap.geneList)
Ap.geneList.low[(which(names(Ap.geneList.low) %in% Ap.low))] <- 1
# check correct number of values set to 1
table(Ap.geneList.low)

# Run GSEA
Ap.low.gsea <- gsea(genelist = Ap.geneList.low, geneID2GO = geneID2GO)
#pandoc.table(Ap.low.BP.ResTable)
write.table(Ap.low.gsea[,"GO.ID"], file = "results/Ap_low_gsea.txt", row.names = FALSE, sep = "\t", quote = FALSE)
```


**4) Bimodal**

Genes with *Bimodal* expression in both colonies

```{r gsea_bimodal}
Ap.bim <- Ap.response.type[which(Ap.response.type$A22.type == "Bimodal" & Ap.response.type$Ar.type == "Bimodal"), "Transcript"]
# create gene list, setting value to 1 for "bim" transcripts
Ap.geneList.bim <- rep(0, length = length(Ap.geneList))
names(Ap.geneList.bim) <- names(Ap.geneList)
Ap.geneList.bim[(which(names(Ap.geneList.bim) %in% Ap.bim))] <- 1
# check correct number of values set to 1
table(Ap.geneList.bim)

# Run GSEA
Ap.bim.gsea <- gsea(genelist = Ap.geneList.bim, geneID2GO = geneID2GO)
#pandoc.table(Ap.bim.gsea)
write.table(Ap.bim.gsea[,"GO.ID"], file = "results/Ap_bim_gsea.txt", row.names = FALSE, sep = "\t", quote = FALSE)
```


**5) Intermediate**

Genes with *Intermediate* expression in both colonies

```{r gsea_intermediate}
Ap.int <- Ap.response.type[which(Ap.response.type$A22.type == "Intermediate" & Ap.response.type$Ar.type == "Intermediate"), "Transcript"]
# create gene list, setting value to 1 for "int" transcripts
Ap.geneList.int <- rep(0, length = length(Ap.geneList))
names(Ap.geneList.int) <- names(Ap.geneList)
Ap.geneList.int[(which(names(Ap.geneList.int) %in% Ap.int))] <- 1
# check correct number of values set to 1
table(Ap.geneList.int)

# Run GSEA
Ap.int.gsea <- gsea(genelist = Ap.geneList.int, geneID2GO = geneID2GO)
#pandoc.table(Ap.int.gsea)
write.table(Ap.int.gsea[,"GO.ID"], file = "results/Ap_int_gsea.txt", row.names = FALSE, sep = "\t", quote = FALSE)
```

```{r gsea_table, echo=FALSE, eval=TRUE, results='asis'}
## merge results into single table
Ap.high.gsea$Type <- "High"
Ap.low.gsea$Type <- "Low"
Ap.bim.gsea$Type <- "Bimodal"
Ap.int.gsea$Type <- "Intermediate"
# combine
Ap.gsea <- rbind(Ap.high.gsea, Ap.low.gsea, Ap.bim.gsea, Ap.int.gsea)
# reorder
Ap.gsea1 <- Ap.gsea[,c("Type", "GO.ID", "Term", "Annotated", "Significant", "Expected", "parentchild")]
colnames(Ap.gsea1)[7] <- "P"
# display
pandoc.table(Ap.gsea1, split.tables = "Inf")
```


Next, I perform GSEA for genes in each functional type in one colony but not the other (e.g. the set difference) to gain insight on differences between the colonies.

**A22 'High' genes not in Ar**

```{r gsea_high_A22}
A22.high <- Ap.response.type[which(Ap.response.type$A22.type == "High" & Ap.response.type$Ar.type != "High"), "Transcript"]
# create gene list, setting value to 1 for "bim" transcripts
A22.geneList.high <- rep(0, length = length(Ap.geneList))
names(A22.geneList.high) <- names(Ap.geneList)
A22.geneList.high[(which(names(A22.geneList.high) %in% A22.high))] <- 1
# check correct number of values set to 1
table(A22.geneList.high)

# Run GSEA
A22.high.gsea <- gsea(genelist = A22.geneList.high, geneID2GO = geneID2GO)
#pandoc.table(A22.high.gsea)
```


**Ar 'High' genes not in A22**

```{r gsea_high_Ar}
Ar.high <- Ap.response.type[which(Ap.response.type$A22.type != "High" & Ap.response.type$Ar.type == "High"), "Transcript"]
# create gene list, setting value to 1 for "bim" transcripts
Ar.geneList.high <- rep(0, length = length(Ap.geneList))
names(Ar.geneList.high) <- names(Ap.geneList)
Ar.geneList.high[(which(names(Ar.geneList.high) %in% Ar.high))] <- 1
# check correct number of values set to 1
table(Ar.geneList.high)

# Run GSEA
Ar.high.gsea <- gsea(genelist = Ar.geneList.high, geneID2GO = geneID2GO)
```


**A22 'Low' genes not in Ar**

```{r gsea_low_A22}
A22.low <- Ap.response.type[which(Ap.response.type$A22.type == "Low" & Ap.response.type$Ar.type != "Low"), "Transcript"]
# create gene list, setting value to 1 for "bim" transcripts
A22.geneList.low <- rep(0, length = length(Ap.geneList))
names(A22.geneList.low) <- names(Ap.geneList)
A22.geneList.low[(which(names(A22.geneList.low) %in% A22.low))] <- 1
# check correct number of values set to 1
table(A22.geneList.low)

# Run GSEA
A22.low.gsea <- gsea(genelist = A22.geneList.low, geneID2GO = geneID2GO)
```


**Ar 'Low' genes not in A22**

```{r gsea_low_Ar}
Ar.low <- Ap.response.type[which(Ap.response.type$A22.type != "Low" & Ap.response.type$Ar.type == "Low"), "Transcript"]
# create gene list, setting value to 1 for "bim" transcripts
Ar.geneList.low <- rep(0, length = length(Ap.geneList))
names(Ar.geneList.low) <- names(Ap.geneList)
Ar.geneList.low[(which(names(Ar.geneList.low) %in% Ar.low))] <- 1
# check correct number of values set to 1
table(Ar.geneList.low)

# Run GSEA
Ar.low.gsea <- gsea(genelist = Ar.geneList.low, geneID2GO = geneID2GO)
```



**A22 'Bimodal' genes not in Ar**

```{r gsea_bim_A22}
A22.bim <- Ap.response.type[which(Ap.response.type$A22.type == "Bimodal" & Ap.response.type$Ar.type != "Bimodal"), "Transcript"]
# create gene list, setting value to 1 for "bim" transcripts
A22.geneList.bim <- rep(0, length = length(Ap.geneList))
names(A22.geneList.bim) <- names(Ap.geneList)
A22.geneList.bim[(which(names(A22.geneList.bim) %in% A22.bim))] <- 1
# check correct number of values set to 1
table(A22.geneList.bim)

# Run GSEA
A22.bim.gsea <- gsea(genelist = A22.geneList.bim, geneID2GO = geneID2GO)
```


**Ar 'Bimodal' genes not in A22**

```{r gsea_bim_Ar}
Ar.bim <- Ap.response.type[which(Ap.response.type$A22.type != "Bimodal" & Ap.response.type$Ar.type == "Bimodal"), "Transcript"]
# create gene list, setting value to 1 for "bim" transcripts
Ar.geneList.bim <- rep(0, length = length(Ap.geneList))
names(Ar.geneList.bim) <- names(Ap.geneList)
Ar.geneList.bim[(which(names(Ar.geneList.bim) %in% Ar.bim))] <- 1
# check correct number of values set to 1
table(Ar.geneList.bim)

# Run GSEA
Ar.bim.gsea <- gsea(genelist = Ar.geneList.bim, geneID2GO = geneID2GO)
```



**A22 'Intermediate' genes not in Ar**

```{r gsea_int_A22}
A22.int <- Ap.response.type[which(Ap.response.type$A22.type == "Intermediate" & Ap.response.type$Ar.type != "Intermediate"), "Transcript"]
# create gene list, setting value to 1 for "bim" transcripts
A22.geneList.int <- rep(0, length = length(Ap.geneList))
names(A22.geneList.int) <- names(Ap.geneList)
A22.geneList.int[(which(names(A22.geneList.int) %in% A22.int))] <- 1
# check correct number of values set to 1
table(A22.geneList.int)

# Run GSEA
A22.int.gsea <- gsea(genelist = A22.geneList.int, geneID2GO = geneID2GO)
```


**Ar 'Intermediate' genes not in A22**

```{r gsea_int_Ar}
Ar.int <- Ap.response.type[which(Ap.response.type$A22.type != "Intermediate" & Ap.response.type$Ar.type == "Intermediate"), "Transcript"]
# create gene list, setting value to 1 for "bim" transcripts
Ar.geneList.int <- rep(0, length = length(Ap.geneList))
names(Ar.geneList.int) <- names(Ap.geneList)
Ar.geneList.int[(which(names(Ar.geneList.int) %in% Ar.int))] <- 1
# check correct number of values set to 1
table(Ar.geneList.int)

# Run GSEA
Ar.int.gsea <- gsea(genelist = Ar.geneList.int, geneID2GO = geneID2GO)
```


Make a table with results.

```{r gsea_table_2, echo=FALSE, eval=TRUE, results='asis'}
A22.high.gsea$Type <- "High"
A22.low.gsea$Type <- "Low"
A22.int.gsea$Type <- "Intermediate"
A22.bim.gsea$Type <- "Bimodal"
A22.gsea <- rbind(A22.high.gsea, A22.low.gsea, A22.int.gsea, A22.bim.gsea)
A22.gsea$Colony <- "ApVT"

Ar.high.gsea$Type <- "High"
Ar.low.gsea$Type <- "Low"
Ar.bim.gsea$Type <- "Bimodal"
Ar.int.gsea$Type <- "Intermediate"
Ar.gsea <- rbind(Ar.high.gsea, Ar.low.gsea, Ar.int.gsea, Ar.bim.gsea)
Ar.gsea$Colony <- "ApNC"

# combine
Ap.setdiff.gsea <- rbind(A22.gsea, Ar.gsea)
# reorder
Ap.setdiff.gsea <- Ap.setdiff.gsea[,c("Colony", "Type", "GO.ID", "Term", "Annotated", "Significant", "Expected", "parentchild")]
colnames(Ap.setdiff.gsea)[8] <- "P"

Ap.setdiff.gsea
```



## Visualize responsive transcripts

Organize data

```{r plot_data}
# extract TPM data for thermally-responsive transcripts
resp.TPM.dt.sub <- TPM.dt.sub[names(responsive.lms)]
setkey(resp.TPM.dt.sub, Transcript)
str(resp.TPM.dt.sub)
length(unique(resp.TPM.dt.sub$Transcript))
# scale transcripts so can compare
resp.TPM.dt.sub[,TPM.scaled:=scale(TPM), by = Transcript]
```

Make plots for all genes expressed at *High* temps in GO category "GO:0006950: response to stress"

```{r plot_GOstress, echo=FALSE, eval=FALSE, cache=TRUE}
high.stress.terms <- GO0006950.responsive[intersect(with(GO0006950.responsive, grep("High", Ar.type)), with(GO0006950.responsive, grep("High", A22.type))), ]


# in low category
low.genes <- GO0006950.responsive[union(with(GO0006950.responsive, grep("Low", Ar.type)), with(GO0006950.responsive, grep("Low", A22.type))), ]
# in bimodal category
bim.genes <- GO0006950.responsive[union(with(GO0006950.responsive, grep("Bimodal", Ar.type)), with(GO0006950.responsive, grep("Bimodal", A22.type))), ]
# in intermediate category
int.genes <- GO0006950.responsive[union(with(GO0006950.responsive, grep("Intermediate", Ar.type)), with(GO0006950.responsive, grep("Intermediate", A22.type))), ]



# Line plot, expression against temp, faceted by colony
trp1 <- ggplot(resp.TPM.dt.sub[high.stress.terms$Transcript], aes(x=val, y=TPM.scaled, group=Transcript)) + 
  geom_line() + 
  facet_grid(. ~ colony) + 
  scale_y_continuous(name="Expression (scaled)") +
  scale_x_continuous(name=expression(paste("Temperature ", degree, "C")))
print(trp1)

# Smooth plot, expression against temp, faceted by colony
png("results/GO_stress_High.png")
trp2 <- ggplot(resp.TPM.dt.sub[high.stress.terms$Transcript], aes(x=val, y=TPM.scaled, group=Transcript)) + 
  geom_smooth(method = "lm", formula = y ~ poly(x, 2)) + 
  facet_grid(. ~ colony) + 
  scale_y_continuous(name="Expression (scaled)") +
  scale_x_continuous(name=expression(paste("Temperature ", degree, "C")))
print(trp2)
dev.off()
```

Make plots for all genes expressed at *Low* temps in GO category "GO:0006950: response to stress"

```{r plot_GOstress_low, echo=FALSE, eval=FALSE, cache=TRUE}
low.stress.terms <- GO0006950.responsive[intersect(with(GO0006950.responsive, grep("Low", Ar.type)), with(GO0006950.responsive, grep("Low", A22.type))), ]

# Line plot, expression against temp, faceted by colony
trp3 <- ggplot(resp.TPM.dt.sub[low.stress.terms$Transcript], aes(x=val, y=TPM.scaled, group=Transcript)) + 
  geom_line() + 
  facet_grid(. ~ colony) + 
  scale_y_continuous(name="Expression (scaled)") +
  scale_x_continuous(name=expression(paste("Temperature ", degree, "C")))
print(trp3)

# Smooth plot, expression against temp, faceted by colony
trp4 <- ggplot(resp.TPM.dt.sub[low.stress.terms$Transcript], aes(x=val, y=TPM.scaled, group=Transcript)) + 
  geom_smooth(method = "lm", formula = y ~ poly(x, 2)) + 
  facet_grid(. ~ colony) + 
  scale_y_continuous(name="Expression (scaled)") +
  scale_x_continuous(name=expression(paste("Temperature ", degree, "C")))
print(trp4)
```


## Shiny interactive web-app

To assist visualization of specific transcripts, I made a interactive web-app using the [shiny](http://www.rstudio.com/shiny/) package. The scripts for this app are in the sub-directory `.\ApRxN-shinyapp`.

Export data for interactive shiny app. 

```{r shiny_file, eval=FALSE, echo=FALSE}
shiny.dt <- TPM.dt.sub[Ap.response.type$Transcript]

# scale expression values 
shiny.dt[,TPM.scaled:=scale(TPM), by = Transcript]
str(shiny.dt)
write.csv(shiny.dt, file = "ApRxN-shinyapp/responsive.lms.TPM.csv", quote = TRUE, row.names = FALSE)
```




```{r rData, echo=FALSE}
# save RData for manuscript file
save(tt2, Ap.df, w1, w2, file = "ms.RData")
```


## Session information ##

```{r session}
save.image()
sessionInfo()
```

## References

```{r references, results='asis'}
bibliography()
```
