---
title: Technical Report for **Thermal reactionomes reveal divergent responses to thermal
  extremes in warm and cool-climate ant species**
author: '[John Stanton-Geddes](john.stantongeddes.research@gmail.com)'
output:
  word_document: default
  pdf_document:
    fig_caption: yes
    includes:
      in_header: header.tex
    keep_tex: yes
    latex_engine: xelatex
    number_sections: no
    toc_depth: 2
  html_document:
    keep_md: yes
---

```{r setup, echo=FALSE, results='hide', message = FALSE, warning=FALSE}
# Global settings
options(stringsAsFactors=FALSE, digits = 3, scipen = 1)
options(width = 80)

# Load libraries
library(R.utils)
library(ggplot2)
library(knitr)
library(knitcitations)
library(pander)
library(stringr)
library(data.table)
library(RCurl)
library(plyr)
library(MASS)
library(xtable)
library(methods) # required by `knitcitations` when file is `knit` from command-line
library(reshape2)
library(RColorBrewer)
library(coin)
library(gridExtra)

# Load Bioconductor libraries
#source("http://bioconductor.org/biocLite.R")
#biocLite("topGO")
library(topGO) 

# load custom functions
source("scripts/RxNseq.R")

# knitr options
opts_chunk$set(cache=TRUE)

# directory to save results
resultsdir <- "results/"
```

## Summary ##
  
In this technical report, which accompanies the manuscript **Thermal reactionomes reveal divergent responses to thermal extremes in warm and cool-climate ant species**, we:

1. Describe the *de novo* assembly of the transcriptome for two ant species within the *Aphaenogaster rudis-picea-fulva* species complex `r citep("10.1155/2012/752815")`.
2. Measure the reaction norm across a thermal gradient (0°C to 38.5°C) for all transcripts and characterize the *thermal reactionome* as the set of transcripts that are thermally-responsive.
3. Quantitatively test three mechanistic hypotheses of thermal adapation:.
  - The *Enhanced response* hypothesis proposes that species extend their thermal limits through a stronger induced response to provide greater protection from more frequently encountered stressors.
  - The *Tolerance hypothesis* proposes that existing inducible stress responses become insufficient or prohibitively costly as environmental stressors increase in frequency.
  - The *Genetic assimilation hypothesis* proposes that exposure to more extreme stressors selects for a shift from inducible to constitutive expression of stress-response genes.
  
We quantified the relative importance of each of these mechanisms to thermal adaptation in these two species by evalutating differences in the reactionomes between species. We used gene set enrichment analysis to provide additional insight on the cellular processes underlying the transcriptome-wide changes in expression. 

## Data ##

The raw Illumina fastq files are available from the NCBI short read archive [link tbd]. The assembled transcriptome, annotation and expression values are available from the Harvard Forest Archives, HF-113 [http://harvardforest.fas.harvard.edu/data-archive] datasets [hf113-38](http://harvardforest.fas.harvard.edu/data/p11/hf113/hf113-38-ap-ann-table.txt), [hf113-41](http://harvardforest.fas.harvard.edu/data/p11/hf113/hf113-41-ap-trans.tar) and [hf113-42](http://harvardforest.fas.harvard.edu/data/p11/hf113/hf113-42-ap-gene-exp-quant.tgz). These processed files are used for this analysis due to the computational demands of re-generating the files, but the exact commands for each of these steps are documented below.


## Sample description ##

Two ant colonies were used for the transcriptome sequencing. The first, designated *A22*, was collected at Molly Bog, Vermont in August 2012 by Nick Gotelli and Andrew Nguyen. The second colony, designated *Ar*, was collected by Lauren Nichols in Raleigh, North Carolina. These colonies were maintained in the lab for 6 months prior to sample collection. Bernice DeMarco (Michigan State University) identified colony *A22* as *A. picea* and *Ar* as *A. carolinensis*. 

For each colony, three ants were exposed to one of 12 temperature treatments, every 3.5°C ranging from 0°C to 38.5°C, for one hour in glass tubes in a water bath. The ants were flash frozen and stored at -80C until RNA was extracted using a two step extraction; [RNAzol RT](http://www.mrcgene.com/rnazol.htm) (Molecular Research Center, Inc) followed by an [RNeasy Micro](http://www.qiagen.com/products/catalog/sample-technologies/rna-sample-technologies/total-rna/rneasy-micro-kit) column (Qiagen). Samples from each colony were pooled and sequenced in separate lanes on a 100bp paired-end run of an Illumina HiSeq at the University of Minnesota Genomics Center, yielding 20e6 and 16e6 reads for the A22 and Ar samples, respectively.


## Transcriptome assembly ##

The Illumina reads were filtered using the program [Trimmomatic](http://www.usadellab.org/cms/?page=trimmomatic) `r citep("10.1093/nar/gks540")` to remove Ilumina adapter sequences, trim bases with PHRED quality scores less than 15 in a 4 bp sliding window and remove final trimmed sequences with length less than 36 bp. The code used was 

~~~
java -jar trimmomatic-0.30.jar PE -threads 40 -phred33 -trimlog trimmomatic.log \\
sample.R1.fastq sample.R2.fastq sample.R1.trimmed.paired.fastq sample.R1.trimmed \\
.unpaired.fastq sample.R2.trimmed.paired.fastq sample.R2.trimmed.unpaired.fastq \\
ILLUMINACLIP:TruSeq3-PE.fa:2:30:10 LEADING:3 TRAILING:3 SLIDINGWINDOW:4:15 MINLEN:36
~~~

where "sample" was replaced by each sample. This filtering yielded 339,845,787 properly paired reads and 16,860,885 unpaired reads. 

Properly paired and unpaired reads passing the Trimmomatic filter were combined and used in de novo transcriptome assembly using the program [Trinity](http://trinityrnaseq.sourceforge.net/) `r citep("10.1038/nbt.1883")`. Unpaired reads resulting from Trimmomatic were concatenated to the set of left (R1) reads per Trinity usage documentation. Assembly on a single 40 CPU machine with 1TB of memory required about 20 hours of compute time.

~~~
Trinity.pl --seqType=fq --JM=90G —left=all.R1.trimmed.fastq —right=all.R2.trimmed.fastq \\
--output=trinity --CPU=40 --inchworm_cpu=40 --bflyCPU=5
~~~

This assembly contained 100,381 unique components (~genes) in 126,172 total transcripts (Table 1). 

As we were assembling two divergent colonies into a single transcriptome, we suspected that this assembly would be susceptible to known problems of errors during assembly (e.g. chimeric transcripts that are fusions of two transcripts) and redundancy `r citep("10.1186/1471-2164-14-328")`. To account for this, we performed two post-assembly processing steps.

First, we ran the program [cap3](http://seq.cs.iastate.edu/) `r citep("10.1101/gr.9.9.868")` setting the maximum gap length and band expansion size to 50 `-f 50 -a 50`, no end clipping as the reads were already filtered `k 0`, requiring 90% identity for assembly, and a minimum overlap length of 100 bp `-o 100`. The percent identity threshold of 90% was chosen to liberally collapse orthologous contigs from the two colonies that may have been assembled separately. 

    cap3 Trinity.fasta -f 50 -a 50 -k 0 -p 90 -o 100 > Trinity_cap3.out

The output of `cap3` gives assembled "contigs" and unassembled "singlets" that were concatenated into a single file.

    # check the number of contigs clustered
    grep -c "Contig" Trinity.fasta.cap.contigs
    grep -c "comp" Trinity.fasta.cap.singlets
    # compare to contigs from Trinity output
    grep -c "comp" Trinity.fasta

    # Combine contigs and singlets from CAP3
    cat Trinity.fasta.cap.contigs Trinity.fasta.cap.singlets > Trinity_cap3.fasta

The output file "Trinity.fasta.cap.info" gives specific information on which contigs were collapsed.

Subsequent to running `cap3`, we ran [uclust](http://drive5.com/usearch/manual/uclust_algo.html) to cluster sequences completely contained within longer sequences, again specificing a 90% identity cutoff for clustering. 

    # sort
    uclust --sort Trinity_cap3.fasta --output Trinity_cap3_sorted.fasta
    # cluster by 90% similarity threshold
    uclust --input Trinity_cap3_sorted.fasta --uc Trinity_cap3_uclust.out --id 0.90
    # convert uclust to fasta format
    uclust --uc2fasta Trinity_cap3_uclust.out --input Trinity_cap3_uclust.fasta

These post-processing step removed `r round((126172-105536)/126172 * 100, 0)`% of the initial reads (Table 1).

```{r assemstats, results = "asis", echo=FALSE}
# make table
Trinity <- c("126,172", "100,389,539", "358", "795", "16,201", "1,631")
Reduced <- c("105,536", "62,648,997", "320", "593", "15,491", "895")

assemstats <- rbind(Trinity, Reduced)
colnames(assemstats) <- c("Total contigs", "Total length", "Median contig size", "Mean contig size", "N50 contig", "N50 Length")

pandoc.table(assemstats, style="rmarkdown", caption = "Statistics for Trinity and cap3+uclust reduced transcriptome assemblies")
```

To remove contigs that are likely contaminants from bacterial, archael, virus or human sources, we used the program [DeconSeq](http://deconseq.sourceforge.net/) r citep("10.1371/journal.pone.0017288"). We downloaded the bacteria, virus, archae and human [databases of contaminants](ftp://edwards.sdsu.edu:7009/deconseq/db), modified the `DeconSeqConfig.pm` file as described [here](http://www.vcru.wisc.edu/simonlab/bioinformatics/programs/install/deconseq.htm) to point to the databases, and ran DeconSeq specifiying 95% identity over 50% the length of contig

    deconseq.pl -c 50 -i 95 -f Trinity_cap3_uclust.fasta -d Trinity_cap3_uclust \\
    -dbs hsref,bast,vir,arch
	
This resulted in removing 5,675 contigs as contaminants, leaving 99,861 "clean" contigs. We spot-checked the contaminants by BLAST and confirmed that they matched bacteria, human or viral sources by greater than 95%. For expression quantification, we use the full assembly to ensure that "contaminant" reads are assigned to the contaminants. After quantification, these transcripts will then be removed from further analyses.

Running Trinity and subsequent programs is time and memory-intensive so the final assembly is downloaded and used for all further analyses. The downloaded archive contains the "clean" and "contaminant" sequences after filtering with DeconSeq.

~~~
# download filtered Trinity assembly, uncompress and move
wget http://harvardforest.fas.harvard.edu/data/p11/hf113/hf113-40-ap-trans.tar

# move and extract
mkdir -p results/
mkdir -p results/trinity-full/
mv hf113-40-ap-trans.tar results/trinity-full/.
tar -xvf hf113-40-ap-trans.tar
~~~

To examine the species distribution of BLAST hits in the transcriptome assembly, I used the program [Krona](http://sourceforge.net/p/krona/home/krona/) r citep("doi:10.1186/1471-2105-12-385") which provides a taxonomic profile of all transcripts by BLAST annotation to species.

The interactive visualization is available [here](http://johnstantongeddes.org/assets/files/ApTranscriptome_FA.html).


## Transcriptome annotation ##

Annotation was performed by uploading the reduced assembly "Trinity_cap3_uclust.fasta" to the web-based annotation program [FastAnnotator](http://fastannotator.cgu.edu.tw/index.php).

Results were available as job ID [13894410176993](http://fastannotator.cgu.edu.tw/job.php?jobid=13894410176993#page=basicinfo).

This annotation file can be downloaded from the Harvard Forest [archive](http://harvardforest.fas.harvard.edu/data/p11/hf113/hf113-36-ap-ann-table.csv), or read directly into R.  

```{r download, echo = TRUE}
# URL for annotation file
annotation.URL <- getURL("http://johnstantongeddes.org/assets/files/ApTranscriptome/ApTranscriptome_AnnotationTable_20140113.txt")

# load 
annotation.file <- read.csv(textConnection(annotation.URL), header = TRUE, sep = "\t", stringsAsFactors = FALSE)
str(annotation.file)

# Convert to data.table
annotation.table <- data.table(annotation.file)
str(annotation.table)
```

## Identification of thermally-responsive genes ##

### Gene expression ###

I quantified gene expression using [sailfish](http://www.cs.cmu.edu/~ckingsf/software/sailfish/index.html). To run this program, first make sure that PATHs to the software libraries are set up correctly as described on the sailfish website. 

An index of the assembly is built with the command:                                                 

~~~
sailfish index -t results/trinity-full/Trinity_cap3_uclust.fasta -o results/trinity-full/sailfish-index-Trinity-cap3-uclust -k 20 -p 4
~~~

Once this is done, expression is quantified for the Trimmomatic filtered reads from each species-treatment sample separately. Note that for each sample, there are four filtered read files:

- paired.left.fastq
- paired.right.fastq
- unpaired.left.fastq
- unpaired.right.fastq

~~~
# make a directory for the expression values
mkdir -p results/trinity-full/sailfish-expression-Trinity-cap3-uclust
# change to "trinity-full" directory
cd results/trinity-full
# for each sample, run the following command
sailfish quant -i sailfish-index-Trinity-cap3-uclust -o sailfish-expression-Trinity-cap3-uclust/A22-0 -l "T=SE:S=U" -r A22-0_ATCACG.unpaired.left.fastq A22-0_ATCACG.unpaired.right.fastq A22-0_ATCACG.paired.left.fastq A22-0_ATCACG.paired.right.fastq -p 4
~~~
                                           
While it is possible to separately specify the paired-end and orphaned single-end reads in Sailfish v0.6.3, the results are exactly the same as if they are all entered as SE.	

These files are downloaded for convenience:

~~~
# download gene expression quantification
wget http://harvardforest.fas.harvard.edu/data/p11/hf113/hf113-41-ap-gene-exp-quant.tgz
# http://johnstantongeddes.org/assets/files/ApTranscriptome/sailfish_quant_20140916.tar
# check md5sum
md5sum sailfish_quant_20140916.tar
# 26102c7ef86cf30b5f8e923640378185

# move and extract
mkdir -p /results/trinity-full/sailfish-expression-Trinity-cap3-uclust
mv sailfish_quant_20140916.tar results/trinity-full/.
tar -xvf sailfish_quant_20140916.tar
~~~

For each sample, there is a directory containting a file *quant_bias_corrected.sf*. This file has the following columns, following a number of header lines:

1. Transcript ID
2. Transcript Length
3. Transcripts per Million (TPM): computed as described in `r citep("10.1093/bioinformatics/btp692")`, and is meant as an estimate of the number of transcripts, per million observed transcripts, originating from each isoform.
4. Reads Per Kilobase per Million mapped reads (RPKM): classic measure of relative transcript abundance, and is an estimate of the number of reads per kilobase of transcript (per million mapped reads) originating from each transcript.

The TPM column for each sample was extracted and combined into a matrix for each species.

**Preliminary [examination](https://minilims1.uvm.edu/BCProject-26-Cahan/methods.html#clustering-of-samples) of the data indicated that the A22_7 and Ar_7 samples may have been switched, so I remove these values from the combined expression data set for the two species.**  The 'Ar_7' sample clusters with all the other A22 samples, while the 'A22_7' sample clusters with all the other Ar samples. The parsimonious explanation for this is a labeling mistake. Removing these samples, rather than relabeling, is the conservative approach. 

![clustering](results/aphaenogaster_sample_clustering.png)


```{r load_expression_data, eval=TRUE, echo=FALSE, results='hide'}
# sailfish expression directory
sfexpressionroot <- "results/trinity-full/sailfish-expression-Trinity-cap3-uclust/"

samples <- c("A22-0", "A22-10", "A22-14", "A22-17", "A22-21", "A22-24", "A22-28", "A22-31", "A22-35", "A22-38", "A22-3", "A22-7", "Ar-0", "Ar-10", "Ar-14", "Ar-17", "Ar-21", "Ar-24", "Ar-28", "Ar-31", "Ar-35", "Ar-38", "Ar-3", "Ar-7")
    
# read in each file using loop
for (j in 1:length(samples)) {
    samp <- samples[j]
    trtval <- as.numeric(str_split_fixed(samp, "-", 2)[2])
    outpre <- gsub("-", "_", samp)
    outname <- paste(outpre, "_quant", sep="")
    read.sailfish.quant(filein=paste(sfexpressionroot, samp, "_quant/quant_bias_corrected.sf", sep=""), outname=outname, samp = samp, trtval = trtval)
}

# combine into long format data.table
A22.TPM <- data.table(rbind(A22_0_quant, A22_3_quant, A22_7_quant, A22_10_quant, A22_14_quant, A22_17_quant, A22_21_quant, A22_24_quant, A22_28_quant, A22_31_quant, A22_35_quant, A22_38_quant))
A22.TPM[,colony:="A22"]

Ar.TPM <- data.table(rbind(Ar_0_quant, Ar_3_quant, Ar_7_quant, Ar_10_quant, Ar_14_quant, Ar_17_quant, Ar_21_quant, Ar_24_quant, Ar_28_quant, Ar_31_quant, Ar_35_quant, Ar_38_quant))
Ar.TPM[,colony:="Ar"]

# combine colonies into single data.frame
TPM.dt <- rbind(A22.TPM, Ar.TPM)
TPM.dt$colony <- as.factor(TPM.dt$colony)
str(TPM.dt)

# drop temperature 7 
setkey(TPM.dt, val)
TPM.dt.sub <- TPM.dt[val != 7] 

length(unique(TPM.dt.sub$Transcript))

# set "trt" to true values - truncated in file names for convenience
TPM.dt.sub[val==3,val:=3.5]
TPM.dt.sub[val==10,val:=10.5]
TPM.dt.sub[val==17,val:=17.5]
TPM.dt.sub[val==24,val:=24.5]
TPM.dt.sub[val==31,val:=31.5]
TPM.dt.sub[val==38,val:=38.5]
str(TPM.dt.sub)
unique(TPM.dt.sub$val)
```

```{r exp_correlations, echo=FALSE, results='hide'}
cortable <- ddply(TPM.dt.sub, .(val), summarize, cor(TPM[which(colony=="A22")], TPM[which(colony=="Ar")]))
```

\newpage

Note that expression levels at each temperature treatment are highly correlated between the two colonies, indicating common patterns of expression response to temperature.

```{r exp_correlations_table, echo=FALSE, results='asis'}
colnames(cortable) <- c("Temperature", "r")
pandoc.table(cortable, style="rmarkdown", caption = "Correlations between species for gene expression at temperature treatment", round=2)
```


### Remove *contaminant* transcripts

With gene expression quantified using all reads, I removed the transcripts identified as contaminants by DeconSeq.

```{r remove_contaminants}
# extract transcript names from fasta file rather than loading whole file
system("grep '^>' results/trinity-full/Trinity_cap3_uclust_cont.fa | cut -f 1 -d ' ' | sed 's/^.{1}//' > results/trinity-full/cont.list")

# read file
cont.list <- read.table("results/trinity-full/cont.list")
# remote ">" leader
cont.list$V1 <- gsub(">", "", cont.list$V1)
str(cont.list)

# remove from TPM.dt.sub
setkey(TPM.dt.sub, Transcript)
TPM.dt.sub <- TPM.dt.sub[!cont.list]
str(TPM.dt.sub)
length(unique(TPM.dt.sub$Transcript))

# remove from annotation.table
setkey(annotation.table, Sequence.Name)
annotation.table <- annotation.table[!cont.list]
str(annotation.table)
``` 

### Regression-model to identify thermally-responsive genes

To identify transcripts (roughly equivalent to genes) that show thermal responsiveness, I fit the following linear model to each transcript:

$$ log(TPM + 1) = \beta_0 + \beta_1(species) + \beta_2(temp) + \beta_3(temp^2) + \beta_4(species * temp) + \beta_5(species * temp^2) + \epsilon $$

where TPM is transcripts per million. 

(1) Identify transcripts with overall significant model fit. Adjust *P* values for multiple testing using FDR and retain transcripts with FDR < 0.05. Use log-transformed response to account for outliers.


```{r modp}
# define model for RxN function
model <-  "log(TPM+1) ~ colony + val + I(val^2) + colony:val + colony:I(val^2)"

# calculate overall P value and R^2 for each transcript
RxNpval <- ddply(TPM.dt.sub, .(Transcript), .inform="TRUE", modpFunc)
```

Of the `r nrow(RxNpval)` transcripts, `r length(which(RxNpval$pval < 0.05))` have models with P < 0.05.

Many of these are likely false positives, so I adjusted P-values using false discovery rate (FDR).

```{r padjust}
RxNpval$padj <- p.adjust(RxNpval$pval, method = "fdr")
# Plot FDR values against initial pvalues
par(mfrow = c(2,1))
hist(RxNpval$pval)
hist(RxNpval$padj)

# subset to significant transcripts
pre.signif.transcripts <- RxNpval[which(RxNpval$padj < 0.05), ]
```

At the 5% FDR significance threshold, there are `r nrow(pre.signif.transcripts)` transcripts with an overall significant model.

Feedback from reviewers questioned whether this was sufficient to remove all false positives. To examine this, we performed a resampling simulation to determine how many transcripts would be identified as significant using the above approach. Specifically, we randomly re-assigned the expression levels to the temperature treatments for each species, fit the same linear model as for the real data, calculated the number of significant transcripts under the 5% FDR threshold, and repeated this 100 times. We then took the 95th percentile of the distribution of significant transcripts from the resampled datasets as the true null level of false positives.

As this resampling is slow, I ran it in a separate file (ApTranscriptome_resample.Rmd) and load the results here.

```{r resample}
# load resampling results
load("data/resample.Rda")
```

Using this approach, the empirical false positive, *after* adjusting *P*-values for FDR, is `r quantile (D$num_signif, probs = 0.95)`, indicating that only `r nrow(pre.signif.transcripts) - quantile(D$num_signif, probs = 0.95)` are true positives.

```{r}
# order by padj
pre.signif.transcripts <- pre.signif.transcripts[order(pre.signif.transcripts$padj), ]
# number to keep
keep <- nrow(pre.signif.transcripts) - quantile(D$num_signif, probs = 0.95)
signif.transcripts <- pre.signif.transcripts[1:keep, ]

# extract significant transcripts
sig.TPM.dt.sub <- TPM.dt.sub[signif.transcripts$Transcript]
```

(2) Fit linear model to significantly responsive transcripts, perform stepAIC to retain only significant terms, and save `lm` output to list

```{r RxNlmAIC, echo=TRUE, eval=TRUE, results='hide'}
# perform model selection for responsive transcripts
# need to use `try` to avoid stopping on error for AIC at Infinity
RxNlmAIC <- try(dlply(sig.TPM.dt.sub, .(Transcript), lmFunc))
```

The set of transcripts with significant expression patterns include those with expression that differs by species, temperature and the interaction of species and temperature. I am specifically interested in the thermally-responsive transcripts (temperature and species x temperature) so I subset the significant transcripts to examine these. 

```{r responsive_transcripts}
interaction.lms <- RxNlmAIC[which(Map(grepFunc, RxNlmAIC, term = "colonyAr:") == TRUE)]
other.lms <- RxNlmAIC[setdiff(names(RxNlmAIC), names(interaction.lms))]
temperature.lms <- other.lms[which(Map(grepFunc, other.lms, term = "val") == TRUE)]
colony.lms <- other.lms[setdiff(names(other.lms), names(temperature.lms))]

# combine all responsive transcripts
responsive.lms <- c(temperature.lms, interaction.lms)
rm(other.lms)

quadratic.lms <- RxNlmAIC[which(Map(grepFunc, RxNlmAIC, term = "2") == TRUE)]
```


```{r table_results, echo=FALSE, results='asis'}
clist <- c('Total', 'Colony', 'Temperature', 'Temperature:Colony')
siglist <- c(length(RxNlmAIC),
             length(colony.lms),
             length(temperature.lms),
             length(interaction.lms)
             )

sigtable <- data.frame(Coefficient = clist, "Number significant" = siglist)

pandoc.table(sigtable, style="rmarkdown", big.mark = ",", caption = "Number of transcripts with expression that depends on species, temperature or their interaction at 5% FDR  out of 98,186 total transcripts.")
```

### Thermal-response expression categories ###

The previous section identified the transcripts with thermally-responsive expression. In this section, I determined the shape of the expression response to temperature for each transcript. Categories of expression response are:

- **High** - increase expression with temperature
- **Low** - decrease expression with temperature
- **Intermediate** - maximum expression at intermediate temperatures (14 - 28C)
- **Bimodal** - expressed greater than one standard deviation of expression at both low and high temperatures
- **Not Responsive** - not thermally-responsive in one species but does respond in the other

For the transcripts where thermal-responsive expression depends on species, I determined the functional type of the expression response separately for each species. 

```{r expression_shape, eval=TRUE, echo=TRUE}
# calculate response type for all temperature responsive transcripts
sig.response.type <- ldply(responsive.lms, .progress = "none", .inform = TRUE, RxNtype)
colnames(sig.response.type)[which(colnames(sig.response.type) == ".id")] <- "Transcript"
sig.response.type <- data.table(sig.response.type)
setkey(sig.response.type, Transcript)
str(sig.response.type)

# calculate response type for transcripts with species x temperature interaction only
interaction.response.type <- ldply(interaction.lms, .progress = "none", .inform = TRUE, RxNtype)
colnames(interaction.response.type)[which(colnames(interaction.response.type) == ".id")] <- "Transcript"
interaction.response.type <- data.table(interaction.response.type)
setkey(interaction.response.type, Transcript)
str(interaction.response.type)

# calculate response types for transcripts without interactions
temperature.response.type <- ldply(temperature.lms, .progress = "none", .inform = TRUE, RxNtype)

# save results to file
write.table(file = paste(resultsdir, "Ap_responsive_transcripts_", Sys.Date(), ".txt", sep = ""), sig.response.type, row.names = FALSE, col.names = TRUE, quote = FALSE, sep = "\t")
```

Next, I compared the number of thermally-responsive in each response category between the two colonies. 

```{r temperature_response}
A22.type.table <- table(sig.response.type$A22.type)
Ar.type.table <- table(sig.response.type$Ar.type)

# table
(Ap.type.table <- rbind(A22.type.table, Ar.type.table))
rownames(Ap.type.table) <- list("A. picea", "A. carolinensis")

# proportion
Ap.type.table.prop <- round(Ap.type.table / (sum(Ap.type.table)/2) * 100)

# test if the number of transcripts in each category differs from null expectations using Pearson Chi-square test
(chi1 <- chisq.test(Ap.type.table))

# test if the total number of responsive transcripts differs between species 
resp.table <- matrix(nrow=2, ncol=2, dimnames = list(c("R", "NR"), c("Ap", "Ac")))
resp.table[1,1] <- sum(A22.type.table[which(names(A22.type.table) != "NotResp")])
resp.table[2,1] <- nrow(annotation.table) - sum(A22.type.table[which(names(A22.type.table) != "NotResp")])
resp.table[1,2] <- sum(Ar.type.table[which(names(Ar.type.table) != "NotResp")])
resp.table[2,2] <- nrow(annotation.table) - sum(Ar.type.table[which(names(Ar.type.table) != "NotResp")])
resp.table

(chi2 <- chisq.test(resp.table))

# test if the number of Low responsive transcripts differs between species 
low.table <- matrix(nrow=2, ncol=2, dimnames = list(c("Low", "NotLow"), c("Ap", "Ac")))
low.table[1,1] <- A22.type.table[which(names(A22.type.table) == "Low")]
low.table[2,1] <- sum(A22.type.table[which(names(A22.type.table) != "Low")])
low.table[1,2] <- Ar.type.table[which(names(Ar.type.table) == "Low")]
low.table[2,2] <- sum(Ar.type.table[which(names(Ar.type.table) != "Low")])
low.table

(chi3 <- chisq.test(low.table))

# test if the number of Intermediate responsive transcripts differs between species 
intermediate.table <- matrix(nrow=2, ncol=2, dimnames = list(c("Intermediate", "Others"), c("Ap", "Ac")))
intermediate.table[1,1] <- A22.type.table[which(names(A22.type.table) == "Intermediate")]
intermediate.table[2,1] <- sum(A22.type.table[which(names(A22.type.table) != "Intermediate")])
intermediate.table[1,2] <- Ar.type.table[which(names(Ar.type.table) == "Intermediate")]
intermediate.table[2,2] <- sum(Ar.type.table[which(names(Ar.type.table) != "Intermediate")])
intermediate.table

(chi4 <- chisq.test(intermediate.table))

# test if the number of High responsive transcripts differs between species 
high.table <- matrix(nrow=2, ncol=2, dimnames = list(c("High", "Others"), c("Ap", "Ac")))
high.table[1,1] <- A22.type.table[which(names(A22.type.table) == "High")]
high.table[2,1] <- sum(A22.type.table[which(names(A22.type.table) != "High")])
high.table[1,2] <- Ar.type.table[which(names(Ar.type.table) == "High")]
high.table[2,2] <- sum(Ar.type.table[which(names(Ar.type.table) != "High")])
high.table

(chi5 <- chisq.test(high.table))

# test if the number of Bimodal responsive transcripts differs between species 
bimodal.table <- matrix(nrow=2, ncol=2, dimnames = list(c("Bimodal", "Others"), c("Ap", "Ac")))
bimodal.table[1,1] <- A22.type.table[which(names(A22.type.table) == "Bimodal")]
bimodal.table[2,1] <- sum(A22.type.table[which(names(A22.type.table) != "Bimodal")])
bimodal.table[1,2] <- Ar.type.table[which(names(Ar.type.table) == "Bimodal")]
bimodal.table[2,2] <- sum(Ar.type.table[which(names(Ar.type.table) != "Bimodal")])
bimodal.table

(chi6 <- chisq.test(bimodal.table))
```

The total number of responsive transcripts does not differ between species, though their groupings into expression categories does differ. 

## Test of marginal frequencies between transcripts in each expression category between species

The question of biological interest is whether the marginal frequencies differ between the two species. Statistically, this can be addressed using the generalized [McNemar's test](http://en.wikipedia.org/wiki/McNemar's_test) of marginal homogeneity. 

```{r mh_test}
type.table <- table(Acar = sig.response.type$Ar.type, Apic = sig.response.type$A22.type)
# reorder
tt2 <- type.table[c("Low", "Intermediate", "High", "Bimodal", "NotResp"), c("Low", "Intermediate", "High", "Bimodal", "NotResp")]

# McNemar's test
mh.test <- mh_test(as.table(tt2))

# calculate contribution of each off-diagonal to Z~0~
d1 <- sum(tt2[1,2:5]) - sum(tt2[2:5,1])
d2 <- sum(tt2[2,c(1,3:5)]) - sum(tt2[c(1,3:5),2])
d3 <- sum(tt2[3,c(1:2,4:5)]) - sum(tt2[c(1:2,4:5),3])
d4 <- sum(tt2[4,c(1:3,5)]) - sum(tt2[c(1:3,5),4])
d5 <- sum(tt2[5,1:4]) - sum(tt2[1:4,5])

# proportion of test statistic due to off-diagonals
dsum <- sum(abs(d1), abs(d2), abs(d3), abs(d4), abs(d5))
abs(d1)/dsum
abs(d2)/dsum
abs(d3)/dsum
abs(d4)/dsum
abs(d5)/dsum

# e2 (Acar Intermediate) contributes ~48% to Z~0~
```

To identify specific cells that deviate from null expectation, I calculated the expected observations assuming that marginal frequencies equal the overall frequency for each expression category, then determined which cells have deviations between observed and expected that are greater than the overall X^2^ statistic.

```{r chi_posthoc}
# overall mean for each class
rs <- rowSums(tt2)
cs <- colSums(tt2)
(gm <- (rs + cs) / sum(tt2 * 2))

# calculate observed values for each cell using overall mean for each expression type
Ec <- outer(gm, gm, "*") * sum(tt2)

# get deviations of observed from expected
Ec.dev <- tt2 - Ec

# calculate chi-squared deviation 
Ec.cells <- sign(tt2 - Ec) * (tt2 - Ec)^2 / Ec

# which of these are significantly great than expected chi-squared?
```

To visualize this, I made a heatmap that showed the deviation of the actual number of transcripts in each cell from the expected count.

```{r mh_plot}
md <- reshape2::melt(Ec.dev)
  
mh_plot <- qplot(x=Apic, y=Acar, data=md, fill=value, geom="tile", ylim = rev(levels(md$Acar))) +
  scale_fill_gradient2(limits=c(-200, 200), low="#f1a340", high="#998ec3") +
  theme(axis.title = element_text(face="italic")) +
  labs(x = "A. picea", y = "A. carolinensis")

mh_plot
# make subplot of rows that match hypotheses

# A. carolinensis **Low** and **Bimodal**
md_sub1 <- subset(md, Acar %in% c("Bimodal", "Low"), )
md_sub1 <- droplevels(md_sub1)

mh_plot_sub1 <- qplot(y=Apic, x=Acar, data=md_sub1, fill=value, geom="tile", xlim = rev(levels(md_sub1$Acar))) + 
  scale_fill_gradient2(limits=c(-205, 205), low="#f1a340", high="#998ec3") +
  theme(axis.title = element_text(face="italic")) +
  labs(y = "A. picea", x = "A. carolinensis") +
  theme(legend.position = "right") + 
  geom_rect(xmin = 0.5, xmax = 1.5, ymin = 3.5, ymax = 4.5, fill = "transparent", linetype=2, colour="black") +
  geom_rect(xmin = 1.5, xmax = 2.5, ymin = 0.5, ymax = 1.5, fill = "transparent", linetype=2, colour="black")

mh_plot_sub1

# A. picea **High** and **Bimodal**
md_sub2 <- subset(md, Apic %in% c("High", "Bimodal"), )
md_sub2 <- droplevels(md_sub2)

mh_plot_sub2 <- qplot(y=Acar, x=Apic, data=md_sub2, fill=value, geom="tile", xlim = rev(levels(md_sub2$Apic))) + 
  scale_fill_gradient2(limits=c(-205, 205), low="#f1a340", high="#998ec3") +
  theme(axis.title = element_text(face="italic")) +
  labs(y = "A. carolinensis", x = "A. picea") + 
  theme(legend.position = "none") + 
  geom_rect(xmin = 0.5, xmax = 1.5, ymin = 3.5, ymax = 4.5, fill = "transparent", linetype=2, colour="black") +
  geom_rect(xmin = 1.5, xmax = 2.5, ymin = 2.5, ymax = 3.5, fill = "transparent", linetype=2, colour="black")

mh_plot_sub2

# arrange plots to make figure for manuscript
mh_plot_sub1 <- mh_plot_sub1 + ggtitle('A') + theme(plot.title = element_text(hjust=0))
mh_plot_sub2 <- mh_plot_sub2 + ggtitle('B') + theme(plot.title = element_text(hjust=0))

png("results/matched_observations_Fig2.png")
grid.arrange(mh_plot_sub1, mh_plot_sub2, ncol=2, widths = c(3.75, 3))
dev.off()

# full figure for supplemental
png("results/matched_observations_FigS1.png")
mh_plot
dev.off()
```


The *enhanced response hypothesis* posits that temperature adaptation uses existing response mechanisms, which should result in significant overlap in response. This pattern would be apparent as an excess of transcripts with the same response pattern between species (e.g. more shared "Low" transcripts) and a deficit of transcripts that shift to other response patterns or become not responsive. 

```{r er_cartoon}
enhanced_response_dat <- md_sub1
enhanced_response_dat$value <- 0
# set non-shared response to random negative value 
enhanced_response_dat[which(enhanced_response_dat$Acar == "Low"), "value"] <- runif(5, min=-20, max=-10)
enhanced_response_dat[which(enhanced_response_dat$Acar == "Bimodal"), "value"] <- runif(5, min=-20, max=-10)
# change shared response to positive
enhanced_response_dat[which(enhanced_response_dat$Acar == "Low" & enhanced_response_dat$Apic == "Low"), "value"] <- 30
enhanced_response_dat[which(enhanced_response_dat$Acar == "Bimodal" & enhanced_response_dat$Apic == "Bimodal"), "value"] <- 30

enhanced_response_dat

# plot
enhanced_response_plot <- ggplot(enhanced_response_dat, aes(y = Apic, x = Acar, fill=value)) + 
  geom_tile(colour = "#ffffff", size =1) +
  xlim(rev(levels(enhanced_response_dat$Acar))) + 
  scale_fill_gradient2(limits=c(-100, 100), low="#f1a340", high="#998ec3") +
  theme(axis.title = element_text(face="italic")) +
  labs(y = "A. picea", x = "A. carolinensis") +
  theme(legend.position = "right") + 
  geom_rect(xmin = 0.5, xmax = 1.5, ymin = 3.5, ymax = 4.5, fill = "transparent", linetype=2, colour="black") +
  geom_rect(xmin = 1.5, xmax = 2.5, ymin = 0.5, ymax = 1.5, fill = "transparent", linetype=2, colour="black")

# title
enhanced_response_plot <- enhanced_response_plot + ggtitle('Enhanced Response hypothesis') + theme(plot.title = element_text(hjust=0, size = 10))
enhanced_response_plot
```

In contrast, the *tolerance hypothesis* predicts that genes involved in active response will become non-responsive or shift to other response categories in the better-adapted species. 

```{r tol_cartoon}
tolerance_dat <- md_sub1
tolerance_dat$value <- 0
# set non-shared response to random positive value 
tolerance_dat[which(tolerance_dat$Acar == "Low"), "value"] <- runif(5, min=40, max=60)
tolerance_dat[which(tolerance_dat$Acar == "Bimodal"), "value"] <- runif(5, min=40, max=60)
# change shared response to near zero
tolerance_dat[which(tolerance_dat$Acar == "Low" & tolerance_dat$Apic == "Low"), "value"] <- -20
tolerance_dat[which(tolerance_dat$Acar == "Bimodal" & tolerance_dat$Apic == "Bimodal"), "value"] <- -20

tolerance_dat

# plot
tolerance_plot <- ggplot(tolerance_dat, aes(y=Apic, x=Acar, fill=value)) + 
  geom_tile(colour = "#ffffff", size =1) +
  xlim(rev(levels(enhanced_response_dat$Acar))) + 
  scale_fill_gradient2(limits=c(-100, 100), low="#f1a340", high="#998ec3") +
  theme(axis.title = element_text(face="italic")) +
  labs(y = "A. picea", x = "A. carolinensis") +
  theme(legend.position = "none") + 
  geom_rect(xmin = 0.5, xmax = 1.5, ymin = 3.5, ymax = 4.5, fill = "transparent", linetype=2, colour="black") +
  geom_rect(xmin = 1.5, xmax = 2.5, ymin = 0.5, ymax = 1.5, fill = "transparent", linetype=2, colour="black")

# title
tolerance_plot <- tolerance_plot + ggtitle('Tolerance hypothesis') + theme(plot.title = element_text(hjust=0, size = 10))
tolerance_plot

# plot together
png("results/predictions_Fig1.png")
grid.arrange(enhanced_response_plot, tolerance_plot, ncol=2, widths = c(3.7, 3))
dev.off()
```


The number of thermally-responsive in each response category differed between the colonies. For *A. picea*, the majority of the responsive transcripts were in the *Low* category, followed by *Bimodal*, *High*, *Intermediate* and *Not Responsive*. For *A. carolinensis*, the majority of responsive transcripts were in the *Bimodal* and *High* categories, followed by *Intermediate*, *Low* and *Not Responsive*.  

Interestingly, over half of the *High* genes in *AcNC* are *Low* in *ApVT*, and vice versa. In contrast, most of the *Low* genes in one species are also *Low* in the other species.


```{r temperature_response_table, echo=FALSE, results='asis'}
# add row and column totals
tt3 <- rbind(tt2, colSums(tt2))
tt3 <- cbind(tt3, rowSums(tt3))
colnames(tt3)[6] <- "Total"
rownames(tt3)[6] <- "Total"
# move names into table
tt3 <- rbind(rownames(tt3), tt3)
tt3 <- cbind(rownames(tt3), tt3)
# name col/row by species
rownames(tt3) <- c("AcNC", "", "", "", "", "", "")
colnames(tt3) <- c("**ApVT**", "", "", "", "", "", "")

pandoc.table(tt3, style="rmarkdown", caption = "Number of transcripts with maximum expression at high, low, intermediate, both high and low (bimodal) temperatures or are not thermally-responsivefor each species and their overlap.", split.tables = "Inf")
```

Table 4 shows the number of transcripts that fall into each expression type for each each species. The totals for each species *do not* include the `r length(temperature.lms)` transcripts that have consistent temperature responses between the two colonies. 

An interesting observation from the matched observations plot (Fig. 1) is that for **NotResp** genes in *A. carolinensis* the **Bimodal** category in *A. picea* is over-represented. Finding that the mean expression level of these genes in *A. carolinensis* is greater than the expression level at the rearing temperature (25°C) in *A. picea* would be consistent with genetic assimilation. That is, they have evolved to be constitutively activated in *A. carolinensis* and thus are no longer thermally-responsive as they are in *A. picea*.

```{r gen_assimilation_bim}
Ap.bim.Ac.nr <- sig.response.type[which(sig.response.type$A22.type == "Bimodal" & sig.response.type$Ar.type == "NotResp"), ]

# get mean expression level for Ar transcripts
Ap.bim.Ac.nr.TPM <- TPM.dt.sub[Ap.bim.Ac.nr$Transcript]
Ap.bim.Ac.nr.mean.exp.NR <- ddply(Ap.bim.Ac.nr.TPM, .(Transcript), summarize, meanTPM = mean(TPM))

# get expression at optimum for A22 transcripts
Ap.bim.Ac.nr.opt.exp <- Ap.bim.Ac.nr$A22.opt

t.test(Ap.bim.Ac.nr.mean.exp.NR$meanTPM, Ap.bim.Ac.nr.opt.exp)
```

This hypothesis was not supported as expression was higher for the **Bimodal** transcripts in *A. picea* than the non-responsive *A. carolinensis* transcripts.

## Species comparisons

In this section, I performed a number of comparisons of the thermal reactionomes between the *Ar* and *A22* species. Specifically, I:

- compared profiles of the temperature of maximum expression for thermally-responsive transcripts
- compared basal expression at optimal temperature for thermally-responsive genes
- compared degree of upregulation to constitutive expression
- compared the critical temperature of transcript upregulation, T~on~, between species for transcripts induced in response to high or low temperatures 
- compared the critical temperature of transcript downregulation, T~off~, between species for transcripts induced in response to high or low temperatures 

For these tests, I predicted expression for responsive transcripts.

```{r responsive_pred}
# subset to transcripts with significant temperature x species interaction
resp.TPM.dt.sub <- TPM.dt.sub[names(interaction.lms)]

# apply predFunc to all responsive transcripts
resp.TPM.dt.sub.pred <- ddply(resp.TPM.dt.sub, .(Transcript), .inform="TRUE", predFunc)

# setkey to Transcript and colony 
resp.TPM.dt.sub.pred <- data.table(resp.TPM.dt.sub.pred)
setkey(resp.TPM.dt.sub.pred, Transcript, colony)
```

I checked the proportion of responsive transcripts against those that have a BLAST hit.

```{r responsive_with_BLAST}
annotation.table.responsive <- annotation.table[which(annotation.table$best.hit.to.nr != "-"), ]

responsive_transcripts_w_BLAST <- length(which(unique(resp.TPM.dt.sub$Transcript) %in% annotation.table.responsive$Sequence.Name))

responsive_transcripts_w_BLAST / round(nrow(annotation.table.responsive))
```


### Compare temperature of maximum expression between species

Probability density function of peak expression for transcripts that differ in expression between *A22* and *Ar*. For this analysis, I used only the transcripts with a significant temperature by species interaction.

```{r max_exp_PDF, warning = FALSE, message = FALSE}
# reshape data
Ap.df <- data.frame(Transcript = rep(interaction.response.type$Transcript, times = 2), 
                    colony = rep(c("ApVT", "AcNC"), 
                                 each = length(interaction.response.type$Transcript)), 
                    max.val = c(interaction.response.type$A22.max, interaction.response.type$Ar.max))

maxexplot <- ggplot(Ap.df, aes(x=max.val, fill=colony)) + 
  geom_density(alpha=0.2, position="identity") + 
  labs(x = "Temperature of maximum expression", y = "Density") +
  theme(axis.title = element_text(size = rel(2))) +
  theme(axis.text = element_text(size = rel(1.2)))
print(maxexplot)

# Figure for presentation
png("results/temp_max_expression.png")
maxexplot + theme(legend.position = "none")
dev.off()
```


### Compare basal expression at optimal temperature for thermally-responsive genes

Genes upregulated in response to thermal stress in one species may have greater basal levels of expression in the other species that experiences those stressful conditions more often. To test this hypothesis, we compared expression levels at the rearing temperature (25C) between the two species for genes in that are either in the 'High' or 'Low' expression group in the other species. Specifically, do genes upregulated at high temperatures in *A22* have greater basal expression at optimal temperatures in *Ar*? 

```{r optimum_expression_comparison}
# list of transcripts that are 'high' expressed in A22
A22.high.transcripts.df <- interaction.response.type[which(interaction.response.type$A22.type == "High"), ]

# Compare expression at optimum temp (A22.opt) between colonies using t-test
t.test(A22.high.transcripts.df$A22.opt, A22.high.transcripts.df$Ar.opt)
boxplot(A22.high.transcripts.df$A22.opt, A22.high.transcripts.df$Ar.opt)

# T test on log-transformed values to control for outliers
t.test(log(A22.high.transcripts.df$A22.opt+1), log(A22.high.transcripts.df$Ar.opt+1))
boxplot(log(A22.high.transcripts.df$A22.opt+1), log(A22.high.transcripts.df$Ar.opt+1))
```

The `t.test` fails to account for the many orders of magnitude difference in expression among transcripts, e.g. non-equal variances. This problem is the key issue in the analysis of differential expression `r citep(c("10.1186/1471-2105-11-94", "10.1038/nprot.2013.099"))`. As my goal is simply to determine if expression is typically greater at optimal temperatures (19.5 C) in *Ar* than *A22* for genes that are up-regulated at high temperatures in *A22*, I use a non-parametric Wilcoxon signed rank-test.

```{r A22_high_wilcoxon}
w1 <- wilcox.test(A22.high.transcripts.df$A22.opt, A22.high.transcripts.df$Ar.opt, alternative = "two.sided", paired = TRUE, conf.int = TRUE)
w1
```

Consistent with expectations, there is greater expression at 25C for *Ar* than *A22* transcripts for the set of transcripts that are transcripts that are up-regulated at high temperatures in *A22*. Note that *A22* had the larger library size so if this was due to TPM not correctly accounting for differences in reads between samples, we would expect to see a positive instead of negative value here.

Next I test the converse: do genes up-regulated at low temperatures in *Ar* have greater basal expression near optimal temperatures in *A22*?

```{r Ar_low_wilcoxon}
# list of transcripts that are 'high' expressed in Ar
Ar.low.transcripts.df <- interaction.response.type[which(interaction.response.type$Ar.type == "Low"), ]

# t-test with log values
t.test(log(Ar.low.transcripts.df$A22.opt+1), log(Ar.low.transcripts.df$Ar.opt+1))
boxplot(log(Ar.low.transcripts.df$A22.opt+1), log(Ar.low.transcripts.df$Ar.opt+1))

# Wilcoxon signed rank-test
w2 <- wilcox.test(Ar.low.transcripts.df$A22.opt, Ar.low.transcripts.df$Ar.opt, alternative = "two.sided", paired = TRUE, conf.int = TRUE)
w2
```

Counter to expectations, expression at optimal temperatures is also greater in *Ar* than *A22* for transcripts upregulated at low temperatures in *Ar*. 

To confirm that there are not sample-level issues, I performed the same comparison using *Intermediate* expressed-genes where I do not expect to see a difference in expression.

```{r intermediate_transcripts}
# list of transcripts that are 'Intermediate' expressed in either colony
Ap.int.transcripts <- interaction.response.type[which(interaction.response.type$Ar.type == "Intermediate" | interaction.response.type$A22.type == "Intermediate"), ]
# T test
t.test(log(Ap.int.transcripts$A22.opt+1), log(Ap.int.transcripts$Ar.opt+1))
# Wilcoxon signed rank-test
w3 <- wilcox.test(Ap.int.transcripts$A22.opt, Ap.int.transcripts$Ar.opt, alternative = "two.sided", paired = TRUE, conf.int = TRUE)
w3
```

The non-parametric test for both comparisions also finds greater expression in *Ar* than *A22* at the optimal temperature.

To visualize this data, I plot the log ratio of inducibility between species against the log ratio of constitutive expression between species. I do this separately for **High** transcripts in *A. picea* and then **Low** transcripts in *A. carolinensis*. 

First, I calculate the degree of expression induction as the proportion difference between the maximum and minimum expression levels for each transcript.

```{r induction_vs_constitutive}
# calculate degree of induction
Ap.induction <- ddply(resp.TPM.dt.sub.pred, .(colony, Transcript), summarise, 
                      upreg = (max(pTPM) - min(pTPM))/min(pTPM) * 100 )
# cast to wide
Ap.induction.wide <- dcast(Ap.induction, Transcript ~ colony, value.var = "upreg")
names(Ap.induction.wide)[names(Ap.induction.wide)=="A22"] <- "A22.upreg"
names(Ap.induction.wide)[names(Ap.induction.wide)=="Ar"] <- "Ar.upreg"

# merge optimum expression, in interaction.response.type data.frame, with predicted TPM data
Ap.induction.wide <- data.table(Ap.induction.wide)
setkey(Ap.induction.wide, Transcript)
interaction.response.type2 <- Ap.induction.wide[interaction.response.type]
```

I extracted the **High** transcripts for *A. picea* and examined the log ratio of inducibility against the log ratio of constitutive expression in *A. picea* over *A. carolinensis**. According to the *genetic assimilation hypothesis* there should be a negative relationship between these ratios. That is, transcripts with greater constitutive expression in *A. carolinensis* should have greater inducibility in response to **High** temperatures in *A. picea*.

```{r log_inducibility}
interaction.response.type2.A22.high <- interaction.response.type2[interaction.response.type2$Transcript %in% A22.high.transcripts.df$Transcript, ]
str(interaction.response.type2.A22.high)

# filter out NotResp transcripts in Ar
interaction.response.type2.A22.high <- interaction.response.type2.A22.high[which(interaction.response.type2.A22.high$Ar.type != "NotResp"), ]

# ratio of upregulation in A22 vs Ar
interaction.response.type2.A22.high$A22.Ar.upreg <- interaction.response.type2.A22.high$A22.upreg / interaction.response.type2.A22.high$Ar.upreg

# ratio of constitutive expression in Ar vs A22
interaction.response.type2.A22.high$A22.Ar.constitutive <- interaction.response.type2.A22.high$A22.opt / interaction.response.type2.A22.high$Ar.opt

# plot!
gg11 <- ggplot(interaction.response.type2.A22.high, aes(x = log(A22.Ar.upreg), y = log(A22.Ar.constitutive))) + 
  geom_point() + 
  geom_smooth(method = "lm") +
  labs(x = expression(paste("log(inducibility ", italic("A. picea / A. carolinensis"), ")")),
       y = expression(paste("log(constitutive expression ", italic("A. picea / A. carolinensis"), ")")))    

gg11

A22.high.lm <- lm(log(A22.Ar.constitutive) ~ log(A22.Ar.upreg), data = interaction.response.type2.A22.high)

summary(A22.high.lm)
```

In contrast to the prediction there is a positive relationship between these ratios. That is, genes with greater upregulation of expression also have greater constitutive expression. 

I repeated the above but for **Low** transcripts in *A. carolinensis*. 

```{r assimilation_Ar_low}
### filter for A. carolinensis *Low* transcripts
interaction.response.type2.Ar.low <- interaction.response.type2[interaction.response.type2$Transcript %in% Ar.low.transcripts.df$Transcript, ]
str(interaction.response.type2.Ar.low)

# filter out NotResp transcripts in A22
interaction.response.type2.Ar.low <- interaction.response.type2.Ar.low[which(interaction.response.type2.Ar.low$A22.type != "NotResp"), ]

# ratio of upregulation
interaction.response.type2.Ar.low$Ar.A22.upreg <- interaction.response.type2.Ar.low$Ar.upreg / interaction.response.type2.Ar.low$A22.upreg

# ratio of constitutive expression
interaction.response.type2.Ar.low$Ar.A22.constitutive <- interaction.response.type2.Ar.low$Ar.opt / interaction.response.type2.Ar.low$A22.opt

gg12 <- ggplot(interaction.response.type2.Ar.low, aes(x = log(Ar.A22.upreg), y = log(Ar.A22.constitutive))) + 
  geom_point() + 
  geom_smooth(method = "lm") +
  labs(x = expression(paste("log(inducibility ", italic("A. carolinensis / A. picea"), ")")),
       y = expression(paste("log(constitutive expression ", italic("A. carolinensis / A. picea"), ")")))                   

gg12

Ar.low.lm <- lm(log(Ar.A22.constitutive) ~ log(Ar.A22.upreg), data = interaction.response.type2.Ar.low)

summary(Ar.low.lm)
```

I combine the two above figures into a single figure.

```{r fig3}
# add labels
gg11 <- gg11 + ggtitle('A') + theme(plot.title = element_text(hjust=0))
gg12 <- gg12 + ggtitle('B') + theme(plot.title = element_text(hjust=0))
# grid.arrange
png(file = "results/genetic_assimilation_Fig4.png")
grid.arrange(gg11, gg12, nrow = 1)
dev.off()
```

As with the previous comparison, I also found no evidence for the genetic assimilation hypothesis. 

### Compare the critical temperature of transcript upregulation, T~on~, between species for transcripts induced in response to high or low temperatures

Thermally-responsive genes could also differ in the critical temperature of gene induction. To examine this, I determined the temperature at which each responsive gene had the greatest increase or decrease in expression.

```{r responsive_TPM}
# extract TPM data for thermally-responsive transcripts
resp.TPM.dt.sub <- TPM.dt.sub[names(responsive.lms)]
setkey(resp.TPM.dt.sub, Transcript)
str(resp.TPM.dt.sub)
length(unique(resp.TPM.dt.sub$Transcript))
# scale transcripts so can compare
resp.TPM.dt.sub[,TPM.scaled:=scale(TPM), by = Transcript]
# rename colonies
resp.TPM.dt.sub$colony2 <- ifelse(resp.TPM.dt.sub$colony == "A22", "ApVT", "AcNC")
```

For next analyses, I extracted the list of gene names by response type.

```{r extract_Transcripts}
interaction.response.type <- as.data.frame(interaction.response.type)
A22.high.transcripts <-  interaction.response.type[which(interaction.response.type$A22.type == "High"), "Transcript"]
Ar.high.transcripts <- interaction.response.type[which(interaction.response.type$Ar.type == "High"), "Transcript"]

A22.low.transcripts <- interaction.response.type[which(interaction.response.type$A22.type == "Low"), "Transcript"]
Ar.low.transcripts <- interaction.response.type[which(interaction.response.type$Ar.type == "Low"), "Transcript"]

A22.bim.transcripts <- interaction.response.type[which(interaction.response.type$A22.type == "Bimodal"), "Transcript"]
Ar.bim.transcripts <- interaction.response.type[which(interaction.response.type$Ar.type == "Bimodal"), "Transcript"]

A22.int.transcripts <- interaction.response.type[which(interaction.response.type$A22.type == "Intermediate"), "Transcript"]
Ar.int.transcripts <- interaction.response.type[which(interaction.response.type$Ar.type == "Intermediate"), "Transcript"]
```

I calculated T~on~ for *High* genes in each species using the observed (rather than predicted) values for each species as by nature of fitting a quadratic function, the maximum change tends to be at the extremes (38.5 or 0) for the predicted values.

```{r T_on_high}
# transcripts expressed at *High* and *Bimodal* temperatures in A22
A22.high.TPM.dt.sub <- resp.TPM.dt.sub.pred[J(union(A22.high.transcripts.df$Transcript, A22.bim.transcripts), "A22")]
setkey(A22.high.TPM.dt.sub, Transcript)
str(A22.high.TPM.dt.sub)

# make data.frame for results
l1 <- length(unique(A22.high.TPM.dt.sub$Transcript))
A22.high.T_on <- data.frame(Transcript = unique(A22.high.TPM.dt.sub$Transcript), colony = rep("ApVT", length = l1), type = rep("High", length = l1), T_on = NA)

# loop across transcripts, calculating T_on

for(i in unique(A22.high.TPM.dt.sub$Transcript)) {
    subdf <- A22.high.TPM.dt.sub[i]
    subdf <- subdf[which(subdf$val > 14), ]
    T_on <- subdf[median(which(diff(subdf$TPM) == max(diff(subdf$TPM))))+1, val]
    A22.high.T_on[which(A22.high.T_on$Transcript == i), "T_on"] <- T_on
}   

# repeat for Ar
Ar.high.TPM.dt.sub <- resp.TPM.dt.sub.pred[J(union(Ar.high.transcripts, Ar.bim.transcripts), "Ar")]
setkey(Ar.high.TPM.dt.sub, Transcript)
str(Ar.high.TPM.dt.sub)

l2 <- length(unique(Ar.high.TPM.dt.sub$Transcript))
Ar.high.T_on <- data.frame(Transcript = unique(Ar.high.TPM.dt.sub$Transcript), colony = rep("AcNC", length = l2), type = rep("High", length = l2), T_on = NA)

for(i in unique(Ar.high.TPM.dt.sub$Transcript)) {
    subdf <- Ar.high.TPM.dt.sub[i]
    subdf <- subdf[which(subdf$val > 14), ]
    T_on <- subdf[median(which(diff(subdf$TPM) == max(diff(subdf$TPM))))+1, val]
    Ar.high.T_on[which(Ar.high.T_on$Transcript == i), "T_on"] <- T_on
}   

# determine if T~on~ is greater in *A22* or *Ar* for *High* genes.

(T_on.high.ttest <- t.test(Ar.high.T_on$T_on, A22.high.T_on$T_on))
```

This test revealed no differences among species in T~on~ for *High* genes.

I performed the same test comparing T~on~ for *Low* genes.

```{r T_on_low}
# transcripts expressed at *Low* temperatures in A22
A22.low.TPM.dt.sub <- resp.TPM.dt.sub.pred[J(union(A22.low.transcripts, A22.bim.transcripts), "A22")]
setkey(A22.low.TPM.dt.sub, Transcript)
str(A22.low.TPM.dt.sub)

# make data.frame for results
l3 <- length(unique(A22.low.TPM.dt.sub$Transcript))
A22.low.T_on <- data.frame(Transcript = unique(A22.low.TPM.dt.sub$Transcript), colony = rep("ApVT", length = l3), type = rep("Low", length = l3), T_on = NA)

# loop across transcripts, calculating T_on
for(i in unique(A22.low.TPM.dt.sub$Transcript)) {
    subdf <- A22.low.TPM.dt.sub[i]
    subdf <- subdf[which(subdf$val < 21), ]
    T_on <- subdf[median(which(diff(subdf$TPM) == max(diff(subdf$TPM))))+1, val]
    A22.low.T_on[which(A22.low.T_on$Transcript == i), "T_on"] <- T_on
}   

# repeat for Ar
Ar.low.TPM.dt.sub <- resp.TPM.dt.sub.pred[J(union(Ar.low.transcripts, Ar.bim.transcripts), "A22")]
setkey(Ar.low.TPM.dt.sub, Transcript)
str(Ar.low.TPM.dt.sub)

l4 <- length(unique(Ar.low.TPM.dt.sub$Transcript))
Ar.low.T_on <- data.frame(Transcript = unique(Ar.low.TPM.dt.sub$Transcript), colony = rep("AcNC", length = l4), type = rep("Low", length = length(unique(Ar.low.TPM.dt.sub$Transcript))), T_on = NA)

for(i in unique(Ar.low.TPM.dt.sub$Transcript)) {
    subdf <- Ar.low.TPM.dt.sub[i]
    subdf <- subdf[which(subdf$val < 21), ]
    T_on <- subdf[median(which(diff(subdf$TPM) == max(diff(subdf$TPM))))+1, val]
    Ar.low.T_on[which(Ar.low.T_on$Transcript == i), "T_on"] <- T_on
}   

(T_on.low.ttest <- t.test(Ar.low.T_on$T_on, A22.low.T_on$T_on))
```

Genes with increased expression at *Low* temperatures are on average turned on 0.2°C higher in *ApVT* than *AcNC*.

Visualize T~on~ for both *Low* and *High* genes on the same plot.

```{r plot_T_on, echo=FALSE, eval=TRUE}
# merge data for plotting
Ap.T_on <- rbind(A22.high.T_on, Ar.high.T_on, A22.low.T_on, Ar.low.T_on)
Ap.T_on$type <- factor(Ap.T_on$type, levels = c("Low", "High"))
Ap.T_on$Species <- ifelse(Ap.T_on$colony == "ApVT", "A. picea", "A. carolinensis")

# combined plot
T_on_plot <- ggplot(Ap.T_on, aes(x = T_on, y = ..density.., colour = Species, group = Species, fill = Species)) +
    facet_grid(. ~ type) +
    geom_histogram(position = "dodge", binwidth = 3) +
    geom_density(adjust = 3, fill = NA, size = 2) + 
    labs(x = "Temperature of gene activation", y = "Density") +
    theme(axis.title = element_text(size = rel(2))) +
    theme(axis.text = element_text(size = rel(1.2)))
#    theme(legend.position = "none")
print(T_on_plot)
```


### Compare the critical temperature of transcript downregulation, T~off~, between species for transcripts induced in response to high or low temperatures 

This analysis is comparable to the one above, but for the *Intermediate* genes that are downregulated, split into the low and high temperature extremes.

```{r T_int_off}
# transcripts expressed at *Intermediate* temperatures in A22
A22.int.TPM.dt.sub <- resp.TPM.dt.sub.pred[J(A22.int.transcripts, "A22")]
setkey(A22.int.TPM.dt.sub, Transcript)
str(A22.int.TPM.dt.sub)

# make data.frame for results
l5 <- length(unique(A22.int.TPM.dt.sub$Transcript))
A22.int.T_off <- data.frame(Transcript = unique(A22.int.TPM.dt.sub$Transcript), colony = rep("ApVT", length = l5), type = rep("Intermediate", length = l5), High = NA, Low = NA)

# loop across transcripts, calculating T_off at both high and low temperatures
for(i in unique(A22.int.TPM.dt.sub$Transcript)) {
    subdf <- A22.int.TPM.dt.sub[i]
    
    subdf.high <- subdf[which(subdf$val >= 21), ]
    High <- subdf.high[median(which(diff(subdf.high$TPM) == min(diff(subdf.high$TPM)))), val]
    A22.int.T_off[which(A22.int.T_off$Transcript == i), "High"] <- High
    
    subdf.low <- subdf[which(subdf$val <= 21), ]
    Low <- subdf.low[median(which(diff(subdf.low$TPM) == min(diff(subdf.low$TPM)))), val]
    A22.int.T_off[which(A22.int.T_off$Transcript == i), "Low"] <- Low
}   

# repeat for Ar
Ar.int.TPM.dt.sub <- resp.TPM.dt.sub.pred[J(Ar.int.transcripts, "Ar")]
setkey(Ar.int.TPM.dt.sub, Transcript)
str(Ar.int.TPM.dt.sub)

# make data.frame for results
l6 <- length(unique(Ar.int.TPM.dt.sub$Transcript))
Ar.int.T_off <- data.frame(Transcript = unique(Ar.int.TPM.dt.sub$Transcript), colony = rep("AcNC", length = l6), type = rep("Intermediate", length = l6), High = NA, Low = NA)

# loop across transcripts, calculating T_off at both high and low temperatures
for(i in unique(Ar.int.TPM.dt.sub$Transcript)) {
    subdf <- Ar.int.TPM.dt.sub[i]
    
    subdf.high <- subdf[which(subdf$val >= 21), ]
    High <- subdf.high[median(which(diff(subdf.high$TPM) == min(diff(subdf.high$TPM)))), val]
    Ar.int.T_off[which(Ar.int.T_off$Transcript == i), "High"] <- High
    
    subdf.low <- subdf[which(subdf$val <= 21), ]
    Low <- subdf.low[median(which(diff(subdf.low$TPM) == min(diff(subdf.low$TPM)))), val]
    Ar.int.T_off[which(Ar.int.T_off$Transcript == i), "Low"] <- Low
}

# compare T_off among species
(T_off.int.high.ttest <- t.test(Ar.int.T_off$High, A22.int.T_off$High))

(T_off.int.low.ttest <- t.test(Ar.int.T_off$Low, A22.int.T_off$Low))
```

As with genes turned on at high temperatures, *A. carolinensis* **Intermediate** genes are down-regulated on at higher temperatures than *A. picea* **Intermediate** genes. Towards low temperatures, the mean temperature of downregulation was not different between species.

Visualize T~off~ for both *Low* and *High* genes on the same plot.

```{r activation_plot, echo=FALSE, eval=TRUE}
# reshape data
Ar.T_off.df <- melt(Ar.int.T_off, id.vars = c("Transcript", "colony", "type"), 
                 measure.vars = c("High", "Low"),
                 variable.name = "class",
                 value.name = "T_off")

A22.T_off.df <- melt(A22.int.T_off, id.vars = c("Transcript", "colony", "type"), 
                 measure.vars = c("High", "Low"),
                 variable.name = "class",
                 value.name = "T_off")


# merge data for plotting
Ap.T_off <- rbind(Ar.T_off.df, A22.T_off.df)
Ap.T_off$Species <- ifelse(Ap.T_off$colony == "ApVT", "A. picea", "A. carolinensis")
# change factor levels to "Low" first
Ap.T_off$class = factor(Ap.T_off$class, levels = c("Low", "High"))


# combined plot
T_off_plot <- ggplot(Ap.T_off, aes(x = T_off, y = ..density.., colour = Species, group = Species, fill = Species)) +
    facet_grid(. ~ class) +
    geom_histogram(position = "dodge", binwidth = 3) +
    geom_density(adjust = 3, fill = NA, size = 2) + 
    labs(x = "Temperature of gene activation", y = "Density") +
    theme(axis.title = element_text(size = rel(2))) +
    theme(axis.text = element_text(size = rel(1.2)))
#    theme(legend.position = "none")
print(T_off_plot)
```

I combined these gene activation comparisons into a single plot.

```{r combined_plot}
gg2 <- ggplot(Ap.T_on, aes(x = T_on, y = ..density.., colour = Species, group = Species, fill = Species)) +
    facet_grid(. ~ type) +
    geom_histogram(position = "dodge", binwidth = 3) +
    geom_density(adjust = 3, fill = NA, size = 2) + 
    labs(x = "Critical temperature of upregulation", y = "Density") +
    theme(legend.position = "none")  

gg3 <- ggplot(Ap.T_off, aes(x = T_off, y = ..density.., colour = Species, group = Species, fill = Species)) +
    facet_grid(. ~ class) +
    geom_histogram(position = "dodge", binwidth = 3) +
    geom_density(adjust = 3, fill = NA, size = 2) + 
    labs(x = "Critical temperature of downregulation", y = "Density") +
    theme(legend.position = "bottom", legend.text = element_text(size = 12, face = 3))  

# grid.arrange
gg2 <- gg2 + ggtitle('A') + theme(plot.title = element_text(hjust=0))
gg3 <- gg3 + ggtitle('B') + theme(plot.title = element_text(hjust=0))

png("results/crit_temp_regulation_Fig3.png")
grid.arrange(gg2, gg3, nrow = 2, heights = c(.5,.6))
dev.off()
```

Next, I tested if the mean expression level for all responsive genes differed among colonies.

```{r mean_exp}
t.test(log(interaction.response.type$A22.opt), log(interaction.response.type$Ar.opt))
# expression level at optimum temp greater in Ar than A22

# non-parametric test
w4 = wilcox.test(interaction.response.type$A22.opt, 
                 interaction.response.type$Ar.opt,
                 alternative = "two.sided", paired = TRUE, conf.int = TRUE)


mean.exp <- ddply(resp.TPM.dt.sub.pred, .(colony, Transcript), summarise,
                  mean.TPM = mean(pTPM))

t.test(log(mean.exp[which(mean.exp$colony == "A22"), "mean.TPM"]), 
       log(mean.exp[which(mean.exp$colony == "Ar"), "mean.TPM"]))

# non-parametric test
w5 = wilcox.test(mean.exp[which(mean.exp$colony == "A22"), "mean.TPM"],
                 mean.exp[which(mean.exp$colony == "Ar"), "mean.TPM"],
                 alternative = "two.sided", paired = TRUE, conf.int = TRUE)


plot(log(mean.exp[which(mean.exp$colony == "A22"), "mean.TPM"]), 
     log(mean.exp[which(mean.exp$colony == "Ar"), "mean.TPM"]),
     xlab = "log(expression) A. picea",
     ylab = "log(expression) A. carolinensis")
abline(a=0, b=1)
```

On average, *A. carolinensis* has higher expression than *A. picea*.

## Gene set enrichment analysis for thermal-responsive genes

### Functional annotation

In the previous section, I identified transcripts that show significant responses in expression. Next, I add gene annotation and ontology information to these transcripts.  

```{r annotation}
setkey(annotation.table, Sequence.Name)
signif.transcripts <- data.table(signif.transcripts)
setkey(signif.transcripts, Transcript)
signif.transcripts <- annotation.table[signif.transcripts]
setnames(signif.transcripts, "Sequence.Name", "Transcript")
```

```{r ann_responsive, echo=TRUE}
responsive.lms.ann.type <- signif.transcripts[names(responsive.lms)]
# combine responsive.lms.ann with "type" information
cols <- c("Transcript", "A22.type", "Ar.type")
subtab <- sig.response.type[, cols, with = FALSE]
setkey(subtab, "Transcript")
responsive.lms.ann.type <- base::merge(responsive.lms.ann.type, subtab, by = "Transcript", all = TRUE)
str(responsive.lms.ann.type)
write.csv(responsive.lms.ann.type, file = paste(resultsdir, "Ap_responsive_genes_", Sys.Date(), ".csv", sep = ""), row.names = FALSE)
```


### Proportion of responsive transcripts annotated

Of the responsive transcripts, `r round(length(which(responsive.lms.ann.type$best.hit.to.nr != "-"))/length(responsive.lms.ann.type$best.hit.to.nr)*100, 0)`% are annotated, while `r round(length(which(annotation.table$best.hit.to.nr != "-"))/nrow(annotation.table) * 100, 0)`% of all transcripts are annotated.

### Candidate gene enrichment

First, I explore of candidate genes with the GO term "response to stress" are enriched in the responsive data set. 

```{r GO_contingency_test}
# GO 'response to stress' hits in responsive transcripts
GO0006950.responsive <- responsive.lms.ann.type[grep("GO:0006950", responsive.lms.ann.type$GO.Biological.Process), list(Transcript, best.hit.to.nr, A22.type, Ar.type)]

# in high category
GO0006950.responsive[union(with(GO0006950.responsive, grep("High", Ar.type)), with(GO0006950.responsive, grep("High", A22.type))), ]
# in low category
GO0006950.responsive[union(with(GO0006950.responsive, grep("Low", Ar.type)), with(GO0006950.responsive, grep("Low", A22.type))), ]
# in bimodal category
GO0006950.responsive[union(with(GO0006950.responsive, grep("Bimodal", Ar.type)), with(GO0006950.responsive, grep("Bimodal", A22.type))), ]
# in intermediate category
GO0006950.responsive[union(with(GO0006950.responsive, grep("Intermediate", Ar.type)), with(GO0006950.responsive, grep("Intermediate", A22.type))), ]

# Chi-square test to see if 'response to stress' related genes overrepresented in responsive.lms compared to full list
resp.stress.responsive.count <- nrow(responsive.lms.ann.type[grep("GO:0006950", responsive.lms.ann.type$GO.Biological.Process), list(Transcript, best.hit.to.nr)])
# GO 'response to stress' hits in all transcripts
resp.stress.all.count <- nrow(annotation.table[grep("GO:0006950", annotation.table$GO.Biological.Process), list(Sequence.Name, best.hit.to.nr)])

GO.stress.table <- matrix(rbind(resp.stress.responsive.count, nrow(responsive.lms.ann.type) - resp.stress.responsive.count, resp.stress.all.count, nrow(annotation.table) - resp.stress.all.count), nrow = 2)

GO.stress.Xsq <- chisq.test(GO.stress.table)
GO.stress.Xsq
```

While `r nrow(GO0006950.responsive)` "response to stress" genes are in the responsive set, this is not enriched compared to the whole transcriptome. 

```{r grep_shock}
hsp_responsive <- responsive.lms.ann.type[grep("shock", responsive.lms.ann.type$best.hit.to.nr), list(Transcript, best.hit.to.nr, A22.type, Ar.type)]
hsp_responsive

hsp_all <- annotation.table[grep("shock", annotation.table$best.hit.to.nr), list(Sequence.Name, best.hit.to.nr)]
nrow(hsp_all)
```

There are no heat shock protein-related genes in the responsive transcripts.

### Gene set enrichment analysis

I used [topGO](http://www.bioconductor.org/packages/2.12/bioc/html/topGO.html) to perform gene set enrichment analysis (GSEA) seperately for each expression type (bimodal, intermediate, high, low). With the GSEA results, I applied the `selectFDR` function to select transcripts with adjusted P < 0.05. For many of the responsive categories, it appeared that there is considerable redundancy in the enriched GO terms. I used the [GOSemSim](http://www.bioconductor.org/packages/release/bioc/html/GOSemSim.html) package to determine semantic similarity of enriched GO terms `r citep("10.1093/bioinformatics/btq064")` and then performed hierarhichal clustering based on this information distance.

For the GSEA, I first created a gene ID to GO term map file.

```{r geneid2go, echo=TRUE, eval=TRUE, cache=TRUE}
# create geneid2go.map file from FastAnnotator annotationtable.txt
mapname = "results/geneid2go.map"
if(file.exists(mapname)) print("Gene ID to GO term file exists") else{
    geneid2GOmap(as.data.frame(annotation.table), ontology = c("BP", "CC", "MF"), mapname = mapname)
}
```

Then I read the map file into memory.

```{r readmap, echo=TRUE, eval=TRUE}
# read mappings file
geneID2GO <- readMappings(file = mapname)
str(head(geneID2GO))
```

### GSEA for thermally-responsive transcripts

Using this gene2GO map file, I performed GSEA for:

**1) all responsive transcripts**

```{r gsea_all, results='hide'}
# create geneList. note that NA values cause problems with topGO
# so set any NA to 1 as need to retain for GO analysis
Ap.geneList <- RxNpval$padj
Ap.geneList[which(is.na(Ap.geneList))] <- 1
stopifnot(length(which(is.na(Ap.geneList))) == 0)
names(Ap.geneList) <- RxNpval$Transcript
str(Ap.geneList)

# Function to select genes with adjusted P-value < 0.05
selectFDR <- function(padj) {
    return(padj < 0.05)
}

# create topGOdata object
Ap.BP.GOdata <- new("topGOdata",
                 description = "BP gene set analysis", ontology = "BP",
                 allGenes = Ap.geneList, 
                 geneSel = selectFDR,
                 nodeSize = 10,
                 annot = annFUN.gene2GO, gene2GO = geneID2GO)

#Ap.BP.GOdata

# perform enrichment analysis using parentchild method
Ap.BP.resultParentChild <- runTest(Ap.BP.GOdata, statistic = 'fisher', algorithm = 'parentchild')
Ap.BP.resultParentChild

# table results
Ap.BP.ResTable <- GenTable(Ap.BP.GOdata, parentchild = Ap.BP.resultParentChild, topNodes = 131)
Ap.BP.ResTable
```

As the molecular processes involved in cold and hot temperature tolerance are different, I performed GSEA separately for each response type and compiled results in a single table.

```{r gsea_Ap_high, results = 'hide'}
sig.response.type <- as.data.frame(sig.response.type)
## Genes with *High* expression in both colonies
Ap.high <- sig.response.type[which(sig.response.type$Ar.type == "High" & sig.response.type$A22.type == "High"), "Transcript"]
Ap.geneList.high <- rep(0, length = length(Ap.geneList))
names(Ap.geneList.high) <- names(Ap.geneList)
Ap.geneList.high[(which(names(Ap.geneList.high) %in% Ap.high))] <- 1
# check correct number of values set to 1
table(Ap.geneList.high)
# run GSEA
Ap.high.gsea <- gsea(genelist = Ap.geneList.high, geneID2GO = geneID2GO, plotpath = NA)
```

A single GO term, *`r paste(Ap.high.gsea[, 1:2], collapse = " ")`*, is enriched for **High** genes in both species. 

```{r gsea_Ap_low, results = "hide"}
## Genes with *Low* expression in both colonies
Ap.low <- sig.response.type[which(sig.response.type$Ar.type == "Low" & sig.response.type$A22.type == "Low"), "Transcript"]
Ap.geneList.low <- rep(0, length = length(Ap.geneList))
names(Ap.geneList.low) <- names(Ap.geneList)
Ap.geneList.low[(which(names(Ap.geneList.low) %in% Ap.low))] <- 1
# check correct number of values set to 1
table(Ap.geneList.low)
# Run GSEA
Ap.low.gsea <- gsea(genelist = Ap.geneList.low, geneID2GO = geneID2GO)
```

The GO terms, *`r paste(Ap.low.gsea[, 2], collapse = ", ")`*, are enriched for **Low** genes in both species. 

```{r gsea_Ap_bim, resuts = 'hide'}
# Genes with *Bimodal* expression in both colonies
Ap.bim <- sig.response.type[which(sig.response.type$A22.type == "Bimodal" & sig.response.type$Ar.type == "Bimodal"), "Transcript"]
# create gene list, setting value to 1 for "bim" transcripts
Ap.geneList.bim <- rep(0, length = length(Ap.geneList))
names(Ap.geneList.bim) <- names(Ap.geneList)
Ap.geneList.bim[(which(names(Ap.geneList.bim) %in% Ap.bim))] <- 1
# check correct number of values set to 1
table(Ap.geneList.bim)
# Run GSEA
Ap.bim.gsea <- gsea(genelist = Ap.geneList.bim, geneID2GO = geneID2GO)
```

The GO terms, *`r paste(Ap.bim.gsea[, 2], collapse = ", ")`*, are enriched for **Bimodal** genes in both species. 

```{r gsea_Ap_int, results = 'hide'}
# genes with *Intermediate* expression in both colonies
Ap.int <- sig.response.type[which(sig.response.type$A22.type == "Intermediate" & sig.response.type$Ar.type == "Intermediate"), "Transcript"]
# create gene list, setting value to 1 for "int" transcripts
Ap.geneList.int <- rep(0, length = length(Ap.geneList))
names(Ap.geneList.int) <- names(Ap.geneList)
Ap.geneList.int[(which(names(Ap.geneList.int) %in% Ap.int))] <- 1
# check correct number of values set to 1
table(Ap.geneList.int)
# run GSEA
Ap.int.gsea <- gsea(genelist = Ap.geneList.int, geneID2GO = geneID2GO)
```

The GO terms, *`r paste(Ap.int.gsea[, 2], collapse = ", ")`*, are enriched for **Intermediate** genes in both species. 

```{r Ap_gsea_intersect_save, echo = FALSE, eval = FALSE}
Ap.high.gsea$Type <- "High"
Ap.low.gsea$Type <- "Low"
Ap.bim.gsea$Type <- "Bimodal"
Ap.int.gsea$Type <- "Intermediate"
Ap.gsea.intersect <- rbind(Ap.high.gsea, Ap.low.gsea, Ap.int.gsea, Ap.bim.gsea)

Ap.gsea.intersect <- Ap.gsea.intersect[,c("Type", "GO.ID", "Term", "Annotated", "Significant", "Expected", "parentchild")]
colnames(Ap.gsea.intersect)[which(colnames(Ap.gsea.intersect) == "parentchild")] <- "P"

write.csv(Ap.gsea.intersect, file = paste(resultsdir, "Ap_gsea_intersect_", Sys.Date(), ".csv", sep = ""), row.names = FALSE)
```

GSEA for genes that are **Bimodal** in *A. picea* and downregulated (**Intermediate**) in *A. carolinensis*.

```{r gsea_A22_bim_Ar_int, results = 'hide'}
A22.bim.Ar.int <- sig.response.type[which(sig.response.type$Ar.type == "Intermediate" & sig.response.type$A22.type == "Bimodal"), "Transcript"]
A22.bim.Ar.int.geneList <- rep(0, length = length(Ap.geneList))
names(A22.bim.Ar.int.geneList) <- names(Ap.geneList)
A22.bim.Ar.int.geneList[(which(names(A22.bim.Ar.int.geneList) %in% A22.bim.Ar.int))] <- 1
# check correct number of values set to 1
table(A22.bim.Ar.int.geneList)
# run GSEA
A22.bim.Ar.int.gsea <- gsea(genelist = A22.bim.Ar.int.geneList, geneID2GO = geneID2GO, plotpath = NA)
```

The GO terms, *`r paste(A22.bim.Ar.int.gsea[, 2], collapse = ", ")`*, are enriched for genes that are **Bimodal** in *A. picea* and downregulated (**Intermediate**) in *A. carolinensis*.


### GSEA for each functional type in each species

*A. picea*

```{r gsea_A22_high, results = 'hide'}
# A22 'High' genes
A22.geneList.high <- rep(0, length = length(Ap.geneList))
names(A22.geneList.high) <- names(Ap.geneList)
A22.geneList.high[(which(names(A22.geneList.high) %in% A22.high.transcripts))] <- 1
A22.high.gsea <- gsea(genelist = A22.geneList.high, geneID2GO = geneID2GO)

# A22 'Low' genes
A22.geneList.low <- rep(0, length = length(Ap.geneList))
names(A22.geneList.low) <- names(Ap.geneList)
A22.geneList.low[(which(names(A22.geneList.low) %in% A22.low.transcripts))] <- 1
A22.low.gsea <- gsea(genelist = A22.geneList.low, geneID2GO = geneID2GO)

# A22 'Bim' genes
A22.geneList.bim <- rep(0, length = length(Ap.geneList))
names(A22.geneList.bim) <- names(Ap.geneList)
A22.geneList.bim[(which(names(A22.geneList.bim) %in% A22.bim.transcripts))] <- 1
A22.bim.gsea <- gsea(genelist = A22.geneList.bim, geneID2GO = geneID2GO)

# A22 'Int' genes
A22.geneList.int <- rep(0, length = length(Ap.geneList))
names(A22.geneList.int) <- names(Ap.geneList)
A22.geneList.int[(which(names(A22.geneList.int) %in% A22.int.transcripts))] <- 1
A22.int.gsea <- gsea(genelist = A22.geneList.int, geneID2GO = geneID2GO)
```

*A. carolinensis*

```{r gsea_Ar_high, results = 'hide'}
# Ar 'High' genes
Ar.geneList.high <- rep(0, length = length(Ap.geneList))
names(Ar.geneList.high) <- names(Ap.geneList)
Ar.geneList.high[(which(names(Ar.geneList.high) %in% Ar.high.transcripts))] <- 1
Ar.high.gsea <- gsea(genelist = Ar.geneList.high, geneID2GO = geneID2GO)

# Ar 'Low' genes
Ar.geneList.low <- rep(0, length = length(Ap.geneList))
names(Ar.geneList.low) <- names(Ap.geneList)
Ar.geneList.low[(which(names(Ar.geneList.low) %in% Ar.low.transcripts))] <- 1
Ar.low.gsea <- gsea(genelist = Ar.geneList.low, geneID2GO = geneID2GO)

# Ar 'Bim' genes
Ar.geneList.bim <- rep(0, length = length(Ap.geneList))
names(Ar.geneList.bim) <- names(Ap.geneList)
Ar.geneList.bim[(which(names(Ar.geneList.bim) %in% Ar.bim.transcripts))] <- 1
Ar.bim.gsea <- gsea(genelist = Ar.geneList.bim, geneID2GO = geneID2GO)

# Ar 'Int' genes
Ar.geneList.int <- rep(0, length = length(Ap.geneList))
names(Ar.geneList.int) <- names(Ap.geneList)
Ar.geneList.int[(which(names(Ar.geneList.int) %in% Ar.int.transcripts))] <- 1
Ar.int.gsea <- gsea(genelist = Ar.geneList.int, geneID2GO = geneID2GO)
```

Combined gene set enrichment analysis for each species and category into single table.

```{r gsea_combine, eval = TRUE}
# combine into single table
A22.high.gsea$Type <- "High"
A22.low.gsea$Type <- "Low"
A22.int.gsea$Type <- "Intermediate"
A22.bim.gsea$Type <- "Bimodal"
A22.gsea <- rbind(A22.high.gsea, A22.low.gsea, A22.int.gsea, A22.bim.gsea)
A22.gsea$Species <- "ApVT"

Ar.high.gsea$Type <- "High"
Ar.low.gsea$Type <- "Low"
Ar.bim.gsea$Type <- "Bimodal"
Ar.int.gsea$Type <- "Intermediate"
Ar.gsea <- rbind(Ar.high.gsea, Ar.low.gsea, Ar.int.gsea, Ar.bim.gsea)
Ar.gsea$Species <- "AcNC"

# combine
Ap.gsea.by.species <- rbind(A22.gsea, Ar.gsea)
# reorder
Ap.gsea.by.species <- Ap.gsea.by.species[,c("Species", "Type", "GO.ID", "Term", "Annotated", "Significant", "Expected", "parentchild")]
colnames(Ap.gsea.by.species)[8] <- "P"

write.csv(Ap.gsea.by.species, file = paste(resultsdir, "Ap_gsea_by_species", Sys.Date(), ".csv", sep = ""), row.names = FALSE)
```


```{r rData, echo=FALSE}
# save RData for manuscript file
save(Ap.type.table, annotation.table, annotation.table.responsive, Ap.type.table.prop, 
     chi1, chi2, chi3, chi4, chi5, chi6, D, mh.test, pre.signif.transcripts,  
     responsive_transcripts_w_BLAST, sigtable, signif.transcripts, T_on.low.ttest, 
     T_on.high.ttest, T_off.int.high.ttest, T_off.int.low.ttest, w3, tt2,
     file = paste("doc/ms_", Sys.Date(), ".RData", sep = ""))
```


## Session information
 
```{r session}
save.image()
sessionInfo()
```

## References

```{r references, echo = FALSE, results='asis'}
bibliography()
```
