---
title: Thermal reactionomes reveal adaptive responses to thermal extremes in warm
  and cool-climate ant species
author: ' '
csl: pnas1.csl
output:
  word_document: default
  pdf_document:
    fig_caption: yes
    includes:
      in_header: header.tex
    keep_tex: yes
    latex_engine: xelatex
    number_sections: no
    toc_depth: 2
  html_document: null
includes: header.tex
bibliography: ApTranscriptome.bib
---

\linenumbers

\newpage

**Authors** 

John Stanton-Geddes \textsuperscript{1}, Andrew Nguyen \textsuperscript{1}, Lacy Chick \textsuperscript{2}, James Vincent \textsuperscript{3}, Mahesh Vangala \textsuperscript{3}, Robert R. Dunn \textsuperscript{4}, Aaron M. Ellison \textsuperscript{5}, Nathan J. Sanders \textsuperscript{2,6}, Nicholas J. Gotelli \textsuperscript{1}, Sara Helms Cahan \textsuperscript{1}
           
\vspace{10 mm}           
           
**Affilitiations**

\textsuperscript{1} Department of Biology, University of Vermont, Burlington, Vermont 05405 \newline
\textsuperscript{2} Department of Ecology and Evolutionary Biology, University of Tennessee, Knoxville, Tenessee 37996 \newline
\textsuperscript{3} Vermont Genetics Network, University of Vermont, Burlington, Vermont 05405 \newline
\textsuperscript{4} Department of Biological Sciences, North Carolina State University, Raleigh, North Carolina 27695 \newline
\textsuperscript{5} Harvard Forest, Harvard University, Petersham, Massachusetts 01336 \newline
\textsuperscript{6} Center for Macroecology, Evolution and Climate, University of Copenhagen

\newpage

# Abstract 

The distributions of species and their responses to climate change are in part determined by their thermal tolerances. However, little is known about how thermal tolerance evolves. To test whether evolutionary extension of thermal limits is accomplished through enhancing the cellular stress response (*enhanced defense*), consititutively elevated expression of protective genes (*genetic assimilation*) or shifting from damage resistance to passive mechanisms of thermal stability (*tolerance*), we conducted an analysis of the *reactionome*; the reaction norm for all genes in an organism's transcriptome measured across an experimental gradient. We characterized thermal reactionomes of two common ant species in the eastern U.S the northern cool-climate *Aphaenogaster picea* and the southern warm-climate *Aphaenogaster carolinensis* across 12 temperature intervals that spanned their entire thermal breadth. We found that at least 8% of all genes were thermally responsive. The majority of upregulation was specific to low temperatures, and consistent with the *enhanced defense hypothesis* the cool-adapted *A. picea* induced expression of more genes in response to extreme temperatures than *A. carolinensis*, including heat shock proteins. In contrast, under high temperatures the warm-adapted *A. carolinensis* downregulated many of the genes upregulated in *A. picea* and required more extreme temperatures to induce changes in gene expression, consistent with the *tolerance* hypothesis. We found no evidence for a trade-off between constitutive and inducible gene expression as predicted by the *genetic assimilation hypothesis*. These results are consistent with expectations that adaptive increases in upper thermal limits may require an evolutionary shift in response mechanism away from damage repair toward tolerance and prevention. 


# Significance Statement

Coping with progressive changes in climate requires both short-term physiological acclimation and long-term adaptation. To better understand the relationship between these two mechanisms, we compared gene expression as a function of temperature in two closely-related ant species: the cool-climate *Aphaenogaster picea* and the warm-climate *A. carolinensis*. We found substantial differences in their patterns of gene expression. *Aphaenogaster picea* responded to extreme temperatures primarily through upregulation, with a robust heat shock response. In contrast, *A. carolinensis* required more extreme temperatures to trigger a protective response and downregulated metabolic genes. These results suggest that adaptation to chronic high temperatures is not just a more extreme version of the acclimation response, but instead involves a qualitative shift in mechanisms from those used to cope with rarely encountered stressors.


# Introduction

Temperature regulates biological activity and shapes diversity from molecular to macroecological scales [@brown2004; @kingsolver2009]. Many species, especially small-bodied arthropods, live at temperatures close to their thermal limits and are at risk from current increases in temperature [@deutsch2008; @kingsolver2013; @sunday2014]. Thermal tolerance, the ability of individuals to maintain function and survive thermal extremes, depends on a complex interplay between the structural integrity of cellular components and activation of physiological response mechanisms to prevent and/or repair damage [@huey1989; @richter2010]. Thermal defense strategies are shaped by the environmental regime organisms experience [@angilletta2003] and thermal limits vary considerably among species and populations [@cowles1939; @deutsch2008; @kingsolver2013; @hoffmann2013]. These differences in thermal tolerance are largely genetic [@krebs1997; @kellermann2012] with a highly polygenic basis [@krebs1998; @williams2012; @morgan2006; @takahashi2011]. Outside of candidate genes [@krebs1998], little is known about the evolution of thermal tolerance or the link between short-term physiological acclimation and longer-term adaptation to novel temperature regimes. This information is critical for understanding the adaptive potential of species to future climates [@hoffmann2008].

To address this gap of knowledge, we need information on the extent to which selection has acted upon the diversity and plasticity of genes involved in thermal tolerance [@hoffmann2008; @somero2011]. In recent years, whole organism gene expression approaches (e.g. transcriptomics) using high-throughput RNA sequencing (RNAseq) technology have been widely applied to identify genes involved in thermal tolerance [@meyer2011; @teets2012; @barshis2013; @oneil2014] and other traits. Such studies typically use an ANOVA-type experimental or sampling design, with only a few environmental levels, and often find only a few dozen to hundred genes with differential expression in different thermal regimes. However, temperature and other environmental factors vary continuously in nature. As a result, such comparisons (e.g. high vs. low temperatures) are likely to miss key differences that are due not just to whether it is hot, but rather how hot it is. Continuous variation is better characterized with a reaction norm approach, which describes variation in the phenotype of a single genotype across an environmental gradient [@gomulkiewicz1992]. Reaction norms differ not only in mean values, but also in their shapes [@hoffmann2013; @murren2014], and differences in the shape of reaction norms are often greater than differences in mean values at both the species and population levels [@murren2014]. 

Here, we extend the reaction norm approach to RNAseq analysis with the *reactionome*, which we define as a characterization of the reaction norm for all genes in an organism's transcriptome across an environmental gradient. We use this method to identify genes that are thermally responsive in two closely-related eastern North American ant species, *Aphaenogaster carolinensis* (part of the *A. rudis*-species complex, B. DeMarco personal communication) and *A. picea* [@umphrey1996]. *Aphaenogaster* are some of the most common ant in eastern North America [@king2013] where they are keystone seed dispersers [@ness2009; @zelikova2011; @rodriguez-cabal2012]. *Aphaenogaster carolinensis* experiences warmer (mean annual temperature (MAT) at collection site = 14.6°C) and less variable (temperature seasonality = 7,678) temperatures than *A. picea* (MAT = 4.6°C, seasonality = 10,008; [@hijmans2005]), with correspondingly higher critical maximum temperatures (44.7°C for *A. carolinensis* versus 41.4°C for *A. picea*; see Methods) and minimum temperatures (6.1°C for *A. carolinensis* versus -0.1°C for *A. picea*) allowing us to investigate adaptation to both lower and upper thermal extremes in this system.

```{r climate, echo=FALSE, results='hide', message=FALSE, warning=FALSE}
library(raster)
w <- getData('worldclim', var='bio', res=10)
#names(w)
# Plot first bioclim dimension (MAT) to check download worked
#plot(w, 1, col=colorRampPalette(c("blue","green","red"))(255))
# sites
sites <- data.frame(site = c("Ac", "Ap"), lon=c(-78.874, -72.702), lat= c(36.037, 44.508))
# extract bioclimatic variables from worldclim data. Note bioclim temperature data is multiplied by 10 (no decimal), so transform by dividing by 10.
dbio <- extract(w, sites[,c("lon","lat")])
# bio1 = MAT
# bio4 = Temperature Seasonality (standard deviation *100)
dbio.vars <- cbind(sites, dbio[,1], dbio[,4])
colnames(dbio.vars)[which(colnames(dbio.vars) == "dbio[, 1]")] <- "MATx10"
dbio.vars$MAT <- dbio.vars$MATx10/10
colnames(dbio.vars)[which(colnames(dbio.vars) == "dbio[, 4]")] <- "Seasonality"
dbio.vars
```

To characterize the thermal reactionome, we collected whole organism RNAseq data at 12 temperatures (every 3.5°C between 0 and 38.5°C) using Illumina short read sequencing technology, generated the *de novo* transcriptome using Trinity [@grabherr2011], quantified gene expression using Sailfish [@patro2014], and characterized the reaction norm for each gene using a custom analysis. 

We used these data to quantitatively test three mechanistic hypotheses of thermal adaptation. First, the *enhanced defense hypothesis* [@hofmann1996; @feder1999; @kultz2005] proposes that species extend their thermal limits through a stronger induced responses to provide greater protection from more frequently encountered stressors. This hypothesis would predict that the cool-adapted *A. picea* would induce more genes in response to low temperatures than would the warm-adapted *A. carolinensis*, which would induce more genes in response to high temperatures. Additionally, upregulation of molecular pathways shared by both species should also be stronger for more commonly experienced thermal extremes; low temperatures for *A. picea*, high temperatures for *A. carolinensis*. Second, the *tolerance hypothesis* [@cowles1939; @fields2001] proposes that existing inducible stress responses become insufficient or prohibitively costly as environmental stressors increase in frequency, resulting in a shift away from induced response in favor of structural changes [@lockwood2012] or behavioral adapations [@gordon2013; @sunday2014]. This hypothesis predicts adaptation to stress should be associated with lower transcriptional responsiveness and less sensitivity to temperature perturbation, as well as a shift to an alternate suite of tolerance genes and pathways. Finally, the *genetic assimiliation hypothesis* [@waddington1953; @sikkink2014] proposes that exposure to more extreme stressors selects for a shift from inducible to constitutive expression of stress-response genes. This hypothesis predicts that transcripts responsive to high temperatures in *A. picea* will have higher constitutive expression in *A. carolinensis*, whereas transcripts responsive to low temperatures in *A. carolinensis* will have higher constitutive expression in *A. picea*. Of course, these hypotheses are not mutually exclusive. By taking a reactionome approach, we can quantify the relative importance of each of these mechanisms to thermal adapatation.


# Results

```{r load, echo=FALSE, results='hide', message=FALSE, warning=FALSE}
#options(scipen = 1, digits = 3)
load("ms_slim_2014-10-22.RData")
library(data.table)
library(ggplot2)
library(reshape2)
library(pander)
library(RColorBrewer)
library(topGO)

options(digits = 3)
panderOptions('round', 3)
panderOptions('keep.trailing.zeros', TRUE)
```

## Reaction norms of thermally-responsive transcripts differ between species

The combined *Aphaenogaster* transcriptome assembly contained 98,186 transcripts. About half of these (51,246) transcripts had a signficant BLAST hit, of which 50% (25,797) had a top hit to Insecta and 37% (18,854) had a top hit to Formicidae. Of the Formicidae hits, we identified `r prettyNum(sum(sigtable[sigtable$Coefficient %in% c("Temperature", "Temperature:Colony"), "Number.significant"]), big.mark = ",", scientific = FALSE)` transcripts, or 8% of the total, with expression patterns that were thermally responsive at a 5% False Discovery Rate [@benjamini1995]. Of these 9,052 transcripts, 80% (...) had a non-linear relationship with temperature that would likely have been missed with a standard differential expression experiment (e.g. high vs. low temperature). The proportion of responsive transcripts is the same if we focus only on those transcript with a BLAST hit (4,618 transcripts out of 51,246). However, as with all *de novo* transcriptome assemblies, this assembly is fragmented due to partial contigs and alternative transcripts [@vijay2013] suggesting that this estimate of 8% is likely a lower bound for the true proportion of genes that are thermally responsive. 

```{r type_table, echo=FALSE, results='asis', fig.cap = "Table of the number of thermally-responsive transcripts by expression type for *A. carolinensis* and *A. picea*. **Low** are transcripts with increased expression at low temperatuers (< 10°C), **Intermediate** are transcripts with maximum expression between 10 - 30°C, **High** are transcripts with increased expression at high temperatures (> 31°C), **Bimodal** are transcripts with increased expression at both low and high temperatures, while **NotResp** are transcripts that are not thermally responsive in one species but are in the other species."}

rownames(Ap.type.table) <- list("A. picea", "A. carolinensis")

pandoc.table(Ap.type.table[,c("Low", "Intermediate", "High", "Bimodal", "NotResp")], style="rmarkdown",
             big.mark = ",", split.tables = Inf, 
             caption = "Table of the number of thermally-responsive transcripts by expression type for *A. carolinensis* and *A. picea*. **Low** are transcripts with increased expression at low temperatures (< 10°C), **Intermediate** are transcripts with maximum expression between 10 - 30°C, **High** are transcripts with increased expression at high temperatures (> 31°C), **Bimodal** are transcripts with increased expression at both low and high temperatures, while **NotResp** are transcripts that are not thermally responsive in one species but are in the other species.")
```

Though the total number of thermally-responsive transcripts did not differ between species ($\chi^2$~`r chi1$parameter`~ = `r round(chi1$statistic, 2)`, *P* = `r round(chi1$p.value, 2)`) there was significant qualitative and quantitative variation in thermal responses among transcripts and between the two species. To examine this, we grouped the thermally-responsive transcripts for each species into five expression categories based on their reaction norms: transcripts that had greatest expression at high (> 31°) temperatures (**High**), low (< 10°C) temperatures (**Low**), intermediate (10 - 30°C) temperatures (**Intermediate**), both high and low temperatures (**Bimodal**) or were not thermally responsive in that species (**Not Responsive**). For the **Bimodal** group, we required that expression at both low and high temperatures was at least one standard deviation greater than the expression at midpoint (19.5°C). The majority of the responsive genes were in the **Low** category for both species (42% and 54% for *A. carolinensis* and *A. picea*, respectively), followed by **Intermediate** (29%), **High** (12%),**Bimodal** (10%) and **Not Responsive** (8%) for *A. carolinensis*, and **Bimodal** (17%), **High** (14%), **Intermediate** (10%) and **Not Responsive** (6%) for *A. picea*. 

## Expression response to thermal extremes differs between species

The two species differed in the number of transcripts in each response category (Table 1, $\chi^2$~`r chi1$parameter`~ = `r chi1$statistic`, *P* < `r max(chi1$p.value, 0.001)`). *Aphaenogaster picea* induced significantly more transcripts in response to both temperature extremes (**Bimodal** transcripts in Table 1; $\chi^2$~`r chi6$parameter`~ = `r chi6$statistic`, *P* < `r max(chi6$p.value, 0.001)`) than did *A. carolinensis*, which downregulated more transcripts under these conditions (**Intermediate** transcripts in Table 1; $\chi^2$~`r chi4$parameter`~ = `r chi4$statistic`, *P* < `r max(chi4$p.value, 0.001)`). Consistent with the *enhanced defense* hypothesis, *A. picea* induced 1,146 (30%) more transcripts in response to low temperatures than *A. carolinensis* (**Low** transcripts in Table 1; $\chi^2$~`r chi3$parameter`~ = `r chi3$statistic`, *P* < `r max(chi3$p.value, 0.001)`). However, consistent with the predictions of the *tolerance hypothesis* but not the *enhanced defense hypothesis*, at high temperatures *A. carolinensis* induced expression of ~14% fewer transcripts than *A. picea* (**High** transcripts in Table 1; $\chi^2$~`r chi5$parameter`~ = `r chi5$statistic`, *P* < `r max(chi5$p.value, 0.001)`). 

```{r echo=FALSE, results='hide'}
# results of Stuart-Maxwell test
mh.test
```

In addition to comparing the overall number of transcripts in each category, we also examined the extent to which reaction norms for individual transcripts were shared between species. Expression patterns differed for 6,792 (65%) of the thermally responsive transcripts, most of which shifted from one responsive category to another. These shifts were not randomly distributed (Stuart-Maxwell test of marginal homogeneity $\chi^2$~4~ = 1,259, *P* < 0.001, Figure S1) indicating that these differences are not simply due to neutral processes. The *enhanced defense* and *tolerance* hypotheses make opposing predictions concerning overlap in response patterns between the two species. The *enhanced defense hypothesis* posits that temperature adaptation uses existing defense mechanisms, which should result in significant overlap in response, whereas the *tolerance hypothesis* predicts that genes involved in active defense will become non-responsive or shift to other response categories in the better-adapted species. For each temperature extreme, we tested these predictions by focusing on the genes upregulated by the less-adapted species (*A. carolinensis* for low temperatures, *A. picea* for high temperatures) and testing whether those genes displayed the same response profile in the better-adapted species. As with the overall results, the patterns of association differed for the two temperature extremes. Genes upregulated at low temperatures in *A. carolinensis* (**Low** and **Bimodal** transcripts) were significantly biased toward this same category and away from all other expression categories in *A. picea*, suggesting shared response pathways (Fig. 1A). In contrast, genes upregulated in response to high temperatures in *A. picea* (**High** and **Bimodal**) were only weakly associated with this same category in *A. carolinensis* (Fig. 1B). Instead, these genes were less likely than expected to be upregulated in any context, with a significant excess in the **Not Responsive** and **Intermediate** categories (Fig. 1B).
 
![Deviations from expected numbers of transcripts in matched observation data for transcript expression type data. The color of each cell represents the deviation of the observed from the expected number of transcripts based on hypothetical equivalence of the marginal frequencies (Blue = excess, Red = deficit). Shown are **Low**, **High** and **Bimodal** expression types for (A) *A. picea* and (B) *A. carolinensis* against all expression types in the opposite species. The dashed boxes indicate the diagonals (e.g. **High** - **High**).](../results/matched_observations_Fig1.png)
 
The molecular processes in the set of transcripts that switched from upregulation in *A. picea* to downregulation in *A. carolinensis* included `r nrow(Ar.int.A22.low.gsea) + nrow(Ar.int.A22.bim.gsea)` GO terms (Table S1), of which `r round((length(grep("metab", Ar.int.A22.low.gsea$Term)) + length(grep("metab", Ar.int.A22.bim.gsea$Term))) / (nrow(Ar.int.A22.low.gsea) + nrow(Ar.int.A22.bim.gsea))*100, 0)`% (`r (length(grep("metab", Ar.int.A22.low.gsea$Term)) + length(grep("metab", Ar.int.A22.bim.gsea$Term)))`) were related to metabolism. In addition, there were GO terms related to repair (e.g. histone modification), transcript regulation (e.g. positive regulation of cellular process) and cellular organization (e.g. transport, organelle organization). Altogether, these results suggest that in response to both thermal extremes, the warmer-climate species, *A. carolinensis*, primarily shuts down metabolism whereas the cooler-climate species, *A. picea*, increases metabolism and initiates damage repair responses. 

## *A. carolinensis* has greater inertia of expression change to increases in temperature than does *A. picea*

As an additional test of the *tolerance hypothesis*, we examined the critical temperature of gene induction in response to increasing and decreasing temperatures. We compared between species the mean temperatures of transcript upregulation, defined as the temperature at which the transcript showed the greatest positive change in expression. In support of the *tolerance hypothesis* but not the *enhanced defense hypothesis*, transcripts turned on at high temperatures were induced on average at `r round(T_on.high.ttest$estimate[1], 1)`°C in *A. carolinensis* compared to `r round(T_on.high.ttest$estimate[2], 1)`°C in *A. picea* (*t*~`r round(T_on.high.ttest$parameter, 0)`~ = `r round(T_on.high.ttest$statistic, 1)`, *P* < `r max(T_on.high.ttest$p.value, 0.001)`; Fig. 2). For low temperatures, there was no difference in the mean temperature of transcript upregulation (`r round(T_on.low.ttest$estimate[1], 1)`°C for *A. carolinensis* compared to `r round(T_on.low.ttest$estimate[2], 1)`°C for *A. picea*; *t*~`r prettyNum(T_on.low.ttest$parameter, digits=2)`~ = `r round(T_on.low.ttest$statistic, 1)`, *P* < `r round(max(T_on.low.ttest$p.value, 0.001), 3)`; Fig. 2). 

![Histogram with smooth density estimate of temperature of maximum change in expression for transcripts that have increased expression at **Low** and **High** temperatures. Red bars and lines are for *A. carolinensis* while blue bars and lines are for *A. picea*.](../results/crit_temp_regulation_Fig2.png)

Similarly, for the **Intermediate** transcripts we compared the mean temperatures of transcript downregulation of transcript expression between species at both high (> 21°) and low (< 21°C) temperatures. Thresholds for downregulation were shifted toward cooler temperatures than were the corresponding thresholds for gene activation, with a shift of approximately ... at the higher temperature extremes and a shift of ...  at low temperature extremes. Again, there was greater inertia of gene expression only in response to high temperature extremes. With increasing temperatures, **Intermediate** transcripts were downregulated at `r round(T_off.int.high.ttest$estimate[1], 1)`°C for *A. carolinensis* compared to `r round(T_off.int.high.ttest$estimate[2], 1)` for *A. picea* (*t*~`r round(T_off.int.high.ttest$parameter, 0)`~ = `r round(T_off.int.high.ttest$statistic, 1)`, *P* < `r round(max(T_off.int.high.ttest$p.value, 0.001), 3)`), whereas with decreasing temperatures the species did not differ in mean temperature of downregulation (*A. carolinensis* `r round(T_off.int.low.ttest$estimate[1], 1)` compared to *A. picea* `r round(T_off.int.low.ttest$estimate[1], 1)`; *t*~`r round(T_off.int.low.ttest$parameter, 0)`~ = `r round(T_off.int.low.ttest$statistic, 1)`, *P* = `r round(max(T_off.int.low.ttest$p.value, 0.001), 3)`).

## No evidence for *genetic assimilation*

We tested the *genetic assimilation hypothesis*, a shift from inducible to constitutive expression of stress-response genes, by comparing the log ratios of relative inducibility to relative constitutive expression. Specifically, this hypothesis predicted a negative relationship between the two. We found no evidence of such a relationship for either temperature extreme; transcripts upregulated at high temperatures in *A. picea* (Fig. 3A) or transcripts upregulated at low temperatures in *A. carolinensis* (Fig. 3B). In fact, for both comparisons we found a weakly positive relationship. 

![Scatterplots of (A) log ratio of inducibility in *A. picea* over inducibility in *A. carolinensis* against log ration of constitutive expression in *A. carolinensis* over constitutive expression in *A. picea* for transcripts with **High** expression in *A. picea*, and (B) log ratio of inducibility in *A. carolinensis* over inducibility in *A. picea* against log ration of constitutive expression in *A. picea* over constitutive expression in *A. carolinensis*. Blue lines and confidence intervals are from are ordinary least squares regressions.](../results/genetic_assimilation_Fig3.png)

In addition, the thermally responsive transcripts in *A. carolinensis*, regardless of expression pattern, had higher constitutive expression than those in *A. picea*, including those with **Intermediate** expression profiles in both species (Wilcoxon V = `r prettyNum(w3$statistic)`, *P* < `r max(w3$p.value, 0.001)`). An important exception to this pattern is the set of transcripts that had **High** or **Bimodal** expression in *A. picea* but were not thermally responsive in *A. carolinensis* (blue cells in bottom row of Fig. 1). Rather than having constitutively elevated expression as predicted, these genes were uniformly low or not expressed (mean TPM = 0.08) in *A. carolinensis*. 



# Discussion

The potential for many species to persist in face of climate change will depend in part upon their thermal tolerances. However, for most species little is known about how plasticity or adaptive changes in gene expression underlie thermal tolerance. By using a *reactionome* approach, we were able to quantitatively describe plasticity in transcript expression across a thermal gradient, and identify evolutionary changes in gene expression associated with shifts in thermal tolerance between the ant species *Aphaenogaster picea* and *A. carolinensis*. We found large-scale, non-linear patterns of gene expression changes in response to temperature, with both quantitative and qualitative differences between species, consistent with different mechanisms of thermal adaptation to low and high temperature extremes. 

Under the *enhanced defense* hypothesis, stress-adapted species are hypothesized to induce a stronger and earlier response to extreme conditions. We found evidence for this hypothesis at low temperatures: although the lower thermal limit for *A. picea* is substantially lower than *A. carolinensis*, the two species upregulated responsive transcripts at the same threshold temperature (Fig. 2A). Moreover, the transcripts upregulated in *A. picea* included nearly the entire set upregulated in *A. carolinensis* as well as an additional set of ~1,000 transcripts (Table 1), enriched for metabolism and damage repair functions (Table S1). Two non-mutually exclusive hypotheses may explain this pattern. First, surviving prolonged low temperatures, such as would be experienced during overwintering, generally requires advance production of specialized cryoprotectants [@neelakanta2012] and a suite of preparatory physiological modifications [@denlinger2002]. The northern species *A. picea* may induce a greater response to survive the longer winter period. Alternatively, the response to low temperatures may reflect countergradient expression to counteract reduction in enzyme efficiency, and maintain activity as temperature declines [@lockwood2012]. This requirement may be under stronger selection in *A. picea* given the shorter growing season that would necessitate foraging under a broader range of temperatures. 

In contrast to cold tolerance, the enhanced upper thermal limit in *A. picea* is best explained by the *tolerance* hypothesis. High temperatures were associated with significantly fewer upregulated transcripts in *A. carolinensis*, and a large proportion (361 of 1,238 transcripts) of those transcripts upregulated in *A. picea* were either downregulated or expressed at negligible levels overall in *A. carolinensis*. Similarly, significantly higher temperatures were required to induce the expression response for *A. carolinensis* than *A. picea* towards the high thermal limit (Fig. 2B-D). This difference between species is also evident among candidate genes known to be involved in stress response. Of the `r nrow(hsp_all)` transcripts annotated to heat shock proteins in the transcriptome, `r length(which(hsp_responsive$A22.type %in% c("Bimodal", "High", "Low")))` were upregulated to thermal extremes in *A. picea* whereas only `r length(which(hsp_responsive$Ar.type %in% c("Bimodal", "High", "Low")))` were upregulated in *A. carolinensis*. Altogether, these results suggest that mechanisms other than the heat shock response are acting to maintain protein stability in face of temperature increases, which may include constitutive defenses [@barshis2013],  enhanced proteome stability [@ghosh2010] or behavioral changes [@sunday2014] to tolerate thermal stress. These differences are in line with expectations that *A. carolinensis*, with a growing season over twice the length of its northern congener, may be better able to afford to restrain from foraging in suboptimal conditions. In fact, restrained foraging under stressful conditions by the red harvester ant *Pogonomyrmex barbatus* has been shown to increase colony fitness [@gordon2013].

The one hypothesis that did not receive support was the *genetic assimilation hypothesis*, which predicts that exposure to more frequent stressors will select for a shift from inducible to constitutive expression of stress-response transcripts. This constrasts with other recent studies on adaptation in field populations to thermal stress [@barshis2013], although in a short-term selection experiment for heat tolerance, Sikkink et al. [-@sikkink2014] also found no evidence for genetic assimilation at the expression level after 10 generations of selection for heat tolerance in *Caenorhabditis remanei*, even though there was a substantial increase in heat tolerance and loss of plasticity in the evolved populations. Both the genetic assimilation and tolerance routes routes to increasing thermal limits are functionally similar in that they emphasize damage prevention rather than repair. Whether a particular taxon evolves one strategy over another may be related to the intensity, frequency and duration of temperature stress in a given environment. 

Given the differences in the patterns of thermal responsiveness between species (Fig. 1, 2), it is worth noting a number of similarities. In both species, there are 3 -- 4 times more transcripts upregulated at low than high temperatures  (Table 1). A potential explanation for this pattern is that selection has been stronger on upper than lower thermal limits, thereby reducing genetic variation at high temperatures [@kingsolver2013; @kelly2013]. This explanation is consistent with the observation in *Aphaenogaster rudis* [@warren2013] and other ectotherms [@addo-bediako2000; @hoffmann2013] that critical maximum temperatures vary less among taxa than do critical minimum temperatures. Among closely-related *Drosophila* species, limits to thermal maxima appear to be due to a scarcity of genetic variation in thermal tolerance [@kellermann2012]. To the extent that the number of thermally responsive genes is related to additive genetic variation, our results also suggest that adaptability to high temperatures is more genetically limited than low temperatures, perhaps placing these northern species at risk of competitive replacement by more heat-tolerant southern species [@rodriguez-cabal2012]. The two species also showed similar numbers of genes with a **Bimodal** response (Table 1), indicating a generalized stress-response role. Critical maximum and minimum temperatures are hypothesized to be genetically correlated [@addo-bediako2000; @hoffmann2013], but this was not evident in terms of gene expression.  Only 15-20% of transcripts upregulated in response to temperature are bimodal, and although upper activation thresholds differed significantly between species, lower thresholds were virtually identical (Fig. 2). This suggests that species do not face a fundamental trade-off between these two limits and may be able to shift upper and lower thermal limits independently to match requirements of more seasonally variable environments. 

A major contribution of this study is the construction of a reactionome for gene expression data (but see Hodgins-Davis et al [-@hodgins-davis2012] for a similar approach using differential expression among many levels). A number of issues have hampered RNA-seq studies to date. Namely, lists of differentially expressed transcripts are prone to false positives [@gonzalez2013], depend on the genetic background of the organism [@sarup2011], identify transcripts that are correlated, but not necessarily causative, with phenotypes [@feder2005] and are prone to "storytelling" [@pavlidis2012]. Our findings are robust to these issues as we focus on the average change in the shape of the reaction norms across many hundreds of responsive transcripts in each species. While we use gene ontology information to interpret our results, the key findings about differential plasticity of expression between species do not depend on functional annotation. Thus, our method provides an example of how RNA-seq data can be used to understand gain meaningful insight on the process of evolution.

Moreover, by characterizing responses across thousands of transcripts, the reactionome approach can help to distinguish selection from neutral drift in gene expression [@khaitovich2004; @whitehead2006; @ogasawara2009]. While we cannot rule out drift as a source of variation for individual transcripts, we would not expect to see systematic differences in , expression type categories or critical temperature thresholds as we do here (Fig. 1, Fig. S1). Thus, our method provides an example of how focusing on transcriptome-wide changes in gene expression, as opposed to identifying lists of differentially-expressed genes, can provide meaningful insight on the process of evolution. 

As with all *de novo* transcriptomes, our work has some caveats. First, given that all sequenced ant genomes to date have about only 18,000 genes [@gadau2012], our transcriptome is highly fragmented. Although we took steps to remove contaminants and redundant transcripts, some likely remain, in addition to partially assembled transcripts. A genome assembly, in progress, will help to reduce fragmentation. Second, the quality of the annotation for a non-model system such as *Aphaenogaster* is not as good as it would be for the model organisms *Drosophila* and *Apis*. Finally, the mapping of changes in gene expression to organismal fitness is far from direct [@feder2005], and large differences in patterns of gene expression may have only small effects on organismal fitness. 

Our results are congruent with evidence from other systems [@barshis2013] that thermally-stressful habitats select for investment in tolerance, while organisms from less stressful environments rely on plastically-induced resistance. Although the heat-shock response is one of the most conserved across living organisms [@kultz2005], it is energetically expensive, particularly under chronic stress conditions [@hoekstra2013].  Under such circumstances, it may be advantageous to proactively prevent thermal damage even at the cost of reduced metabolic efficiency, either by maintaining a higher constitutive level of chaperone proteins (ref) or by increasing the thermal stability of proteins at the expense of catalytic activity (Arnold et al. 2001).  Thus, although in the short term increasing temperature stress leads to a quantitatively stronger induced response, adapting to such stress over evolutionary time appears to require a qualitative shift in mechanism of resistance that can alter not only the magnitude, but the sign of gene expression change in response to temperature.  Whether such a shift would be possible in the compressed time frame of projected climate change, particularly for long-lived organisms such as ants, is likely to be critical in determining the capacity of populations to adapt to more frequent and long-lasting stressors. 

## Conclusion

In this work, we have brought reaction norms to the genomic era by characterizing the thermal reactionomes of two temperate ant species, *Aphaenogaster picea* and *A. carolinensis*. At least 10% of their transcriptomes are thermally responsive. For both species, more transcripts responded to low than high temperatures. Together with physiological data showing more variation in critical minimum than maximum temperatures, characterization of the reactionome suggests that these species, and likely other ectotherms, have more limited genetic accomodation to high than low temperatures. Our results also indicate that these two ant species have different responses to thermal extremes. *A. picea* responds by increasing metabolism and activating heat shock proteins and other protective molecules, whereas *A. carolinensis* shuts down metabolism and must rely on other mechanisms for thermal tolerance. The thermal reactionomes of these two species provide key insights into the genetic basis of thermal tolerance, and a resource for the future study of ecological adaptation in ant species. Finally, the reactionome itself illustrates a new direction for characterizing acclimation and adaptation in a changing climate. 


# Material and Methods

## Samples

Ants of the genus *Aphaenogaster* are some of the most abundant in eastern North America [@lubertazzi2012], and species as well as populations within species differ in the critical maximum and minimum temperatures [@warren2013]. Temperature is a potentially strong selective force for ground-nesting ant populations, which must tolerate seasonally freezing winters and hot summers. On shorter time scales, individual workers can experience extreme thermal environments when they leave the thermally buffered ant nest to forage for food [@diamond2012]. 

We used colonies of the ant species *Aphaenogaster picea*, collected from Molly Bog, Vermont (University of Vermont Natural Areas; 44.508°N, -72.702°W) and *Aphaenogaster carolinensis*, collected from Durham, North Carolina (36.037° N -78.874° W). The phylogeny of *Aphaeogaster* is currently unresolved [@umphrey1996], but putative species identification was possible based on morphology (Bernice DeMarco, pers. communication), and there is about 2% pairwise divergence between these clades using five nuclear genes (Bernice DeMarco, unpublished data). Colonies of these species were maintained in common conditions at ?? for 6 months prior to experimentation. 

From each species, we haphazardly collected 12 ants from a single colony at the same time on 12 days. Each day, the 12 ants were placed in glass tubes into a water bath maintained at one of 12 randomly-assigned temperatures (0° to 38.5°C, in 3.5° increments) for one hour. The minimum and maximum temperatures were selected based on previous work showing that these temperatures are close to the critical minimum and maximum temperatures for *Aphaenogaster* [@warren2013]. At the end of the hour, the ants were flash frozen in liquid nitrogen and stored at -80°C. We extracted mRNA by homogenizing three pooled ants in 500 uL of RNAzol buffer with zirconium silicate beads in a Bullet Blender (Next Advance; Averill Park, NY), followed by RNAzol extraction (Molecular Research Center Inc; Cincinnati, OH) and then an RNeasy micro extraction (Qiagen Inc; Valencia, CA) following the manufacturer's instructions. 

## Sequencing, assembly and annotation

For each species, the 12 samples were barcoded and sequenced in a single lane of 2 x 100bp paired-end reads on an Illumina HiSeq 1500 yielding 200 and 160 million reads for the *A. picea* and *A. carolinensis* samples respectively. Reads were filtered to remove Illumina adapter sequences and low quality bases using the program Trimmomatic [@lohse2012]. 

We assembled the sequenced reads into the full set of mRNA transcripts, the transcriptome, for the combined data set from both species using the Trinity *de novo* transcriptome assembly program [@grabherr2011]. *De novo* transcriptome assembly is prone to falsely identifying alternative transcripts and identifying inaccurate transcripts that are chimeric (e.g. regions of two separate transcripts that assemble into a false, or chimeric, third transcript) [@yang2013]. We removed potentially false transcripts by first running the program `CAP3` [@huang1999] to cluster sequences with greater than 90% similarity and merge transcripts with overlaps longer than 100 bp and 98% similar in length. Second, we ran the program `uclust` which clusters sequences completely contained within longer sequences at greater than 90% similarity (see Supplemental Information for exact parameters). We used liberal values (90% similarity) to merge orthologous transcripts in the two species that may not have assembled together in the initial *de novo* transcriptome assembly. To identify contaminant sequences, we screened our full transcriptome using the program `DeconSeq` [@schmieder2011] with the provided bacteria, virus, archae and human databases of contaminants. 

The Trinity *de novo* transcriptome assembly for both species assembled together included 126,172 transcripts with a total length of 100 million bp. Filtering to remove redundant or chimeric reads resulted in an assembly with 105,536 transcripts. The total length was 63 million bp with an N~50~ length of 895 bp and a mean transcript size of 593 bp. Of the 105,536 filtered transcripts, 55,432 had hits to the NCBI-nr database. Of these, 38,711 transcripts mapped to GO terms, 1,659 transcripts were identified to an enzyme and 18,935 transcripts mapped to a domain with >50% coverage. We removed 5,675 transcripts identified as known contaminants, leaving 99,861 clean transcripts. Of these, 47% have best hits to transcripts from the class Insecta [(Supplementary Krona plot)](http://johnstantongeddes.org/assets/files/ApTranscriptome_FA.html).

To determine the putative function of the transcripts, we used functional annotation of the transcriptome assembly using the web-based tool `FastAnnotator` [@chen2012] which annotates and classifies transcripts by Gene Ontology (GO) term assignment, enzyme identification and domain identification. 

## Thermally-responsive transcripts

We quantified expression of each transcript using the program `Sailfish` [@patro2014] and use the bias-corrected transcripts per million (TPM) [@wagner2012] as our measure of transcript expression. We included the contaminant transcripts identified by DeconSeq at the quantification stage to avoid incorrectly assigning reads to other transcripts, but removed these from further analyses. As preliminary examination of the data (see supplemental...) indicated that 7°C samples may have been mis-labeled, we omitted these data from the analysis. The expression values were highly correlated between species at each temperature treatment (r^2^ > 0.98) indicating that assembling the transcriptome with data from both species was justified (Supplemental).

To identify transcripts that have significant changes in expression across the thermal gradient, we fit the regression model

$$ log(TPM + 1) = \beta_0 + \beta_1(species) + \beta_2(temp) + \beta_3(temp^2) + \beta_4(species * temp) + \beta_5(species * temp^2) + \epsilon $$
    
to each transcript. We used $log(TPM + 1)$ as the response to control for skew in the expression data. For a continuous predictor such as temperature, this regression approach is preferred to an ANOVA approach as it can reveal non-linear responses such as hump-shaped or threshold effects [@cottingham2005]. This method is robust to over-dispersion because we expect errors in the read count distribution [@anders2010] to be independent with respect to temperature. To correct for multiple testing, we calculated adjusted *P* values using the False Discovery Rate (FDR) approach of Benjamini and Hochberg [-@benjamini1995], retaining transcripts under the 5% FDR threshold.

We identified thermally-responsive transcripts as the subset under the FDR threshold that had significant $\beta_2(temp)$, $\beta_3(temp^2)$, $\beta_4(species * temp)$ or $\beta_5(species * temp^2)$ terms after step-wise model selection by AIC. All analyses were performed in R [@r_core_team2013] and are fully reproducible (Supplemental file). 

For each thermally-responsive transcript, we predicted expression levels across the tested thermal range using the final linear model separately for each species. We used the predicted transcript expression values to partition the thermally-responsive transcripts into five expression categories for each species; transcripts that had greatest expression at high (> 31°) temperatures (**High**), low (< 10°C) temperatures (**Low**), intermediate (10 - 30°C) temperatures (**Intermediate**), both high and low temperatures (**Bimodal**) or were not thermally responsive in that species (*NotResp*). For the **Bimodal** group, we required that expression at both low and high temperatures was at least one standard deviation greater than the expression at the mean temperature of 19.5°C.

## Species-level comparisons

We performed comparative analyses between the two species focusing on the transcripts that had a significant species x temperature interaction to determine if the two species differed in the tempertaure of transcript activation and sensitivity of the thermally-responsive transcripts. For each transcript, we predicted expression levels across temperatures for each species using the full linear model, and then grouped transcripts into the four responsive categories, **High**, **Low**, **Intermediate**, and **Bimodal**, as well as a fifth category **Not Responsive**. To test whether the temperature at which thermally-responsive transcripts are activated differs between species, we identified the temperature at which there was the greatest change in expression for each transcript in each species. We grouped the **High** transcripts along with the high temperature end of the **Bimodal** transcripts, and did the same for **Low** transcripts. We then performed a *t*-test to determine if the mean temperature of transcript activation differed between the two species for each group. 

To determine if the sensitivity of transcripts to thermal extremes differed between the species, we used the subset of **Intermediate** and **Bimodal** transcripts in each species. For each **Intermediate** transcript in each species, we randomly sampled 1,000 temperature values weighted by the expression function (e.g. point of maximum expression had highest probability of being sampled). We then calculated the standard deviation of this random draw, and performed a *t*-test comparing the standard deviations of expression between species. Finding a greater standard deviation of expression would be indicative of less sensitivity to changes in temperature. For the **Bimodal** transcripts, we performed the same analysis as with the **Intermediate** transcripts, but with a greater standard deviation being indicative of a more sensitive transcript (convex reaction norm) while a low standard deviation being a transcript with greater inertia of expression in response to temperature changes (nearly flat).Thus, finding one species had overall greater standard deviation of expression for **Bimodal** transcripts would be indicative of greater thermal sensitivity for that species. 

To test for genetic assimilation, we compared constitutive expression levels (expression at 19.5° C) for the subset of transcripts that were responsive to low temperatures (**Low** transcripts) in the southern *A. carolinensis* species, or high temperatures (**High** transcripts) in the northern *A. picea* species using non-parametric two-sided Wilcoxon tests.

## Gene set enrichment analysis

To describe the overall molecular toolkit for thermal tolerance in *Aphaenogaster*, we performed gene set enrichment analysis (GSEA) on the intersection of each expression category in the two species using the `parentChild` algorithm [@grossmann2007] from the package `topGO` [@alexa2006] in R [@r_core_team2013]. Briefly, this approach identifies GO terms that are overrepresented in the significant transcripts relative to all GO terms in the transcriptome, accounting for dependencies among the GO terms. We clustered enriched GO terms from the `topGO` analysis by semantic similarity using the package `GOSemSim` [@yu2010] to reduce redundancy. 

## Data availability

Illumina short-read sequence data is available from the NCBI Short Read Archive BioProject PRJNA260626. The Trinity transcriptome assembly, FastAnnotator annotation file and Sailfish gene expression quantification files are available at [TBD: Harvard Forest Archives]. Reproducible R script for analysis is available from GitHub [@stanton-geddes2014].  


## Acknowledgements

Support made possible by NSF DEB Award #1136644 and the Vermont Genetics Network through Grant Number 8P20GM103449 from the INBRE Program of the National Institute of General Medical Sciences (NIGMS) of the National Institutes of Health (NIH). Its contents are solely the responsibility of the authors and do not necessarily represent the official views of NIGMS or NIH.


# Figure Legends

Figure legends go here.

# Figures

Figures here.

# References



