---
title: "Patterns of stress tolerance versus resistance in the thermal reactionomes from populations of cold and warm-climate ant species"
author: \centering{John Stanton-Geddes \textsuperscript{1}, Andrew Nguyen \textsuperscript{1}, James Vincent \textsuperscript{2}, Mahesh Vangala \textsuperscript{2}, Robert R. Dunn \textsuperscript{3}, Nathan J. Sanders \textsuperscript{4}, Nicholas J. Gotelli \textsuperscript{1}, Sara Helms Cahan \textsuperscript{1} \\ \vspace{5 mm} \textsuperscript{1} Department of Biology, University of Vermont \\ \textsuperscript{2} Vermont Genetics Network, University of Vermont \\ \textsuperscript{3} Department of Biological Sciences, North Carolina State University \\ \textsuperscript{4} Center for Macroecology, Evolution and Climate, University of Copenhagen}
csl: ecology.csl
output:
  pdf_document:
    fig_caption: yes
    includes:
      in_header: header.tex
    keep_tex: yes
    latex_engine: xelatex
    number_sections: yes
    toc_depth: 2
  html_document: null
  word_document: default
includes: header.tex
bibliography: ApTranscriptome.bib
---

\linenumbers
\tableofcontents

\newpage

# Abstract 

**250 words allowed - need to cut 56**

Thermal tolerance is a key trait governing the distributions of species and underlies their future responses to climate change. In this study, we demonstrate how insight on the evolution of thermal tolerance can be gained through characterizing an organism’s thermal “reactionome” – the reaction norm for all genes in an organism's transcriptome measured across a thermal gradient. We characterized the reactionomes of two populations of parapatric ant species of eastern U.S. deciduous forests: the northern cold-climate Aphaenogaster rudis and the southern warm-climate Aphaenogaster carolinensis . After exposing laboratory-acclimated individuals to a range of realistic temperatures (0° - 38.5°C), we found that 10% of genes were thermally responsive. When the two species were compared, the warm-adapted *A. carolinensis* displayed higher basal levels of gene expression, \todo{check this statement} a lower magnitude of upregulation at thermal extremes, upregulated of fewer heat shock proteins, and greater inertia of changes in gene expression in response to thermal extremes. This pattern is consistent with both constitutive thermal defenses and enhanced proteome stability. Moreover, a surprisingly large proportion of genes showed divergent patterns of response to temperature between the two species, suggesting that they have different strategies for coping with temperature stress. *Aphaenogaster carolinensis* shut down more genes than *A. picea* in response to thermal extremes, whereas the same genes tended to be upregulated in *A. picea*. These included genes functionally involved in \todo{check} metabolism, cellular organization and repair. Altogether, these results suggest that *A. picea* actively resists thermal shock by synthesizing heat shock proteins and other protective molecules to maintain function, whereas *A. carolinensis* slows down metabolism and relies upon other mechanisms to tolerate thermal stress. More broadly, this study adds to a growing body of evidence that thermally stressful habitats select for an evolutionary shift in response mechanism away from high-cost damage repair and toward lower-cost prevention, despite enzyme efficiency tradeoffs that such mechanisms entail.  

# Significance Statement

**120-word-maximum statement about the significance of their research paper written at a level understandable to an undergraduate-educated scientist outside their field of specialty**

# Introduction

Temperature plays a prominent role in structuring biological diversity from molecular to macroecological scales [@kingsolver2009]. The ability of individuals to maintain function and survive thermal extremes, thermal tolerance, varies considerably among species and populations [@cowles1939; @deutsch2008; @kingsolver2013; @hoffmann2013]. While organisms differ in fixed traits such as proteome stability [@ghosh2010] and behavior [@sunday2014] that allow them to function in the typical range of temperatures to which they are exposed, tolerance to thermal extremes often involves rapid changes in gene expression [@kultz2005; @somero2011]. These differences within and among species in thermal tolerance are largely genetic [@krebs1997; @kellermann2012] with a highly polygenic basis [@krebs1998; @williams2012; @morgan2006; @takahashi2011]. How this variation of thermal tolerance evolved, and its potential for future evolution [@hoffmann2008], is not well understood. To address this, we need information on the extent to which geographic differentiation and selection has acted upon the diversity of genes, and their plasticity of expression, involved in thermal tolerance.

In recent years, whole organism gene expression approaches (e.g. transcriptomics) using high-throughput RNA sequencing (RNAseq) technology have been widely applied to identify genes underlying thermal tolerance [@meyer2011; @teets2012; @barshis2013; @oneil2014], drought tolerance [@villar2011] and many other traits. These RNAseq studies have used an ANOVA-type experimental or sampleing design with only a few levels of temperature or other environmental stressors tested. However, many ecologically-relevant aspects of the environment such as temperature vary continuously in nature. The reaction norm describes variation in the phenotype of a single genotype in response to a continuous environmental gradient [@gomulkiewicz1992]. Reaction norms differ not only in mean values, but also in their shapes [@hoffmann2013; @murren2014], and differences in the shape of reaction norms are often greater than differences in mean values of either species or populations [@murren2014]. Murren et al [-@murren2014] concluded that many patterns of biologically-relevant plasticity are likely to be missed with ANOVA-style designs that compare only a few discrete levels of an environmental variable. Similarly, gene expression studies that look at just a few environments (e.g. hot vs. cold) are likely to miss key differences among species that are due not just to whether it is hot, but rather how hot it is. 

Moreover, the results provided by RNAseq data do not always directly address the aims of studies. For evolutionary ecology studies, the goal of RNAseq studies often is to identify the genetic basis of environmental adaptation [@meyer2011; @teets2012; @barshis2013; @oneil2014]. However, lists of differentially-expressed genes derived from RNAseq results are prone to false positives [@gonzalez2013], depend on the genetic background of the organism [@sarup2011], may miss genes with small expression changes that have detectable phenotypic effects [@yan2002; @padovan2013], identify genes that are correlated, but not causative, of phenotypes [@fay2004] and may not be directly informative about organismal status [@feder2005]. Genome-wide association (GWA) studies have faced a similar problem: phenotypic effects of a small number of statistically-significant genes are over-estimated while small effects of a large number of other causal genes are missed [@beavis1994; @larsson2013]. Our understanding of the genetic basis of traits, and our ability to predict evolutionary potential, are compromised by these issues. Thus, plant and animal breeders have largely abandoned GWA in favor of genomic prediction [@goddard2007; @yang2010]. 

Here, we present a novel method for RNAseq analysis: the *thermal reactionome*, which is a characterization of the reaction norm for all genes in an organism's transcriptome across a thermal gradient. Rather than focusing on the identity of individual genes with differential expression, we compare overall patterns in the reactionomes of two closely-related temperate-zone species to gain insight into how spatially divergent selection has changed patterns of thermally-responsive gene expression. We apply this approach in a comparative study of two closely related eastern North American ant species, *Aphaenogaster carolinensis* and *A. picea* [@umphrey1996], which have more southern and northern distributions, respectively, and diverged approximately .... years ago (bernice??). Temperature is a potentially strong selective force on ground-nesting *Aphaenogaster* colonies, which must tolerate seasonally freezing winters and hot summers. On shorter time scales, individual workers can experience extreme thermal environments when they leave the thermally buffered ant nest to forage for food [@lubertazzi2012]. 

*A. carolinensis* lives in a climate that is hotter and has a longer growing season than that of *A. picea* (Supplemental Information). Thus, patterns of gene expression underlying tolerance of thermal extremes is likely to differ between these species in a number of ways. First, the number of genes showing plasticity of expression to thermal extremes may differ between the species. Finding more genes are up-regulated to high temperatures in the colder-climate species *A. picea* would indicate that this species relies on activation of the cellular stress response [@kultz2005] and other protective genes to a greater extent than *A. carolinensis*, while the opposite would be true for cold temperatures. Conversely, finding that more genes are down-regulated in response to thermal extremes would indicate that a species relies to a greater extent upon proteome stability [@ghosh2010] and behavioral mechanisms [@sunday2014] to tolerate thermal stress. 

In addition, responsive genes may change in their sensitivity to thermal extremes. Finding that heat-responsive genes are up-regulated at higher temperatures in *A. carolinensis* than *A. picea* would be consistent with local adaptation for greater heat tolerance. That is, more extreme temperatures are required for activation of the heat shock response [@somero2011]. Conversely, finding cold-responsive genes are up-regulated at lower temperatures in *A. picea* than *A. carolinensis* would indicate local adaptation for cold tolerance. Similar patterns may be observed for genes that are down-regulated in response to thermal extremes. Finally, changes in the expression patterns of responsive genes between species can provide insight on the evolution of thermal tolerance. Specifically, if adaptation to thermal extremes works by genetic assimilation [@pigliucci2006], genes up-regulated in response to cold temperatures in *A. carolinensis* may have increased basal expression in the colder-climate species *A. picea*, and vice versa.

Importantly, none of the above hypotheses depend on the functional annotation of the responsive genes, which can be inaccurate in non-model systems [@primmer2013; @skunca2012]. However, insight on the molecular toolkit involved in adaptation to thermal extremes can be gained through circumspect examination of the functional annotation of responsive genes. For example, we expect genes in the cellular stress response [@kultz2005] to be up-regulated in response to thermal extremes, whereas we expect genes involved with cell profliferation and non-essential molecular processes to be down-regulated in response to thermal extremes [@somero2011].


# Results

```{r load, echo=FALSE, results='hide', message=FALSE, warning=FALSE}
options(scipen = 1, digits = 3)
load("../ms.RData")
library(data.table)
library(ggplot2)
library(RColorBrewer)
library(topGO)
```

## Thermally-responsive genes

The Trinity *de novo* transcriptome assembly included 126,172 genes with a total length of 100 million bp. Filtering to remove redundant or chimeric reads resulted in an assembly with 105,536 genes. The total length was 63 million bp with an N~50~ length of 895 bp and a mean gene size of 593 bp. Of the 105,536 filtered genes, 55,432 had hits to the NCBI-nr database. Of these, 38,711 genes mapped to GO terms, 1,659 genes were identified to an enzyme and 18,935 genes mapped to a domain with >50% coverage. We removed 5,675 genes identified as known contaminants, leaving 99,861 clean transcripts. Of these, 47% have best hits to genes from the class Insecta [(Supplementary Krona plot)](http://johnstantongeddes.org/assets/files/ApTranscriptome_FA.html).

```{r mosaic_plot, echo=FALSE, results='asis', fig.cap = "Mosaic plot of the number of thermally-responsive genes by expression type. The width of the columns is proportional to the number of genes for each expression type for *A. carolinensis* while the height of the rows is proportional to the number of genes for each expression type for *A. picea.* The rows are colored by expression type for *A. picea*. The number of genes are labeled in two boxes for scale."}
mosaic.colors <- brewer.pal(5, "Blues")
par(mar = c(5.1, 3.1, 2.1, 7.5))
mosaicplot(tt2, ylab = expression(italic("A. picea")), xlab = "", main = "", color = mosaic.colors, cex.axis = 0.01)
labels <- c("Low", "Intermediate", "High", "Bimodal", "NotResp")
text(c(.1,.5,.75,.85,.99), par("usr")[3] - 0.05, srt = 45, adj = 1, labels = labels, xpd = TRUE, font = 1)
text(.5, par("usr")[4]+.1, "A. carolinensis", xpd = TRUE, font = 3)
text(.15, .7, "2,594")
text(.85, .03, "223", cex = .75)
legend(1.05, .9, fill = mosaic.colors, c("Low", "Intermediate", "High", "Bimodal", "NotResp"), xpd=TRUE)
```

We identified `r prettyNum(sum(sigtable[sigtable$Coefficient %in% c("Temperature", "Temperature:Colony"), "Number.significant"]), big.mark = ",", scientific = FALSE)` genes with expression patterns that are thermally-responsive at 5% FDR. Thus, at a minimum `r round(9100/99861, 2)*100`% of genes are thermally-responsive. However, the closest related ant species with sequenced genomes have only about 17,00 genes [@gadau2012], and assuming the number is similar for *Aphaenogaster*, this suggests that up to 50% of the genome could be thermally-responsive. The reaction norms for each responsive gene can be searched by functional annotation or GO category and viewed interactively at [https://johnsg.shinyapps.io/ApRxN-shinyapp/].

Of the thermally-responsive genes, `r prettyNum(sum(sigtable[sigtable$Coefficient %in% "Temperature:Colony", "Number.significant"]), big.mark = ",", scientific = FALSE)` have an expression pattern that differs between the two species. These differences were not only in magnitude, as the number of transcripts in each expression category differed significantly between the species ($\chi^2$ = `r round(chi1$statistic, 0)`, P < `r max(chi1$p.value, 0.001)`; Fig. 1). In both species, the majority of genes were in the **Low** category (Fig. 1). For *A. carolinensis*, this was followed by **Intermediate**, **High** and then **Bimodal**, whereas for *A. picea* the second largest category was **Bimodal** followed by **High** and finally **Intermediate** (Fig. 1). There were `r colSums(tt2)[which(colnames(tt2) == "NotResp")]` genes that were responsive in *A. carolinensis* but not *A. picea*, and `r rowSums(tt2)[which(rownames(tt2) == "NotResp")]` that were responsive in *A. picea* but not *A. carolinensis*. 

## Species comparisons

To examine how the sensitivity of gene expression to changes in temperature differed between the two species, we compared the temperature of gene activation for **High** and **Low** genes, and the variance in expression for the **Intermediate** and **Bimodal** genes. The **High** genes in *A. carolinensis* were activated on average at `r T_on.high.ttest$estimate[1]`°C compared to `r T_on.high.ttest$estimate[2]`°C for *A. picea* (*t*~df=`r T_on.high.ttest$parameter`~ = `r T_on.high.ttest$statistic`, *P* < `r max(T_on.high.ttest$p.value, 0.001)`; Fig. 2). For genes with increased expression at **Low** temperatures, there was no difference in the mean temperature of gene activation (*t*~df=`r T_on.low.ttest$parameter`~ = `r T_on.low.ttest$statistic`, *P* < `r round(max(T_on.low.ttest$p.value, 0.001), 3)`; Fig. 2). For **Bimodal** expressed genes, we found no difference between species in the variance of the reaction norms (*t*~df=`r round(t.varbim$parameter, 0)`~ = `r round(t.varbim$statistic, 2)`, *P* < `r round(t.varbim$p.value, 2)`). In contrast, **Intermediate** genes from *A. carolinensis* had a broader thermal range of stable gene expression than those from *A. picea* (*t*~df=`r round(t.varint$parameter, 0)`~ = `r round(t.varint$statistic, 2)`, *P* < `r max(t.varint$p.value, 0.001)`, 95% CI difference `r paste(round(t.varint$conf.int[1], 3), "--", round(t.varint$conf.int[2], 3))`). 


```{r T_on_plot, echo=FALSE, warning = FALSE, fig.cap = "Histogram with smooth density estimate of temperature of maximum change in expression for genes that have increased expression at **Low** and **High** temperatures. Bars are Probability density function of peak expression for all thermally-responsive in *A. carolinensis* (blue) and *A. carolinensis* (red) species"}

T_on_plot <- ggplot(Ap.T_on, aes(x = T_on, y = ..density.., colour = Species, group = Species, fill = Species)) +
    facet_grid(. ~ type) +
    geom_histogram(position = "dodge", binwidth = 3) +
    geom_density(alpha = 0.2, adjust = 3) + 
    scale_y_continuous(name="Density") +
    scale_x_continuous(name=expression("Temperature of gene activation"))
print(T_on_plot)
```

We looked for evidence of genetic assimilation by comparing gene expression at intermediate temperatures for genes that were up-regulated at low temperatures in *A. carolinensis* or high temperatures in *A. picea*. Contrary to expectations, for **Low** genes in *A. carolinensis* gene expression at intermediate temperatures was lower in *A. picea* than *A. carolinensis* (Wilcoxon V = `r prettyNum(w2$statistic)`, *P* < `r max(w2$p.value, 0.001)`, 95% CI location shift `r paste(round(w2$conf.int[1], 3), "--", round(w2$conf.int[2], 3))`) . Moreover, for **High** genes in *A. picea* (Wilcoxon V = `r prettyNum(w1$statistic)`, *P* < `r max(w1$p.value, 0.001)`, 95% CI location shift `r paste(round(w1$conf.int[1], 3), "--", round(w1$conf.int[2], 3))`) and **Intermediate** genes in both species (Wilcoxon V = `r prettyNum(w3$statistic)`, *P* < `r max(w3$p.value, 0.001)`, 95% CI location shift `r paste(round(w3$conf.int[1], 3), "--", round(w3$conf.int[2], 3))`), gene expression at intermediate temperatures was always higher in *A. carolinensis*. 

In addition to shifts in position, we quantified the differences between species in the mean and shape of the reaction norms of thermally-responsive genes. Species differed by mean expression (*t*~df=`r t.mean$parameter`~ = `r t.mean$statistic`, *P* < `r max(t.mean$p.value, 0.001)`) which was about `r round(exp(t.mean$estimate))` TPM greater in *A. carolinensis* than *A picea*, and mean slope (*t*~df=`r t.slope$parameter`~ = `r t.slope$statistic`, *P* < `r max(t.slope$p.value, 0.001)`), which was about `r round(exp(t.mean$estimate))` $\Delta$TPM/degree greater in *A. picea*. Curvature (*t*~df=`r t.curvature$parameter`~ = `r t.curvature$statistic`, *P* < `r round(max(t.curvature$p.value, 0.001), 3)`) and wiggle (*t*~df=`r t.wiggle$parameter`~ = `r t.wiggle$statistic`, *P* < `r max(t.wiggle$p.value, 0.001)`) did not differ between species. Of the total differences ($\Delta T$) in reaction norms between species for each gene, about `r round(qlmean[2] * 100, 0)`% (95% CI `r paste(round(qlmean[1] * 100, 0), "--", round(qlmean[3] * 100, 0))`%) and `r round(qlslope[2] * 100, 0)`% (95% CI `r paste(round(qlslope[1] * 100, 0), "--", round(qlslope[3] * 100, 0))`%) were due to shifts in mean and slope, respectively. 


## Gene set enrichment analysis

As the molecular pathways involved in thermal response to cold and hot temperatures differ, we performed gene set enrichment analysis on each expression category separately. First, we looked at the genes shared in each expression category (e.g. the intersection) for each species (Table S1). Out of a total of `r length(score(Ap.BP.resultParentChild))` GO terms scored, we found `r nrow(Ap.gsea.intersect[which(Ap.gsea.intersect$Type == "High"), ])` enriched (*P* < 0.01) terms for **High** genes. Clustering of terms by semantic similarity yielded 3 clusters, with 5 of the 7 terms being involved in "regulation of biosynthetic/metabolic process". For **Low** genes, `r nrow(Ap.gsea.intersect[which(Ap.gsea.intersect$Type == "Low"), ])` GO terms were enriched. These terms grouped into 15 clusters, including "protein modification", "protein metabolic process", "lipid metabolic process", "regulation of signaling", "protein transport/localization", "detection of stimulus", and "chondroitin sulfate biosynthesis". For **Bimodal** genes, `r nrow(Ap.gsea.intersect[which(Ap.gsea.intersect$Type == "Bimodal"), ])` GO terms were enriched. These terms grouped into 5 clusters, including "biological regulation", "regulation of cellular biosynthesis", "gene expression", "signaling", and "protein modification process". For **Intermediate** genes, `r nrow(Ap.gsea.intersect[which(Ap.gsea.intersect$Type == "Intermediate"), ])` GO terms were enriched. These terms grouped into 3 clusters, including "lipid biosynthesis/metabolism", "glycoprotein biosynthesis/metabolism", and "DNA organization". Overall, GSEA indicates that molecular responses to thermal extremes involve large-scale cellular and organizational changes, though the molecular responses to high temperatures are more limited than those to low temperatures. 

We further performed GSEA for the genes that shifted from being shut-down at extreme temperatures, **Intermediate**, in *A. carolinensis* to having increased expression in **Low** temperatures in *A. picea*. This group included `r nrow(Ar.int.A22.low.gsea)` GO terms that grouped into 12 clusters, including "protein metabolic process", "glycosylation" "transport", "lipid metabolic process", "GTP metabolism", "positive regulation of cellular process" and "organelle organization". For **Intermediate** genes in *A. carolinensis* that shifted **Bimodal** in *A. picea*, there were `r nrow(Ar.int.A22.bim.gsea)` enriched GO terms including "phospholipid catabolic process", "L-amino acid transport", "macromolecule metabolic process", "gluatmate receptor signaling pathway" and "histone modification". These results indicate that in response to thermal extremes, the warmer-climate species, *A. carolinensis*, shuts down metabolism to a greater extent than the colder-climate species, *A. picea*, which increases metabolism. 

**Table** Table of enriched GO terms for each expression category in each species.

## Candidate genes

We examined the reaction norms for genes in the GO category "response to stress" which we *a priori* expected to have significant thermal responses (Fig. 3). Though we found `r resp.stress.responsive.count` genes annotated with this GO term among the thermally-responsive trancripts, this number was not enriched compared to the total number in the transcriptome ($\chi^2$ = `r GO.stress.Xsq$statistic`, P < `r round(GO.stress.Xsq$p.value, 2)`).


# Discussion

Thermal tolerance is a key trait for determining the survival of individual organisms, the persistence of populations and the geographic limits of species. In this study, we applied a novel regression design to characterize the thermal reactionome of parapatric ant species, *Aphaenogaster picea* and *Aphaenogaster carolinensis*, found in colder and warmer environments, respectively, that are known to differ in their thermal tolerances [@warren2013]. We found that at least 10%, and potentially 50%, of the genes in these species were thermally-responsive. 

The patterns of gene expression between species differed substantially, with more down-regulation in *A. carolinensis* (**Intermediate** genes) and more up-regulation of gene expression in *A. picea* in response to thermal extremes (**High** and **Low** genes; Fig. 1, 3). The results of the gene set enrichment analysis give an indication of what might drive this pattern. The genes that are up-regulated at cold temperatures (**Low**) in *A. picea* but shut-down **Intermediate** by *A. carolinensis* include metabolism, transport and regulation GO clusters. Thus, *A. carolinensis* shuts-down metabolism to a lesser extent than *A. picea* and relies on other mechanisms such as behavior or proteome stability to tolerate thermal stress. In contrast, *A. picea* increases metabolism to a greater extent than *A. carolinensis*, potentially to mount a greater molecular response to thermal stress. Consistent with this hypothesis, we find that of the 7 thermally-responsive heat shock proteins, all 7 increase expression in response to thermal extremes in *A. picea*, but only 4 do in *A. carolinensis*. This result is consistent with work in corals [@barshis2013] that shows that organisms in more thermally-stressful environments are more likely to invest in a tolerance strategy, while organisms from less stressful environments rely on plastically induced resistance. 


```{r max_exp_PDF, echo = FALSE, warning = FALSE, message = FALSE, fig.cap = "Probability density function of the temperature of maximum expression for all thermally-responsive genes in each species."}
# reshape data
Ap.df$Species <- ifelse(Ap.df$colony == "ApVT", "A. picea", "A. carolinensis")

maxexplot <- ggplot(Ap.df, aes(x=max.val, fill=Species)) + 
  geom_density(alpha=0.2, position="identity") + 
  labs(x = "Temperature of maximum expression", y = "Density")

print(maxexplot)
```

Given the differences in the number of thermally-responsive genes between the two species (Fig. 1), a striking pattern is that 3-4x more genes are upregulated at cold than high temperatures in both species (Fig. 3). A potential explanation for this pattern is that selection has been stronger on upper than lower thermal limits, thereby reducing genetic variation at high temperatures [@kingsolver2013; @kelly2013]. This explanation is consistent with the observation in *Aphaenogaster* [@warren2013] and other ectotherms [@addo-bediako2000; @hoffmann2013] that critical maximum temperatures are more constrained than critical minimum temperatures. Moreover, in an experiment with 94 species of *Drosophila*, Kellermann et al [-@kellermann2012] showed that limited genetic variation among species (e.g. phylogenetic inertia) rather than similar selection pressures underlie greater similarity of upper than lower thermal limits. Thus, when coupled with phenotypic data, our results demonstrate that transcriptomic studies can provide insight on evolutionary potential. For *Aphaenogaster*, our results suggest that adaptability to increased temperatures is limited, and these species risk replacement by more heat-tolerant competitors [@rodriguez-cabal2012]. 

For species or populations that routinely, but unpredictably, experience stressful environments, genetic assimilation is the process by which physiologically-relevant genes change from being environmentally-induced to basally-expressed [@pigliucci2006]. However, in contrast to these expectations, we found lower basal gene expression in *A. picea* for genes that were up-regulated in response to cold temperatures (**Low** genes) in *A. carolinensis*. We also observed greater basal expression in *A. carolinensis* for **High** genes in *A. picea* as well as for genes that were **Intermediate** in both spcies. A potential explanation for the consistently higher basal gene expression in the warmer-environment species, *A. carolinensis*, is that this species has a higher metabolic rate and thus requires greater gene expression.  

Of the `r prettyNum(sum(sigtable[sigtable$Coefficient %in% c("Temperature", "Temperature:Colony"), "Number.significant"]), big.mark = ",", scientific = FALSE)` genes that were thermally-responsive, `r round(with(sigtable, sigtable[Coefficient=="Temperature:Colony","Number.significant"]) / with(sigtable, sum(sigtable[Coefficient %in% c("Temperature", "Temperature:Colony"),"Number.significant"])) * 100, 0)`% differed in their responsiveness by colony. The differences between species were due to changes in the mean ($\Delta M$) and slope ($\Delta S$) of the reaction norms for each thermally-responsive gene. This result contrasts with the findings of Murren et al. [-@murren2014], who found that most of the differences in phenotypic reaction norms were due to shape. As genes are not directly analogous to phenotypes, this contradiction is not surprising. Instead, it indicates that small shifts in the mean of many genes underlying a trait can result in non-linear phenotypic changes.  

To date, transcriptomic studies have performed differential expression analysis between a few different conditions or environments (but see [@hodgins-davis2012]). By using a regression-approach, we are able to quantitatively describe changes in gene expression, rather than simply report the numbers of differentially-expressed genes. Thus, we were able to compare the temperature at which genes were activated near the upper and lower thermal limits between species (Fig. 2). For genes activated at cold temperatures, there was little difference among species with *A. picea* genes turning on at 13°C compared to 12.8°C for *A. carolinensis*. For high temperature genes, *A. carolinensis* genes were turned on at 33.8°C compared to 32.9°C for *A. picea*. 

As with all *de novo* transcriptomes, our work has some important caveats. First, given that all sequenced ant genomes to date have about 18,000 genes [@gadau2012], our transcriptome is highly fragmented. Though we took steps to remove contaminants and redundant transcripts, some likely remain, in addition to partially-assembled transcripts. A genome assembly, in progress, would considerably resolve this problem. Second, the quality of the annotation for a non-model system such as *Aphaenogaster* is poor, as we have to rely upon resources from distantly related *Drosophila* and *Apis*. Finally, the mapping of changes in gene expression to organismal fitness is far from direct [@feder2005]. Large differences in patterns of gene expression may only have small effects on organismal fitness. Given these caveats, transcriptomic studies promise to offer many new insights on organisms' response to environmental change. Candidate gene studies, while informative, fail to capture the complexity of gene expression changes [@jensen2008; @kenkel2013]. In addition, for long-lived species that are intractable to study in the lab, transcriptomics can provide key insights on plasticity and the potential for species to tolerate and adapt to environmental change [@oneil2014; @lemay2013]. 

## Conclusions 

In this work, we bring reaction norms to the genomic era by characterizing the thermal reactionomes of the temperate ant species, *Aphaenogaster picea* and *A. carolinensis*. We found that at least 10% of the transcriptome is thermally-responsive. For both species, more genes responded to cold than hot temperatures. In light of physiological data showing more variation in critical minimum than maximum temperatures, this suggests that these species and other ectotherms have less adaptive potential to hot than cold temperatures. Further, our results indicate that the two species have different responses to thermal extremes. *A. picea* responds by increasing metabolism and activating heat shock and other protective molecules, while *A. carolinensis* shuts down metabolism and must rely on other mechanisms for thermal tolernace. These results provide key insights on the genetic basis of thermal tolerance, and provide a tremendous resource for the future study of ecological adaptation in ant species.



# Material and Methods

Ants are ecologically-important species involved in multiple symbioses¸ and are key contributors to seed dispersal [@lengyel2009; @rodriguez-cabal2012], litter processing and decomposition [@lyford1963; @folgarait1998; @terborgh2001; @holec2006]. Ants of the genus *Aphaenogaster* are some of the most abundant in eastern North America [@lubertazzi2012]. ...previous results from chambers...???

## Samples

We used colonies of the ant species *Aphaenogaster picea*, collected from Molly Bog, Vermont (University of Vermont Natural Areas; 44.508°N, -72.702°W) and *Aphaenogaster carolinensis*, collected from Durham, North Carolina ( 36.037° N -78.874° W). The phylogeny of *Aphaeogaster* is currently unresolved [@umphrey1996], but putative species identification was possible based on morphology (Bernice Bacon DeMarco, pers. communication). Colonies of these species were maintained in the lab for 6 months prior to experimentation. 

From each species, we haphazardly collected 12 ants at the same time on 12 days. Each day, the 12 ants were placed in glass tubes into a water bath at one of 12 randomly-assigned temperatures, every 3.5°C between 0° and 38.5°C, for one hour. The minimum and maximum temperatures were selected based on previous work showing that these temperatures are close to the critical minimum and maximum temperatures for *Aphaenogaster* [@warren2013]. At the end of the hour, the ants were flash frozen in liquid nitrogen and stored at -80°C. We extracted mRNA by homogenizing three pooled ants in 500 uL of RNAzol buffer with zirconium silicate beads in a Bullet Blender (Next Advance; Averill Park, NY), followed by RNAzol extraction (Molecular Research Center Inc; Cincinnati, OH) and then an RNeasy micro extraction (Qiagen Inc; Valencia, CA) following the manufacturer's instructions. 

## Sequencing, assembly and annotation

For each species, the 12 samples were barcoded and sequenced in a single lane of 2 x 100bp paired-end reads on an Illumina HiSeq 1500 yielding 200 and 160 million reads for the *A. picea* and *A. carolinensis* samples respectively. Reads were filtered to remove Illumina adapter sequences and low quality bases using the program Trimmomatic [@lohse2012]. 

We assembled the sequenced reads into the full set of mRNA transcripts, the transcriptome, for the combined data set from both species using the Trinity *de novo* transcriptome assembly program [@grabherr2011]. *De novo* transcriptome assembly is prone to falsely identifying alternative transcripts and identifying inaccurate transcripts that are chimeric (e.g. regions of two separate transcripts that assemble into a false, or chimeric, third transcript) [@yang2013]. We removed potentially false transcripts by first running the program `CAP3` [@huang1999] to cluster sequences with greater than 90% similarity and merge transcripts with overlaps longer than 100 bp and 98% similar in length. Second, we ran the program `uclust` which clusters sequences completely contained within longer sequences at greater than 90% similarity (see Supplemental Information for exact parameters). We used liberal values (90% similarity) to merge orthologous transcripts in the two species that may not have assembled together in the initial *de novo* transcriptome assembly. To identify contaminant sequences, we screened our full transcriptome using the program `DeconSeq` [@schmieder2011] with the provided bacteria, virus, archae and human databases of contaminants. 

To determine the putative function of the transcripts, we performed functional annotation of the transcriptome assembly using the web-based tool `FastAnnotator` [@chen2012] which annotates and classifies genes by Gene Ontology (GO) term assignment, enzyme identification and domain identification. 

We performed subsequent analyses at the transcript level due to limitations of working with a system lacking a reference genome, but refer to these transcripts as genes for simplicity.

## Thermally-responsive genes

We quantified expression of each gene using the program `Sailfish` [@patro2014] and use the bias-corrected transcripts per million (TPM) [@wagner2012] as our measure of gene expression. We included the contaminant genes identified by DeconSeq at the quantification stage to avoid incorrectly assigning reads to other genes, but removed these from further analyses. As preliminary examination of the data (see supplemental...) indicated that 7°C samples may have been mis-labeled, we omitted these data from the analysis. The expression values were highly correlated between species at each temperature treatment (r^2^ > 0.98) indicating that assembling the transcriptome with data from both species was justified (Supplemental).

To identify genes that have significant changes in expression across the thermal gradient, we fit the regression model

$$ log(TPM + 1) = \beta_0 + \beta_1(species) + \beta_2(temp) + \beta_3(temp^2) + \beta_4(species * temp) + \beta_5(species * temp^2) + \epsilon $$
    
to each gene. We used $log(TPM + 1)$ as the response to control for skew in the expression data. For a continuous predictor such as temperature, this regression approach is preferred to an ANOVA approach as it can reveal non-linear responses such as hump-shaped or threshold effects even [@cottingham2005]. This method is robust to over-dispersion as we expect errors in the read count distribution [@anders2010] to be independent with respect to temperature. To correct for multiple testing, we calculated adjusted *P* values using the False Discovery Rate (FDR) approach of Benjamini and Hochberg [-@benjamini1995], retaining genes under the 5% FDR threshold.

We identified thermally-responsive genes as the subset under our FDR threshold that had significant $\beta_2(temp)$, $\beta_3(temp^2)$, $\beta_4(species * temp)$ or $\beta_5(species * temp^2)$ terms after step-wise model selection by AIC. All analyses were performed in R [@r_core_team2013] and are fully reproducible (Supplemental file). 

For each thermally-responsive gene, we predicted expression levels across the tested thermal range using the final linear model separately for each species. We used the predicted gene expression values to partition the thermally-responsive genes into five expression categories for each species; genes that had greatest expression at high (> 31°) temperatures (**High**), low (< 10°C) temperatures (**Low**), intermediate (10 - 30°C) temperatures (**Intermediate**), both high and low temperatures (**Bimodal**) or were not thermally-responsive in that species (*NotResp*). For the **Bimodal** group, we required that expression at both low and high temperatures was at least one standard deviation greater than the expression at the mean temperature of 19.5°C.

## Species-level comparisons

We performed comparative analyses between the two species focusing on the genes that had a significant species x temperature interaction to determine if the two species differed in the tempertaure of gene activation and sensitivity of the thermally-responsive genes. For each gene, we predicted expression levels across temperatures for each species using the full linear model, and then grouped genes into the four responsive categories, **High**, **Low**, **Intermediate**, and **Bimodal**, as well as a fifth category **Not Responsive**. To test whether the temperature at which thermally-responsive genes are activated differs between species, we identified the temperature at which there was the greatest change in expression for each gene in each species. We grouped the **High** genes along with the high temperature end of the **Bimodal** genes, and did the same for **Low** genes. We then performed a *t*-test to determine if the mean temperature of gene activation differed between the two species for each group. 

To determine if the sensitivity of genes to thermal extremes differed between the species, we used the subset of **Intermediate** and **Bimodal** genes in each species. For each **Intermediate** gene in each species, we randomly sampled 1,000 temperature values weighted by the expression function (e.g. point of maximum expression had highest probability of being sampled). We then calculated the standard deviation of this random draw, and performed a *t*-test comparing the standard deviations of expression between species. Finding a greater standard deviation of expression would be indicative of less sensitivity to changes in temperature. For the **Bimodal** genes, we performed the same analysis as with the **Intermediate** genes, but with a greater standard deviation being indicative of a more sensitive gene (convex reaction norm) while a low standard deviation being a gene with greater inertia of expression in response to temperature changes (nearly flat).Thus, finding one species had overall greater standard deviation of expression for **Bimodal** genes would be indicative of greater thermal sensitivity for that species. 

To test for genetic assimilation, we compared basal expression levels (expression at 19.5° C) for the subset of genes that were responsive to cold temperatures (**Low** genes) in the southern *A. carolinensis* species, or hot temperatures (**High** genes) in the northern *A. picea* species using non-parametric two-sided Wilcoxon tests.

Finally, we explored the extent to which differences in the thermal reactionome between the species were due to changes in the mean or shape of the reaction norm for each individual gene. Following the example of Murren et al [-@murren2014], we tested for overall differences in the (a) mean expression value across temperatures (*M*), (b) slope of expression (*S*), (c) curvature of expression (*C*) and (d) wiggle of expression (*W*), which consists of all higher-order differences in shape not capture by the first three measures (formulas given in Appendix). As genes differed by five orders of magnitude in expression levels, we calculated the mean-standardized difference between the species $\Delta M = (Mean_{A. picea} - Mean_{Ar}) / \bar M$, where $\bar M = (Mean_{A. picea} + Mean_{Ar}) / 2$ yielding $\Delta M$, $\Delta S$, $\Delta C$, and $\Delta W$ for the mean, slope, curvature and wiggle of each gene, respectively. Using the mean-standardized values, we performed one-sample two-sided *t*-tests to determine if the overall mean, slope or curvature were greater in *A. picea* than *A. carolinensis*. Further, to quantify the relative contribution of changes in the mean, slope, curvature or wiggle on the overall differences in reaction norms, we defined the *Total* difference in the reaction norm for each gene as $\Delta T = \Delta M + \Delta S + \Delta C + \Delta W$. We then partitioned the contribution of each measure by dividing it by $\Delta T$. 

## Gene set enrichment analysis

To describe the overall molecular toolkit for thermal tolerance in *Aphaenogaster*, we performed gene set enrichment analysis (GSEA) on the intersection of each expression category in the two species using the package `topGO` [@alexa2006; @alexa2010] with the `parentChild` algorithm [@grossmann2007] implemented in R [@r_core_team2013]. Briefly, this approach identifies GO terms that are overrepresented in the significant genes relative to all GO terms in the transcriptome, accounting for dependencies among the GO terms. We clustered enriched GO terms from the `topGO` analysis by semantic similarity using the package `GOSemSim` [@yu2010] to reduce redundancy. 

## Data availability

The raw Illumina reads are available at [...], the script for analysis and version history is available at (http://github.com/johnstantongeddes/ApTranscriptome).  



# Figure Legends




# References

