---
title: Thermal reactionomes reveal adaptive responses to thermal
  extremes in warm and cold-climate ant species
author: ' '
csl: ecology.csl
output:
  pdf_document:
    fig_caption: yes
    includes:
      in_header: header.tex
    keep_tex: yes
    latex_engine: xelatex
    number_sections: no
    toc_depth: 2
  html_document: null
  word_document: default
includes: header.tex
bibliography: ApTranscriptome.bib
---

\linenumbers

\newpage

**Authors** 

John Stanton-Geddes \textsuperscript{1}, Andrew Nguyen \textsuperscript{1}, James Vincent \textsuperscript{2},
Mahesh Vangala \textsuperscript{2}, Robert R. Dunn \textsuperscript{3}, Aaron M. Ellison \textsuperscript{4}, Nathan J. Sanders \textsuperscript{5}, Nicholas J. Gotelli \textsuperscript{1}, Sara Helms Cahan \textsuperscript{1}
           
\vspace{10 mm}           
           
**Affilitiations**

\textsuperscript{1} Department of Biology, University of Vermont, Burlington, Vermont 05405 \newline
\textsuperscript{2} Vermont Genetics Network, University of Vermont, Burlington, Vermont 05405 \newline
\textsuperscript{3} Department of Biological Sciences, North Carolina State University, Raleigh, North Carolina 27695 \newline
\textsuperscript{4} Harvard Forest, Harvard University, Petersham, Massachusetts 01336 \newline
\textsuperscript{5} Center for Macroecology, Evolution and Climate, University of Copenhagen

\newpage

# Abstract 

Thermal tolerance is a key trait governing the distributions of species and is likely useful in predicting their future responses to climate change. To understand how plasticity of gene expression underlies thermal tolerance, we describe the *thermal reactionome*, the reaction norm for all genes in an organism's transcriptome measured across an experimental thermal gradient. We characterize the reactionomes of two common ant species in the eastern U.S: the northern cold-climate *Aphaenogaster picea* and the southern warm-climate *Aphaenogaster carolinensis*. We found that at least 10% of genes were thermally responsive. The warm-adapted *A. carolinensis* displayed higher mean levels of expression for responsive genes, exhibited greater inertia of gene expression change and upregulated fewer heat shock proteins than did *A. picea* in response to thermal extremes. This pattern is consistent with hypotheses of constitutive thermal defense and enhanced proteome stability. *Aphaenogaster carolinensis* also down-regulated expression of more genes in response to thermal extremes than did *A. picea*. A subset of these same genes were upregulated in *A. picea*, including genes mapped to ontology terms such as metabolism, cellular organization, glycoprotein synthesis and DNA repair. These differences indicate that the two closely related species employ different strategies for coping with temperature stress. The northern *A. picea* actively resists thermal shock by synthesizing heat shock proteins and other protective molecules to maintain function, whereas the southern *A. carolinensis* slows down metabolism and relies upon other mechanisms to tolerate thermal stress. These results are in line with expectations that thermally stressful habitats select for an evolutionary shift in response mechanism away from high-cost damage repair and toward lower-cost prevention.  



# Significance Statement

Coping with progressive changes in climate requires both short-term physiological acclimation and long-term adaptive changes. To better understand the relationship between these two timescales, we compared gene expression as a function of temperature in two closely-related ant species: the cool-climate *Aphaenogaster picea* and the warm-climate *A. carolinensis*. Surprisingly, we found divergent transcriptomic responses in these species. *Aphaenogaster picea* responded to high and low temperatures through upregulation of a larger proportion of its transcriptome with a robust heat shock response. In contrast, *A. carolinensis* required more extreme temperatures to trigger a protective response and downregulated genes related to metabolism. These results suggest that adaptation to chronic high temperatures involves qualitatively different mechanisms than those used to cope with rarely encountered stressors, which may limit adaptive potential of northern species in the face of rapidly shifting climatices.



# Introduction

Temperature regulates biological activity and shapes diversity from molecular to macroecological scales [@brown2004; @kingsolver2009]. Many species, especially small insects, live at temperatures close to their thermal limits and are at risk from current increases in temperature [@deutsch2008; @kingsolver2013; @sunday2014]. Thermal tolerance, the ability of individuals to maintain function and survive thermal extremes, depends on a complex interplay between the structural integrity of cellular components and activiation of physiological response mechanisms to prevent and/or repair damage (...refs...). Thermal defense strategies are shaped by the environmental regime organisms experience (...ref...), and thermal limits vary considerably among species and populations [@cowles1939; @deutsch2008; @kingsolver2013; @hoffmann2013]. These differences in thermal tolerance are largely genetic [@krebs1997; @kellermann2012] with a highly polygenic basis [@krebs1998; @williams2012; @morgan2006; @takahashi2011]. Outside of candidate genes [@krebs1998], little is known about the molecular basis of this variation in thermal tolerance or the link between short-term physiological acclimation and longer-term adaptation to novel temperature regimes. This information is critical for understanding the adaptive potential of species to future climates [@hoffman2008].

To address this gap of knowlege, we need information on the extent to which selection has acted upon the diversity genes and their plasticity of expression involved in thermal tolerance [@hoffmann2008; @somero2011]. In recent years, whole organism gene expression approaches (e.g. transcriptomics) using high-throughput RNA sequencing (RNAseq) technology have been widely applied to identify genes involved in thermal tolerance [@meyer2011; @teets2012; @barshis2013; @oneil2014] and other traits. Such studies typically use an ANOVA-type experimental or sampling design, with only a few environmental levels, and often find only a few dozen to hundred genes with differential expression in different thermal regimes. However, temperature and other environmental factors vary continuously in nature. As a result, such comparisons (e.g. hot vs. cold) are likely to miss key differences that are due not just to whether it is hot, but rather how hot it is. Continuous variation is better characterized with a reaction norm approach, which describes variation in the phenotype of a single genotype across an environmental gradient [@gomulkiewicz1992]. Reaction norms differ not only in mean values, but also in their shapes [@hoffmann2013; @murren2014], and differences in the shape of reaction norms are often greater than differences in mean values at both the species and population levels [@murren2014]. 

Here, we extend the reaction norm approach to RNAseq analysis with the *reactionome*, which we define as a characterization of the reaction norm for all genes in an organism's transcriptome across an environmental gradient. We use this method to identify genes that are thermally-responsive in two closely related eastern North American ant species, *Aphaenogaster carolinensis* and *A. picea* [@umphrey1996]. *Aphaenogaster* are some of the most common ant in eastern North America [@king2013] where they are keystone seed dispersers [@ness2009; @zelikova2011; @rodriguez-cabal2012]. *Aphaenogaster carolinensis* experiences significantly warmer and less seasonally variable temperatures than *A. picea* (...data...), with correspondingly higher minimal and maximal critical thermal limits (...CTmin and CTmax from HF/DF...). Experimental warming experiments indicate that *Aphaenogaster* abundance and competitive ability may be significantly impacted by temperature increases (...refs...).

To characterize the thermal reactionome, we collected whole organism RNAseq data at 12 temperatures (every 3.5°C between 0 and 38.5°C) using Illumina short read sequencing technology, generated the *de novo* transcriptome using Trinity [@grabherr2011], quantified gene expression using Sailfish [@patro2014], and characterized the reaction norm for each gene using a custom analysis (Supplemental Material). We used these data to test two specific predictions. First, we predicted that species from more thermally-stressful environments rely to a greater extent upon constitutive tolerance than plastically-induced resistance mechanisms relative to species from cooler environments. [@barshis2013]. This pattern could be manifested through the number of thermally-responsive genes, or their sensitivity to changes in temperature. Finding that fewer genes are thermally responsive in the warmer-climate species *A. carolinensis* would support the former, while finding that more extreme temperatures are required to induce changes in gene expression in *A. carolinensis* would support the latter. Second, we predicted that long-term adaptive changes to temperatures would involve the same genes that are involved in short-term physiological acclimation (e.g. genetic assimilation [@pigliucci2006]). Finding that the genes up-regulated in response to cold temperatures in *A. carolinensis* have increased basal expression in the colder-climate species *A. picea*, and vice versa, would be consistent with this prediction. 


# Results

```{r load, echo=FALSE, results='hide', message=FALSE, warning=FALSE}
#options(scipen = 1, digits = 3)
load("ms_2014-09-26.RData")
library(data.table)
library(ggplot2)
library(reshape2)
library(pander)
library(RColorBrewer)
library(topGO)

options(digits = 2)
panderOptions('round', 2)
panderOptions('keep.trailing.zeros', TRUE)
```

## Transcripts differ in thermal responsiveness between species

The combined *Aphaenogaster* transcriptome assembly contained 98,186 transcripts. About half of these (51,246) transcripts had a signficant BLAST hit, of which 50% (25,797) had a top hit to Insecta and 37% (18,854) had a top hit to Formicidae. Of these, we identified `r prettyNum(sum(sigtable[sigtable$Coefficient %in% c("Temperature", "Temperature:Colony"), "Number.significant"]), big.mark = ",", scientific = FALSE)` transcripts, or 8% of the total, with expression patterns that were thermally-responsive at a 5% False Discovery Rate [@benjamini1995]. Of these, `r prettyNum(length(quadratic.lms), big.mark = ",")` had a non-linear relationship with temperature that would likely have been missed with a standard differential expression approach. The result is the same if we focus only on those transcript with a BLAST hit (4,618 transcripts out of 51,246). However, as with all *de novo* transcriptome assemblies, this assembly is fragmented due to partial contigs and alternative transcripts [@vijay2013] suggesting that this estimate is a lower-bound for the true proportion of genes that are thermally-responsive. 


```{r type_table, echo=FALSE, results='asis', fig.cap = "Table of the number of thermally-responsive transcripts by expression type for *A. carolinensis* and *A. picea*. **Low** are transcripts with increased expression at low temperatuers (< 10°C), **Intermediate** are transcripts with maximum expression between 10 - 30°C, **High** are transcripts with increased expression at high temperatures (> 31°C), **Bimodal** are transcripts with increased expression at both low and high temperatures, while **NotResp** are transcripts that are thermally-responsive in one species but not the other."}

rownames(Ap.type.table) <- list("A. picea", "A. carolinensis")

pandoc.table(Ap.type.table[,c("Low", "Intermediate", "High", "Bimodal", "NotResp")], style="rmarkdown",
             big.mark = ",", split.tables = Inf, 
             caption = "Table of the number of thermally-responsive transcripts by expression type for *A. carolinensis* and *A. picea*. **Low** are transcripts with increased expression at low temperatuers (< 10°C), **Intermediate** are transcripts with maximum expression between 10 - 30°C, **High** are transcripts with increased expression at high temperatures (> 31°C), **Bimodal** are transcripts with increased expression at both low and high temperatures, while **NotResp** are transcripts that are thermally-responsive in one species but not the other.")
```

Transcripts varied both qualitatively and quantitatively in their pattern of response to the temperature gradient. We grouped the thermally-responsive transcripts for each species into five expression categories based on their reaction norms: transcripts that had greatest expression at high (> 31°) temperatures (**High**), low (< 10°C) temperatures (**Low**), intermediate (10 - 30°C) temperatures (**Intermediate**), both high and low temperatures (**Bimodal**) or were not thermally-responsive in that species (**Not Responsive**). For the **Bimodal** group, we required that expression at both low and high temperatures was at least one standard deviation greater than the expression at midpoint (19.5°C). 

We found that the number of transcripts that responded to low temperatures is substantially greater than of that to high temperatures (Table 1). In both species, the majority of transcripts were in the **Low** category (Table 1). For *A. carolinensis*, this was followed by **Intermediate**, **High** and then **Bimodal**, whereas for *A. picea* the second largest category was **Bimodal** followed by **High** and finally **Intermediate**. 

Gene set enrichment analysis (GSEA) of the responsive transcripts indicated that the molecular responses to thermal extremes involve large-scale cellular and organizational changes. Out of a total of `r length(score(Ap.BP.resultParentChild))` GO terms scored, we found `r nrow(Ap.gsea.intersect[which(Ap.gsea.intersect$Type == "Low"), ])` terms enriched (*P* < 0.01) for **Low** transcripts. These terms grouped into 15 clusters, including "protein modification", "protein metabolic process", "lipid metabolic process", "regulation of signaling", "protein transport/localization", "detection of stimulus", and "chondroitin sulfate biosynthesis". For *High* transcripts, we found `r nrow(Ap.gsea.intersect[which(Ap.gsea.intersect$Type == "High"), ])` enriched terms, with 8 of the 9 terms being involved in regulation of biosynthetic or metabolic process".  For **Bimodal** transcripts, `r nrow(Ap.gsea.intersect[which(Ap.gsea.intersect$Type == "Bimodal"), ])` GO terms were enriched and grouped into 5 clusters, including "biological regulation", "regulation of cellular biosynthesis", "transcript expression", "signaling", and "protein modification process". For **Intermediate** transcripts, `r nrow(Ap.gsea.intersect[which(Ap.gsea.intersect$Type == "Intermediate"), ])` GO terms were enriched and grouped into 3 clusters including "lipid biosynthesis/metabolism", "glycoprotein biosynthesis/metabolism", and "DNA organization". 

# Shifts in the expression patterns of thermally-responsive transcripts between species

Of the thermally-responsive transcripts, `r prettyNum(sum(sigtable[sigtable$Coefficient %in% "Temperature:Colony", "Number.significant"]), big.mark = ",", scientific = FALSE)` (`r prettyNum(sigtable[sigtable$Coefficient == "Temperature:Colony", "Number.significant"] / sigtable[sigtable$Coefficient == "Total", "Number.significant"]*100, digits=2)`%) had an expression pattern that differed between the two species. This was not due to a species difference in the overall number of responsive genes (...stat...). Instead, most of these were transcripts that changed from one expression category to another (i.e. the number of transcripts in each expression category differed significantly between species; $\chi^2$ = `r round(chi1$statistic, 0)`, P < `r max(chi1$p.value, 0.001)`; Table 1). These shifts were not randomly distributed (...Stuart-Maxwell test of marginal homotranscriptity ...)

**Intermediate** transcripts in *A. carolinensis* were significantly more likely to have **Low** or **Bimodal** expression in *A. picea* than expected by chance (...stat...), while there was a significant deficit of the reverse shift of *A. picea* **Intermediate** transcripts to any upregulated category in *A. carolinensis* (Fig. 1). Similarly, both **High** and **Bimodal** transcripts in *A. picea* were more likely to be **Not Responsive** in *A. carolinensis* than expected, while the reverse was not significant (Fig. 1). 
 
The molecular processes in the set of transcripts that switched from down-regulation in *A. carolinensis* to up-regulation in *A. picea* included `r nrow(Ar.int.A22.low.gsea) + nrow(Ar.int.A22.bim.gsea)` GO terms, of which ... were related to metabolism. In addition, there were GO terms related to repair (e.g. "histone modification"), transcript regulation (e.g. "positive regulation of cellular process") and cellular organization (e.g. "transport", "organelle organization"). Altogether, these results suggest that in response to thermal extremes, the warmer-climate species, *A. carolinensis*, primarily shuts down metabolism while the colder-climate species, *A. picea*, increases metabolism and initiates damage repair responses. 

...GO terms **High** and **Bimodal** transcripts in *A. picea* that were **Not Responsive** in *A. carolinensis* ...


```{r expression_change, echo=FALSE, warning = FALSE, fig.cap = "Matched expression type data for *A. carolinensis* (rows) and *A. picea* (columns). The color of each cell represent the deviation of the observed from the expected number of transcripts based on hypothetical equivalence of the marginal frequencies. Blue cells are those with significant excesses, while red cells are those with significant absences."}
#tt2

# marginal sums and total
rs <- rowSums(tt2)
cs <- colSums(tt2)
n <- sum(tt2)

# grand mean for each class
gm <- (rs + cs) / sum(tt2 * 2)

# calculate observed values for each cell using overall mean for each expression type
Ec <- outer(gm, gm, "*") *sum(tt2)

# get deviations of observed from expected
Ec.dev <- tt2 - Ec

# overall table P-value
Ec.chistat <- sum((tt2-Ec)^2 / Ec)
# P-value
pchisq(Ec.chistat, df=(nrow(tt2)-1) * (ncol(tt2)-1), lower.tail = FALSE)

# overall minimum chi-squared value `X2.df` for a significantly non-homogenous distribution of counts
df = (ncol(tt2) - 1) * (nrow(tt2) - 1)
X2.df <- qchisq(p = 0.05, df = df, lower.tail = FALSE)

# posthoc result for each cell
Ec.cells <- sign(tt2 - Ec) * (tt2 - Ec)^2 / Ec
Ec.cells.sign.table <- apply(Ec.cells, c(1,2), function(x) ifelse(abs(x) <= X2.df, "0", 
                                                            ifelse(x > X2.df, "+", "-")))

md <- melt(Ec.cells)

# plot
mh_plot <- qplot(x=Apic, y=Acar, data=md, fill=value, geom="tile", ylim = rev(levels(md$Acar))) + 
  scale_fill_gradient2(limits=c(-600, 600)) 
mh_plot
```


## Species differences in the sensitivity of thermally-responsive transcripts to changes in temperature

To examine how the sensitivity of transcript expression to changes in temperature differed between the two species, we compared the mean temperatures of transcript activation, defined as the temperature at which the transcript showed the greatest positive change in expression. **High** and **Low** transcripts were included only for the direction in which they were upregulated, while **Bimodal** genes wre included for analysis at both high and low temperature activation. Transcripts turned on at high temperatures in *A. carolinensis* were activated on average at `r round(T_on.high.ttest$estimate[1], 1)`°C compared to `r round(T_on.high.ttest$estimate[2], 1)`°C for *A. picea* (*t*~`r round(T_on.high.ttest$parameter, 0)`~ = `r round(T_on.high.ttest$statistic, 1)`, *P* < `r max(T_on.high.ttest$p.value, 0.001)`; Fig. 2). For low temperatures, there was no difference in the mean temperature of transcript activation (`r round(T_on.low.ttest$estimate[1], 1)`°C for *A. carolinensis* compared to `r round(T_on.low.ttest$estimate[2], 1)`°C for *A. picea*; *t*~`r round(T_on.low.ttest$parameter, 0)`~ = `r round(T_on.low.ttest$statistic, 1)`, *P* < `r round(max(T_on.low.ttest$p.value, 0.001), 3)`; Fig. 2). 

Similarly, for the **Intermediate** transcripts we compared the mean temperatures of down-regulation of transcript expression between species at both high (> 21°) and low (< 21°C) temperatures. Overall, the thresholds for down-regulation were shifted toward cooler temperatures than the corresponding thresholds for gene activation, with a shift of approximately ... at the higher temperature extremes and ...  at low temperature extremes. Consistent with the results for up-regulated transcripts, we found greater inertia of gene expression to changes in temperature for *A. carolinensis* than *A. picea* drive by a differential response to heat stress. With increasing temperatures, **Intermediate** transcripts were down-regulated at `r round(T_off.int.high.ttest$estimate[1], 1)`°C for *A. carolinensis* compared to `r round(T_off.int.high.ttest$estimate[2], 1)` for *A. picea* (*t*~df=`r round(T_off.int.high.ttest$parameter, 0)`~ = `r round(T_off.int.high.ttest$statistic, 1)`, *P* < `r round(max(T_off.int.high.ttest$p.value, 0.001), 3)`), while with decreasing temperatures the species did not differ in mean temperature of down-regulation (*A. carolinensis* `r round(T_off.int.low.ttest$estimate[1], 1)` compared to *A. picea* `r round(T_off.int.low.ttest$estimate[1], 1)`; *t*~df=`r round(T_off.int.low.ttest$parameter, 0)`~ = `r round(T_off.int.low.ttest$statistic, 1)`, *P* < `r round(max(T_off.int.low.ttest$p.value, 0.001), 3)`).

**ADD PANEL TO FIG2 WITH DOWNREGULATION OF INTERMEDIATE TRANSCRIPTS**


```{r T_on_plot, echo=FALSE, warning = FALSE, fig.cap = "Histogram with smooth density estimate of temperature of maximum change in expression for transcripts that have increased expression at **Low** and **High** temperatures. Red bars and lines are for *A. carolinensis* while blue bars and lines are for *A. picea*."}

T_on_plot <- ggplot(Ap.T_on, aes(x = T_on, y = ..density.., colour = Species, group = Species, fill = Species)) +
    facet_grid(. ~ type) +
    geom_histogram(position = "dodge", binwidth = 3) +
    geom_density(alpha = 0.2, adjust = 3) + 
    scale_y_continuous(name="Density") +
    scale_x_continuous(name=expression("Temperature of transcript activation"))
print(T_on_plot)
```

## Adaptive changes in basal expression are not linked to transcripts involved in short-term physiological acclimation

We tested the hypothesis of adaptation by genetic assimilation, the process by which transcripts change from being environmentally induced to constitutively expressed, in multiple comparisons. First, we predicted that consistent exposure to a temperature stressor should drive a shift from inducibility under stress conditions to continuously elevated expression (e.g. "frontloading", [@barshis2013]). Thus, the warm-adaptated *A. carolinensis* should display higher basal expression of **High** transcripts than *A. picea*, while the reverse should be true for **Low** transcripts.  Instead, we found that transcript expression at optimal temperatures (about 19.5°C) was always greater in *A. carolinensis* than in *A. picea*. This pattern was true for transcripts that had **High** expression in *A. picea* (Wilcoxon V = `r prettyNum(w1$statistic)`, *P* < `r max(w1$p.value, 0.001)`, 95% CI location shift `r paste(round(w1$conf.int[1], 3), "--", round(w1$conf.int[2], 3))`), **Low** expression in *A. carolinensis* (Wilcoxon V = `r prettyNum(w2$statistic)`, *P* < `r max(w2$p.value, 0.001)`, 95% CI location shift `r paste(round(w2$conf.int[1], 3), "--", round(w2$conf.int[2], 3))`), and **Intermediate** transcripts in both species (Wilcoxon V = `r prettyNum(w3$statistic)`, *P* < `r max(w3$p.value, 0.001)`, 95% CI location shift `r paste(round(w3$conf.int[1], 3), "--", round(w3$conf.int[2], 3))`). 

In general, thermally-responsive transcripts in *A. carolinensis* had higher constitutive expression than those in *A. picea*.

The only exception was the set of genes that had **High** or **Bimodal** expression in *A. picea* but were not thermally-responsive in *A. carolinensis* (blue cells in bottom row of Fig. 1). Rather than being constitutively elevated as predicted, these genes were not expressed (...mean TPM...) in *A. carolinensis*. Overall, we found no support for genetic assimilation as the primary mechanism for adaptive shifts in thermal tolerance. 


## Candidate genes

We examined the reaction norms for transcripts in the GO category "response to stress" which we *a priori* expected to have significant thermal responses (Fig. 3). Although `r resp.stress.responsive.count` transcripts were annotated with this GO term among the thermally-responsive trancripts, this number was not enriched compared to the total number in the transcriptome ($\chi^2$ = `r round(GO.stress.Xsq$statistic, 2)`, P < `r round(GO.stress.Xsq$p.value, 2)`). In addition, of the `r nrow(hsp_all)` genes annotated to heat shock proteins in the transcriptome, `r length(which(hsp_responsive$A22.type %in% c("Bimodal", "High", "Low")))` were upregulated to cold or hot temperatures in *A. picea* while only `r length(which(hsp_responsive$Ar.type %in% c("Bimodal", "High", "Low")))` were upregulated in *A. carolinensis*. 


# Discussion

By using a regression-approach, we are able to quantitatively describe changes in transcript expression, rather than simply report the numbers of differentially-expressed transcripts. We found that about 70% of the thermally-responsive transcripts (7,202) have non-linear responses to temperature. 

This result is consistent with the only other work we are aware of that examines reaction norms of gene expression, which found that most genes in yeast have have expression differences in response to copper exposure [@hodgins-davis2012]. Finding that thousands of genes responded to temperature highlights the limitations of previous studies [@meyer2011; @barshis2013; @palumbi2014; @oneil2014] that used only a few temperature levels and found that only a few dozen to hundred genes are involved in thermal responses. Underlying the importance of thermal tolerance as a physiological trait [@kingsolver2009], the genetic basis of thermal tolerance is likely to be much more complex than indicated by these previous studies.  

The patterns of transcript expression between species differed substantially, with more transcripts down-regulated in *A. carolinensis* and more up-regulated in *A. picea* in response to thermal extremes (Table 1, Fig. 1). The results of gene set enrichment analysis give an indication of what might drive this difference. The transcripts that are shut-down by *A. carolinensis* in response to cold temperatures, but up-regulated by *A. picea* include gene ontology clusters involved in metabolism, transport and regulation. Thus, *A. carolinensis*, the more southerly species, shuts-down metabolism to a greater extent than *A. picea* and so must rely on other mechanisms such as behavior or increased proteome stability to tolerate thermal stress. In contrast, *A. picea* increases metabolism to a greater extent than *A. carolinensis* with more extreme temperatures, potentially to mount a greater molecular response to thermal stress. Consistent with this hypothesis, we found that of the 7 thermally-responsive heat shock proteins, all 7 increase expression in response to thermal extremes in *A. picea*, but only 4 do in *A. carolinensis*. This result supports our expectation that *A. picea* would rely to a greater extent upon activation of the cellular stress response and other protective genes to a greater extent than *A. carolinensis*. 

Our results are congruent with evidence from other systems [@barshis2013] that thermally-stressful habitats select for investment in tolerance, while organisms from less stressful environments rely on plastically-induced resistance. Although the heat-shock response is one of the most conserved across living organisms (ref), it is energetically expensive, particularly under chronic stress conditions (ref).  Under such circumstances, it may be advantageous to proactively prevent thermal damage even at the cost of reduced metabolic efficiency, either by maintaining a higher basal level of chaperone proteins (ref) or by increasing the thermal stability of proteins at the expense of catalytic activity (Arnold et al. 2001).  Thus, although in the short term increasing temperature stress leads to a quantitatively stronger induced response, adapting to such stress over evolutionary time appears to require a qualitative shift in mechanism of resistance that can alter not only the magnitude, but the sign of gene expression change in response to temperature.  Whether such a shift would be possible in the compressed time frame of projected climate change, particularly for long-lived organisms such as ants, is likely to be critical in determining the capacity of populations to adapt to more frequent and long-lasting stressors. 

```{r max_exp_PDF, eval = FALSE, echo = FALSE, warning = FALSE, message = FALSE, fig.cap = "Probability density function of the temperature of maximum expression for all thermally-responsive transcripts in each species."}
# reshape data
Ap.df$Species <- ifelse(Ap.df$colony == "ApVT", "A. picea", "A. carolinensis")

maxexplot <- ggplot(Ap.df, aes(x=max.val, fill=Species)) + 
  geom_density(alpha=0.2, position="identity") + 
  labs(x = "Temperature of maximum expression", y = "Density")

print(maxexplot)

####################################
### CHANGE TO REACTION NORM FOR HSPs 
####################################

```


Given the differences in the number of thermally-responsive transcripts between the two species (Fig. 1), it is worth noting the similarity in expression patterns between species. In both species, there are 3-4 $\times$ more transcripts upregulated at cold than high temperatures  (Fig. 3). A potential explanation for this pattern is that selection has been stronger on upper than lower thermal limits, thereby reducing genetic variation at high temperatures [@kingsolver2013; @kelly2013]. This explanation is consistent with the observation in *Aphaenogaster rudis* [@warren2013] and other ectotherms [@addo-bediako2000; @hoffmann2013] that critical maximum temperatures vary less among taxa than do critical minimum temperatures. At least among *Drosophila* taxa, limits to thermal maxima appear to be due to a scarcity of genetic variation in thermal tolerance [@kellermann2012]. In this light, for *A. picea* and *A. carolinensis*, our results suggest that adaptability to increased temperatures is more limited than to cold temperatures, and these species risk replacement by more heat-tolerant competitors [@rodriguez-cabal2012]. 

In contrast to the predictions of the genetic assimilation hypothesis that adaptation occurs via changes from plastically-induced to constitutive expression (e.g. genetic assimilation), we found lower basal transcript expression in *A. picea* for transcripts that were up-regulated in response to cold temperatures (**Low** transcripts) in *A. carolinensis*. We also observed greater basal expression in *A. carolinensis* for **High** transcripts in *A. picea* as well as for transcripts that were **Intermediate** in both species. A potential explanation for the consistently higher basal transcript expression in the warmer-environment species, *A. carolinensis*, is that this species has a higher metabolic rate and thus requires greater transcript expression.    

As with all *de novo* transcriptomes, our work has some caveats. First, given that all sequenced ant genomes to date have about only 18,000 genes [@gadau2012], our transcriptome is highly fragmented. Although we took steps to remove contaminants and redundant transcripts, some likely remain, in addition to partially-assembled transcripts. A genome assembly, in progress, will help to reduce fragmentation. Second, the quality of the annotation for a non-model system such as *Aphaenogaster* is not as good as it would be for distantly-related model organisms *Drosophila* and *Apis*. Finally, the mapping of changes in gene expression to organismal fitness is far from direct [@feder2005]. Large differences in patterns of gene expression may have only small effects on organismal fitness. Given these caveats, transcriptomic studies promise to offer many new insights into how organisms respond and will respond to environmental change, especially for long-lived species that are intractable to study in the lab. 

An additional concern is differences in gene expression among species due to drift. While we do in fact expect some genes to differ in expession due to neutral processes, if this was the primary driver of differences among species we would expect to see equal distribution of counts in the off-diagonals of the matched observations (Fig. 1). Instead we find that certain categories are over-represented, and that the differences in the reaction norm shapes are consistent with *a priori* expectations. 

## Conclusions 

In this work, we have brought reaction norms to the genomic era by characterizing the thermal reactionomes of two temperate ant species, *Aphaenogaster picea* and *A. carolinensis*. At least 10% of their transcriptomes are thermally-responsive. For both species, more genes responded to cold than hot temperatures. Together with physiological data showing more variation in critical minimum than maximum temperatures, characterization of the reactionome suggests that these species, and likely other ectotherms, have less adaptive potential in response to hot than cold temperatures. Our results also indicate that these two ant species have different responses to thermal extremes. *A. picea* responds by increasing metabolism and activating heat shock proteins and other protective molecules, whereas *A. carolinensis* shuts down metabolism and must rely on other mechanisms for thermal tolerance. The thermal reactionomes of these two species provide key insights into the genetic basis of thermal tolerance, and a resource for the future study of ecological adaptation in ant species. Finally, the reactionome itself illustrates a new direction for characterizing acclimation and adaptation in a changing climate. 


# Material and Methods

## Samples

Ants of the genus *Aphaenogaster* are some of the most abundant in eastern North America [@lubertazzi2012], and species as well as populations within species differ in the critical maximum and minimum temperatures [@warren2013]. Temperature is a potentially strong selective force for ground-nesting ant populations, which must tolerate seasonally freezing winters and hot summers. On shorter time scales, individual workers can experience extreme thermal environments when they leave the thermally buffered ant nest to forage for food [@diamond2012]. 

We used colonies of the ant species *Aphaenogaster picea*, collected from Molly Bog, Vermont (University of Vermont Natural Areas; 44.508°N, -72.702°W) and *Aphaenogaster carolinensis*, collected from Durham, North Carolina (36.037° N -78.874° W). The phylogeny of *Aphaeogaster* is currently unresolved [@umphrey1996], but putative species identification was possible based on morphology (Bernice Bacon DeMarco, pers. communication). Colonies of these species were maintained in the lab for 6 months prior to experimentation. 

From each species, we haphazardly collected 12 ants from a single colony at the same time on 12 days. Each day, the 12 ants were placed in glass tubes into a water bath maintained at one of 12 randomly-assigned temperatures (0° to 38.5°C, in 3.5° increments) for one hour. The minimum and maximum temperatures were selected based on previous work showing that these temperatures are close to the critical minimum and maximum temperatures for *Aphaenogaster* [@warren2013]. At the end of the hour, the ants were flash frozen in liquid nitrogen and stored at -80°C. We extracted mRNA by homogenizing three pooled ants in 500 uL of RNAzol buffer with zirconium silicate beads in a Bullet Blender (Next Advance; Averill Park, NY), followed by RNAzol extraction (Molecular Research Center Inc; Cincinnati, OH) and then an RNeasy micro extraction (Qiagen Inc; Valencia, CA) following the manufacturer's instructions. 

## Sequencing, assembly and annotation

For each species, the 12 samples were barcoded and sequenced in a single lane of 2 x 100bp paired-end reads on an Illumina HiSeq 1500 yielding 200 and 160 million reads for the *A. picea* and *A. carolinensis* samples respectively. Reads were filtered to remove Illumina adapter sequences and low quality bases using the program Trimmomatic [@lohse2012]. 

We assembled the sequenced reads into the full set of mRNA transcripts, the transcriptome, for the combined data set from both species using the Trinity *de novo* transcriptome assembly program [@grabherr2011]. *De novo* transcriptome assembly is prone to falsely identifying alternative transcripts and identifying inaccurate transcripts that are chimeric (e.g. regions of two separate transcripts that assemble into a false, or chimeric, third transcript) [@yang2013]. We removed potentially false transcripts by first running the program `CAP3` [@huang1999] to cluster sequences with greater than 90% similarity and merge transcripts with overlaps longer than 100 bp and 98% similar in length. Second, we ran the program `uclust` which clusters sequences completely contained within longer sequences at greater than 90% similarity (see Supplemental Information for exact parameters). We used liberal values (90% similarity) to merge orthologous transcripts in the two species that may not have assembled together in the initial *de novo* transcriptome assembly. To identify contaminant sequences, we screened our full transcriptome using the program `DeconSeq` [@schmieder2011] with the provided bacteria, virus, archae and human databases of contaminants. 

The Trinity *de novo* transcriptome assembly for both species assembled together included 126,172 transcripts with a total length of 100 million bp. Filtering to remove redundant or chimeric reads resulted in an assembly with 105,536 transcripts. The total length was 63 million bp with an N~50~ length of 895 bp and a mean transcript size of 593 bp. Of the 105,536 filtered transcripts, 55,432 had hits to the NCBI-nr database. Of these, 38,711 transcripts mapped to GO terms, 1,659 transcripts were identified to an enzyme and 18,935 transcripts mapped to a domain with >50% coverage. We removed 5,675 transcripts identified as known contaminants, leaving 99,861 clean transcripts. Of these, 47% have best hits to transcripts from the class Insecta [(Supplementary Krona plot)](http://johnstantongeddes.org/assets/files/ApTranscriptome_FA.html).

To determine the putative function of the transcripts, we used functional annotation of the transcriptome assembly using the web-based tool `FastAnnotator` [@chen2012] which annotates and classifies transcripts by Gene Ontology (GO) term assignment, enzyme identification and domain identification. 

## Thermally-responsive transcripts

We quantified expression of each transcript using the program `Sailfish` [@patro2014] and use the bias-corrected transcripts per million (TPM) [@wagner2012] as our measure of transcript expression. We included the contaminant transcripts identified by DeconSeq at the quantification stage to avoid incorrectly assigning reads to other transcripts, but removed these from further analyses. As preliminary examination of the data (see supplemental...) indicated that 7°C samples may have been mis-labeled, we omitted these data from the analysis. The expression values were highly correlated between species at each temperature treatment (r^2^ > 0.98) indicating that assembling the transcriptome with data from both species was justified (Supplemental).

To identify transcripts that have significant changes in expression across the thermal gradient, we fit the regression model

$$ log(TPM + 1) = \beta_0 + \beta_1(species) + \beta_2(temp) + \beta_3(temp^2) + \beta_4(species * temp) + \beta_5(species * temp^2) + \epsilon $$
    
to each transcript. We used $log(TPM + 1)$ as the response to control for skew in the expression data. For a continuous predictor such as temperature, this regression approach is preferred to an ANOVA approach as it can reveal non-linear responses such as hump-shaped or threshold effects [@cottingham2005]. This method is robust to over-dispersion because we expect errors in the read count distribution [@anders2010] to be independent with respect to temperature. To correct for multiple testing, we calculated adjusted *P* values using the False Discovery Rate (FDR) approach of Benjamini and Hochberg [-@benjamini1995], retaining transcripts under the 5% FDR threshold.

We identified thermally-responsive transcripts as the subset under the FDR threshold that had significant $\beta_2(temp)$, $\beta_3(temp^2)$, $\beta_4(species * temp)$ or $\beta_5(species * temp^2)$ terms after step-wise model selection by AIC. All analyses were performed in R [@r_core_team2013] and are fully reproducible (Supplemental file). 

For each thermally-responsive transcript, we predicted expression levels across the tested thermal range using the final linear model separately for each species. We used the predicted transcript expression values to partition the thermally-responsive transcripts into five expression categories for each species; transcripts that had greatest expression at high (> 31°) temperatures (**High**), low (< 10°C) temperatures (**Low**), intermediate (10 - 30°C) temperatures (**Intermediate**), both high and low temperatures (**Bimodal**) or were not thermally-responsive in that species (*NotResp*). For the **Bimodal** group, we required that expression at both low and high temperatures was at least one standard deviation greater than the expression at the mean temperature of 19.5°C.

## Species-level comparisons

We performed comparative analyses between the two species focusing on the transcripts that had a significant species x temperature interaction to determine if the two species differed in the tempertaure of transcript activation and sensitivity of the thermally-responsive transcripts. For each transcript, we predicted expression levels across temperatures for each species using the full linear model, and then grouped transcripts into the four responsive categories, **High**, **Low**, **Intermediate**, and **Bimodal**, as well as a fifth category **Not Responsive**. To test whether the temperature at which thermally-responsive transcripts are activated differs between species, we identified the temperature at which there was the greatest change in expression for each transcript in each species. We grouped the **High** transcripts along with the high temperature end of the **Bimodal** transcripts, and did the same for **Low** transcripts. We then performed a *t*-test to determine if the mean temperature of transcript activation differed between the two species for each group. 

To determine if the sensitivity of transcripts to thermal extremes differed between the species, we used the subset of **Intermediate** and **Bimodal** transcripts in each species. For each **Intermediate** transcript in each species, we randomly sampled 1,000 temperature values weighted by the expression function (e.g. point of maximum expression had highest probability of being sampled). We then calculated the standard deviation of this random draw, and performed a *t*-test comparing the standard deviations of expression between species. Finding a greater standard deviation of expression would be indicative of less sensitivity to changes in temperature. For the **Bimodal** transcripts, we performed the same analysis as with the **Intermediate** transcripts, but with a greater standard deviation being indicative of a more sensitive transcript (convex reaction norm) while a low standard deviation being a transcript with greater inertia of expression in response to temperature changes (nearly flat).Thus, finding one species had overall greater standard deviation of expression for **Bimodal** transcripts would be indicative of greater thermal sensitivity for that species. 

To test for genetic assimilation, we compared basal expression levels (expression at 19.5° C) for the subset of transcripts that were responsive to cold temperatures (**Low** transcripts) in the southern *A. carolinensis* species, or hot temperatures (**High** transcripts) in the northern *A. picea* species using non-parametric two-sided Wilcoxon tests.

## Gene set enrichment analysis

To describe the overall molecular toolkit for thermal tolerance in *Aphaenogaster*, we performed gene set enrichment analysis (GSEA) on the intersection of each expression category in the two species using the package `topGO` [@alexa2006; @alexa2010] with the `parentChild` algorithm [@grossmann2007] implemented in R [@r_core_team2013]. Briefly, this approach identifies GO terms that are overrepresented in the significant transcripts relative to all GO terms in the transcriptome, accounting for dependencies among the GO terms. We clustered enriched GO terms from the `topGO` analysis by semantic similarity using the package `GOSemSim` [@yu2010] to reduce redundancy. 

## Data availability

The raw Illumina reads are available at [...], the script for analysis and version history is available at (http://github.com/johnstantongeddes/ApTranscriptome).  



# Figure Legends




# References

