---
title: Investigation of thermal reactionomes reveals stress tolerance vs. resistance in
  warm and cold-climate ant species
author: \centering{John Stanton-Geddes \textsuperscript{1}, Andrew Nguyen \textsuperscript{1},
  James Vincent \textsuperscript{2}, Mahesh Vangala \textsuperscript{2}, Robert R.
  Dunn \textsuperscript{3}, Nathan J. Sanders \textsuperscript{4}, Nicholas J. Gotelli
  \textsuperscript{1}, Sara Helms Cahan \textsuperscript{1} \\ \vspace{5 mm} \textsuperscript{1}
  Department of Biology, University of Vermont \\ \textsuperscript{2} Vermont Genetics
  Network, University of Vermont \\ \textsuperscript{3} Department of Biological Sciences,
  North Carolina State University \\ \textsuperscript{4} Center for Macroecology,
  Evolution and Climate, University of Copenhagen}
csl: ecology.csl
output:
  pdf_document:
    fig_caption: yes
    includes:
      in_header: header.tex
    keep_tex: yes
    latex_engine: xelatex
    number_sections: yes
    toc_depth: 2
  html_document: null
  word_document: default
includes: header.tex
bibliography: ApTranscriptome.bib
---

\linenumbers

\newpage

# Abstract 

Thermal tolerance is a key trait governing the distributions of species and underlies their future responses to climate change. Here, we characterize the thermal “reactionome” – the reaction norm for all genes in an organism's transcriptome measured across a thermal gradient to understand the genetic basis of thermal tolerance. We characterized the reactionomes of two common ant species in eastern U.S. forests: the northern cold-climate *Aphaenogaster picea* and the southern warm-climate *Aphaenogaster carolinensis*. We found that at least 10% of genes were thermally responsive. The warm-adapted *A. carolinensis* displayed higher basal levels of gene expression, upregulated fewer heat shock proteins, and exhibited greater inertia of gene expression chang than did *A. picea*. This pattern is consistent with both constitutive thermal defenses and enhanced proteome stability. *A. carolinensis* also shut down more genes in response to thermal extremes than *A. picea*. A subset of these genes were upregulated in *A. picea*, including gene ontology terms such as metabolism, cellular organization, glycoprotein synthesis and DNA repair. These differences between species suggest that the two closely related species employ different strategies for coping with temperature stress. Namely *A. picea* actively resists thermal shock by synthesizing heat shock proteins and other protective molecules to maintain function, whereas *A. carolinensis* slows down metabolism and relies upon other mechanisms to tolerate thermal stress. More broadly, this study adds to a growing body of evidence that thermally stressful habitats select for an evolutionary shift in response mechanism away from high-cost damage repair and toward lower-cost prevention.  



# Significance Statement

Successfully coping with progressive changes in climate will require both short-term physiological acclimation and longer-term adaptive changes. To better understand the relationship between these two timescales, we compared gene expression as a function of temperature in two closely-related ant species: the cool-climate *Aphaenogaster picea* and the warm-climate *A. carolinensis*. We found surprisingly divergent transcriptomic responses in the two species. *A. picea* responded to high and low temperatures through upregulation of a larger proportion of its transcriptome with a more robust heat shock response. In contrast, *A. carolinensis* reacted to thermal stress to a greater extent through downregulation, reducing metabolic gene expression and requiring more extreme temperatures to trigger a protective response. These results suggest that adaptation to chronic high temperatures involves qualitatively different mechanisms than those used to cope with rarely-encountered stressors, which may limit adaptive potential of northern species in the face of rapidly shifting climatic zones.



# Introduction

Temperature regulates biological activity and shapes diversity from molecular to macroecological scales [@brown2004; @kingsolver2009]. Thermal tolerance, the ability of individuals to maintain function and survive thermal extremes, varies considerably among species and populations [@cowles1939; @deutsch2008; @kingsolver2013; @hoffmann2013]. These differences in thermal tolerance are largely genetic [@krebs1997; @kellermann2012] with a highly polygenic basis [@krebs1998; @williams2012; @morgan2006; @takahashi2011]. Outside of candidate genes [@krebs1998], little is known about the genetic basis of this variation or how it has evolved. To address this, we need information on the extent to which geographic differentiation and selection has acted upon the diversity of genes, and their plasticity of expression, involved in thermal tolerance. This information is a critical for understanding the adaptive potential of species to future climates [@hoffmann2008] and the link between short-term physiological acclimation and longer-term adaptation to increased temperatures [@somero2010].   

In recent years, whole organism gene expression approaches (e.g. transcriptomics) using high-throughput RNA sequencing (RNAseq) technology have been widely applied to identify genes underlying thermal tolerance [@meyer2011; @teets2012; @barshis2013; @oneil2014] and other traits. These RNAseq studies have used an ANOVA-type experimental or sampling design with only a few environmental levels. However, temperature and other environmental factors vary continuously in nature. As a result, many patterns of biologically-relevant plasticity are likely to be missed with ANOVA-style designs that compare only a few discrete levels of an environmental variable [@murren2014]. Similarly, gene expression studies that look at just a few environments (e.g. hot vs. cold) are likely to miss key differences among species that are due not just to whether it is hot, but rather how hot it is. What is needed is a reaction norm approach, which describes variation in the phenotype of a single genotype in response to a continuous environmental gradient [@gomulkiewicz1992]. Reaction norms differ not only in mean values, but also in their shapes [@hoffmann2013; @murren2014], and differences in the shape of reaction norms are often greater than differences in mean values of either species or populations [@murren2014]. 

Regardless of the experimental design or treatment of temperature as a categorical variable with few levels, an important suite of issues often hampers many RNAseq studies that are designed to identify the genetic basis of environmental adaptation [@meyer2011; @teets2012; @barshis2013; @oneil2014]. Namely, lists of differentially-expressed genes derived from RNAseq results are prone to false positives [@gonzalez2013], depend on the genetic background of the organism [@sarup2011], may miss genes with small expression changes that have detectable phenotypic effects [@yan2002; @padovan2013], identify genes that are correlated, but not causative, of phenotypes [@fay2004] and may not be directly informative about organismal status [@feder2005]. Genome-wide association studies have faced a similar problem: phenotypic effects of a small number of statistically-significant genes are over-estimated while small effects of a large number of other causal genes are missed [@beavis1994; @larsson2013]. Our understanding of the genetic basis of traits, and our ability to predict evolutionary potential, are compromised by these issues. 

Here, we present a novel method for RNAseq analysis: the *reactionome*, which we define as a characterization of the reaction norm for all genes in an organism's transcriptome across an environmental gradient. We use this method to identify genes that are thermally-responsive in two closely related eastern North American ant species, *Aphaenogaster carolinensis* and *A. picea* [@umphrey1996]. Ants are ecologically-important species involved in multiple symbioses¸ and are key contributors to seed dispersal [@lengyel2009; @rodriguez-cabal2012], litter processing and decomposition [@lyford1963; @folgarait1998; @terborgh2001; @holec2006; @wardle2010]. 

To gain insight on the genetic basis of thermal adaptation, we compared the thermal reactionomes of these species with two specific predictions. First, species from more thermally-stressful environments may rely to a greater extent upon constitutive tolerance than plastically-induced resistance mechanisms [@barshis2013]. Finding that fewer genes are up-regulated to high temperatures in the warmer-climate species *A. carolinensis* would support this prediction. Second, spatially-divergent local adaptation leaves a signal at the molecular level [@savolainen2013]. For gene expression, this local adaptation may be manifested through changes in the sensitivity of responsive genes to changes in temperature. Evidence for local adaptation in gene expression would be observed if heat-responsive genes are up-regulated at higher temperatures in *A. carolinensis* than in *A. picea*, and conversely, that cold-responsive genes are activated at colder temperatures in *A. picea*. 

In addition, we used the thermal reactionomes to examine the relationship between short-term physiological acclimation and longer-term adaptive changes to temperature. Specifically, through the process of genetic assimilation [@pigliucci2006], adaptive responses are hypothesized to involve the same molecular pathways as short-term acclimatory responses. Finding that the genes up-regulated in response to cold temperatures in *A. carolinensis* have increased basal expression in the colder-climate species *A. picea*, and vice versa, would be consistent with this prediction. While none of the above predictions depend on the functional annotation of the responsive genes [@pavlidis2012], insight on the molecular toolkit involved in adaptation to thermal extremes can be gained through examination of the functional annotation of responsive genes using gene set enrichment analysis. For example, in response to thermal extremes, we expect genes in the cellular stress response [@kultz2005] to be up-regulated, whereas we expect genes involved with cell profliferation and non-essential molecular processes to be down-regulated [@somero2011]. To test these predictions for these two exemplar species, we collected RNAseq data at 12 temperatures (every 3.5°C between 0° and 38.5°C) using Illumina short read data, generated the transcriptome using Trinity [@grabherr2011], quantified gene expression using Sailfish [@patro2014], and characterized the reaction norm for each gene using a custom analysis.


# Results

```{r load, echo=FALSE, results='hide', message=FALSE, warning=FALSE}
options(scipen = 1, digits = 3)
load("../ms.RData")
library(data.table)
library(ggplot2)
library(reshape2)
library(pander)
library(RColorBrewer)
library(topGO)
```

The combined *Aphaenogaster* transcriptome assembly contained 99,861 transcripts. Of these, we identified `r prettyNum(sum(sigtable[sigtable$Coefficient %in% c("Temperature", "Temperature:Colony"), "Number.significant"]), big.mark = ",", scientific = FALSE)` transcripts (hereafter referred to as genes) with expression patterns that were thermally-responsive at 5% False Discovery Rate. Of these, xxx had a non-linear relationship with temperature that would not have been detected with a standard differential expression approach. At a minimum `r round(9100/99861, 2)*100`% of the transcriptome is thermally-responsive. ~~However, the closest related ant species with sequenced genomes have only about 17,000 genes [@gadau2012]. Assuming that the number of genes is similar for these two *Aphaenogaster* species, this suggests that up to 50% of the genome (9,100 / 17,000) could be thermally-responsive.~~


```{r mosaic_plot, echo=FALSE, results='asis', fig.cap = "Mosaic plot of the number of thermally-responsive genes by expression type. **Low** are genes with increased expression at low temperatuers (< 10°C), **Intermediate** are genes with maximum expression between 10 - 30°C, **High** are genes with increased expression at high temperatures (> 31°C), **Bimodal** are genes with increased expression at both low and high temperatures, while **NotResp** are genes that are thermally-responsive in one species but not the other. The width of the columns is proportional to the number of genes for each expression type for *A. carolinensis* while the height of the rows is proportional to the number of genes for each expression type for *A. picea.* The rows are colored by expression type for *A. picea*. **Low** The total number of genes for two categories (*A. picea* **Low** - *A. carolinensis* **Low**; *A. picea* **NotResp** - *A. carolinensis* **Bimodal**) is labeled in two boxes for perspective."}
mosaic.colors <- brewer.pal(5, "Blues")
par(mar = c(5.1, 3.1, 2.1, 7.5))
mosaicplot(tt2, ylab = expression(italic("A. picea")), xlab = "", main = "", color = mosaic.colors, cex.axis = 0.01)
labels <- c("Low", "Intermediate", "High", "Bimodal", "NotResp")
text(c(.1,.5,.75,.85,.99), par("usr")[3] - 0.05, srt = 45, adj = 1, labels = labels, xpd = TRUE, font = 1)
text(.5, par("usr")[4]+.1, "A. carolinensis", xpd = TRUE, font = 3)
text(.15, .7, "2,594")
text(.85, .03, "223", cex = .75)
legend(1.05, .9, fill = mosaic.colors, c("Low", "Intermediate", "High", "Bimodal", "NotResp"), xpd=TRUE)
```

## Thermal responses involve greater down-regulation in the warmer-climate species

For each species, the thermally-responsive genes were grouped into five expression categories based on their reaction norms: genes that had greatest expression at high (> 31°) temperatures (**High**), low (< 10°C) temperatures (**Low**), intermediate (10 - 30°C) temperatures (**Intermediate**), both high and low temperatures (**Bimodal**) or were not thermally-responsive in that species (**Not Responsive**). For the **Bimodal** group, we required that expression at both low and high temperatures was at least one standard deviation greater than the expression at midpoint (19.5°C). In both species, the majority of genes were in the **Low** category (Fig. 1). For *A. carolinensis*, this was followed by **Intermediate**, **High** and then **Bimodal**, whereas for *A. picea* the second largest category was **Bimodal** followed by **High** and finally **Intermediate** (Fig. 1). Of the thermally-responsive genes, `r prettyNum(sum(sigtable[sigtable$Coefficient %in% "Temperature:Colony", "Number.significant"]), big.mark = ",", scientific = FALSE)` (`r prettyNum(sigtable[sigtable$Coefficient == "Temperature:Colony", "Number.significant"] / sigtable[sigtable$Coefficient == "Total", "Number.significant"]*100, digits=2)`%) had an expression pattern that differed between the two species. While some of these were responsive in only one species (`r colSums(tt2)[which(colnames(tt2) == "NotResp")]` genes were responsive in *A. carolinensis* but not *A. picea*, and `r rowSums(tt2)[which(rownames(tt2) == "NotResp")]` were responsive in *A. picea* but not *A. carolinensis*), most of these were genes that changed from one expression category to another between the species (i.e. the number of genes in each expression category differed significantly between species; $\chi^2$ = `r round(chi1$statistic, 0)`, P < `r max(chi1$p.value, 0.001)`; Fig. 1).

Specifically, all the diagonals (e.g. **Low** -- **Low**) the *A. carolinensis* **Bimodal** -- *A. picea* **Not responsive** and *A. carolinensis* **Not responsive** -- *A. picea* **High** categories were over-represented compared to a hypothetical homogenous distribution for the contingency table. 

In contrast to expectations, more genes did not respond to high temperatures in *A. carolinensis*. In fact, the only differences were for **Intermediate** genes in *A. carolinensis*, which disproportionatly switched to **Low** or **Bimodal** expression in *A. picea*. Thus, *A. carolinensis*, the warmer-climate species, appears to rely upon shutting down gene expression to a greater extent than *A. picea*, which activates gene expression...


```{r expression_change, echo=FALSE}
cont.tab <- structure(unlist(as.vector(tt2)), .Dim = c(5L, 5L), .Dimnames = structure(list(
    c("Low", "Intermediate", "High", "Bimodal", "NotResp"), c("Low", 
    "Intermediate", "High", "Bimodal", "NotResp")), .Names = c("Ac", 
"Ap")))

# marginal sums and total
rs <- rowSums(cont.tab)
cs <- colSums(cont.tab)
n <- sum(cont.tab)

# grand mean for each class
gm <- (rs + cs) / sum(cont.tab * 2)

# calculate observed values for each cell using overall mean for each expression type
Ec <- outer(gm, gm, "*") *sum(cont.tab)

# get deviations of observed from expected
Ec.dev <- cont.tab - Ec

# overall table P-value
Ec.chistat <- sum((cont.tab-Ec)^2 / Ec)
# P-value
pchisq(Ec.chistat, df=(nrow(cont.tab)-1) * (ncol(cont.tab)-1), lower.tail = FALSE)

# overall minimum chi-squared value `X2.df` for a significantly non-homogenous distribution of counts
df = (ncol(cont.tab) - 1) * (nrow(cont.tab) - 1)
X2.df <- qchisq(p = 0.05, df = df, lower.tail = FALSE)

# posthoc result for each cell
Ec.cells <- sign(cont.tab - Ec) * (cont.tab - Ec)^2 / Ec
Ec.cells.sign.table <- apply(Ec.cells, c(1,2), function(x) ifelse(abs(x) <= X2.df, "0", 
                                                            ifelse(x > X2.df, "+", "-")))

# plot
qplot(x=Ap, y=Ac, data=melt(Ec.cells), fill=value, geom="tile") + 
  scale_fill_gradient2(limits=c(-600, 600)) 
```


## Greater inertia of gene expression response to heat stress in the warmer climate species

To examine how the sensitivity of gene expression to changes in temperature differed between the two species, we compared the mean temperatures of gene activation, defined as the temperatures at which the gene showed the greatest change in expression, for genes turned on at high or low temperatures, including *Bimodal* genes. Genes turned on at high temperatures in *A. carolinensis* were activated on average at `r round(T_on.high.ttest$estimate[1], 1)`°C compared to `r round(T_on.high.ttest$estimate[2], 1)`°C for *A. picea* (*t*~df=`r round(T_on.high.ttest$parameter, 0)`~ = `r round(T_on.high.ttest$statistic, 1)`, *P* < `r max(T_on.high.ttest$p.value, 0.001)`; Fig. 2). For genes with increased expression at low temperatures, there was a negligible difference in the mean temperature of gene activation (`r round(T_on.low.ttest$estimate[1], 1)`°C for *A. carolinensis* compared to `r round(T_on.low.ttest$estimate[2], 1)`°C for *A. picea*; *t*~df=`r round(T_on.low.ttest$parameter, 0)`~ = `r round(T_on.low.ttest$statistic, 1)`, *P* < `r round(max(T_on.low.ttest$p.value, 0.001), 3)`; Fig. 2). 

Similarly, for the **Intermediate** genes we compared the mean temperatures of downregulation of gene expression among species at both high (> 21°) and low (< 21°C) temperatures. Consistent with the results for up-regulated genes, we found that with increasing temperatures, the **Intermediate** genes were down-regulated at `r round(T_off.int.high.ttest$estimate[1], 1)`°C for *A. carolinensis* compared to `r round(T_off.int.high.ttest$estimate[2], 1)` for *A. picea* (*t*~df=`r round(T_off.int.high.ttest$parameter, 0)`~ = `r round(T_off.int.high.ttest$statistic, 1)`, *P* < `r round(max(T_off.int.high.ttest$p.value, 0.001), 3)`), while with decreasing temperatures the species did not differ in mean temperature of down-regulation (*A. carolinensis* `r round(T_off.int.low.ttest$estimate[1], 1)` compared to *A. picea* `r round(T_off.int.low.ttest$estimate[1], 1)`; *t*~df=`r round(T_off.int.low.ttest$parameter, 0)`~ = `r round(T_off.int.low.ttest$statistic, 1)`, *P* < `r round(max(T_off.int.low.ttest$p.value, 0.001), 3)`). Overall, these results indicate greater intertia of gene expression to changes in temperature for *A. carolinensis* than *A. picea*. 


```{r T_on_plot, echo=FALSE, warning = FALSE, fig.cap = "Histogram with smooth density estimate of temperature of maximum change in expression for genes that have increased expression at **Low** and **High** temperatures. Red bars and lines are for *A. carolinensis* while blue bars and lines are for *A. picea*."}

T_on_plot <- ggplot(Ap.T_on, aes(x = T_on, y = ..density.., colour = Species, group = Species, fill = Species)) +
    facet_grid(. ~ type) +
    geom_histogram(position = "dodge", binwidth = 3) +
    geom_density(alpha = 0.2, adjust = 3) + 
    scale_y_continuous(name="Density") +
    scale_x_continuous(name=expression("Temperature of gene activation"))
print(T_on_plot)
```

## Adaptive changes in basal expression are not linked to genes involved in short-term physiological acclimation

We looked for evidence of genetic assimilation (the process by which physiologically-relevant genes change from being environmentally-induced to basally-expressed) by comparing gene expression at intermediate temperatures for genes that were up-regulated at low temperatures in *A. carolinensis* or high temperatures in *A. picea*. While finding that genes upregulated to thermal extremes 


Contrary to expectations, for **Low** genes in *A. carolinensis* gene expression at intermediate temperatures was lower in *A. picea* than *A. carolinensis* (Wilcoxon V = `r prettyNum(w2$statistic)`, *P* < `r max(w2$p.value, 0.001)`, 95% CI location shift `r paste(round(w2$conf.int[1], 3), "--", round(w2$conf.int[2], 3))`) . Moreover, for **High** genes in *A. picea* (Wilcoxon V = `r prettyNum(w1$statistic)`, *P* < `r max(w1$p.value, 0.001)`, 95% CI location shift `r paste(round(w1$conf.int[1], 3), "--", round(w1$conf.int[2], 3))`) and **Intermediate** genes in both species (Wilcoxon V = `r prettyNum(w3$statistic)`, *P* < `r max(w3$p.value, 0.001)`, 95% CI location shift `r paste(round(w3$conf.int[1], 3), "--", round(w3$conf.int[2], 3))`), gene expression at intermediate temperatures was always higher in *A. carolinensis*. 


## Gene set enrichment analysis

To describe the molecular toolkit underlying *Aphaenogaster* response to thermal extremes, we used gene set enrichment analysis to identify gene ontology (GO) categories enriched for each expression category separately. First, we looked at the genes shared in each expression category (e.g. the intersection) for each species (Table S1). Out of a total of `r length(score(Ap.BP.resultParentChild))` GO terms scored, we found `r nrow(Ap.gsea.intersect[which(Ap.gsea.intersect$Type == "High"), ])` enriched (*P* < 0.01) terms for **High** genes. Clustering of terms by semantic similarity yielded 3 clusters, with 5 of the 7 terms being involved in "regulation of biosynthetic/metabolic process". For **Low** genes, `r nrow(Ap.gsea.intersect[which(Ap.gsea.intersect$Type == "Low"), ])` GO terms were enriched. These terms grouped into 15 clusters, including "protein modification", "protein metabolic process", "lipid metabolic process", "regulation of signaling", "protein transport/localization", "detection of stimulus", and "chondroitin sulfate biosynthesis". For **Bimodal** genes, `r nrow(Ap.gsea.intersect[which(Ap.gsea.intersect$Type == "Bimodal"), ])` GO terms were enriched. These terms grouped into 5 clusters, including "biological regulation", "regulation of cellular biosynthesis", "gene expression", "signaling", and "protein modification process". For **Intermediate** genes, `r nrow(Ap.gsea.intersect[which(Ap.gsea.intersect$Type == "Intermediate"), ])` GO terms were enriched. These terms grouped into 3 clusters, including "lipid biosynthesis/metabolism", "glycoprotein biosynthesis/metabolism", and "DNA organization". Overall, GSEA indicates that molecular responses to thermal extremes involve large-scale cellular and organizational changes, though the molecular responses to high temperatures are more limited than those to low temperatures. 

We further used gene set enrichment analysis for the genes that shifted from being shut-down at extreme temperatures, **Intermediate**, in *A. carolinensis* to having increased expression in **Low** temperatures in *A. picea*. This group included `r nrow(Ar.int.A22.low.gsea)` GO terms that grouped into 12 clusters, including "protein metabolic process", "glycosylation" "transport", "lipid metabolic process", "GTP metabolism", "positive regulation of cellular process" and "organelle organization". For **Intermediate** genes in *A. carolinensis* that shifted **Bimodal** in *A. picea*, there were `r nrow(Ar.int.A22.bim.gsea)` enriched GO terms including "phospholipid catabolic process", "L-amino acid transport", "macromolecule metabolic process", "gluatmate receptor signaling pathway" and "histone modification". These results indicate that in response to thermal extremes, the warmer-climate species, *A. carolinensis*, shuts down metabolism to a greater extent than the colder-climate species, *A. picea*, which increases metabolism. 

**Table** Table of enriched GO terms for each expression category in each species.


## Candidate genes

We examined the reaction norms for genes in the GO category "response to stress" which we *a priori* expected to have significant thermal responses (Fig. 3). Though we found `r resp.stress.responsive.count` genes annotated with this GO term among the thermally-responsive trancripts, this number was not enriched compared to the total number in the transcriptome ($\chi^2$ = `r GO.stress.Xsq$statistic`, P < `r round(GO.stress.Xsq$p.value, 2)`).


# Discussion

By using a regression-approach, we are able to quantitatively describe changes in gene expression, rather than simply report the numbers of differentially-expressed genes. Specifically, we found that at least 10%, and potentially 50%, of the genes in these species were thermally-responsive. This result is consistent with the only other work we are aware of that examines reaction norms of gene expression, which found that most genes in yeast have variable expression in response to copper exposure [@hodgins-davis2012]. In general, finding thousands of genes that respond to temperature highlights the limitations of previous methods that only found a few dozen to hundred genes are involved in thermal response from recent studies in corals [@meyer2011; @barshis2013; @palumbi2014] and butterflies [@oneil2014]. Underlying the importance of thermal tolerance as a physiological trait [@kingsolver2009], the genetic basis of thermal tolerance is likely to be much more complex than indicated by these previous studies. 

The patterns of gene expression between species differed substantially, with more genes down-regulated in *A. carolinensis* and more up-regulated in *A. picea* in response to thermal extremes (Fig. 3). The results of the gene set enrichment analysis give an indication of what might drive this difference. The genes that are shut-down by *A. carolinensis* in response to cold temperatures, but up-regulated by *A. picea* include gene ontology clusters involved in metabolism, transport and regulation. Thus, *A. carolinensis*, the more southerly species, shuts-down metabolism to a lesser extent than *A. picea* and so must rely on other mechanisms such as behavior (e.g. retreat into its nest) or proteome stability to tolerate thermal stress. In contrast, *A. picea* increases metabolism to a greater extent than *A. carolinensis* with more extreme temperatures, potentially to mount a greater molecular response to thermal stress. Consistent with this hypothesis, we found that of the 7 thermally-responsive heat shock proteins, all 7 increase expression in response to thermal extremes in *A. picea*, but only 4 do in *A. carolinensis*. This result supports our expectation that *A. picea* would rely to a greater extent upon activation of the cellular stress response and other protective genes to a greater extent than *A. carolinensis*. Our results are at least congruent with evidence from other systems [@barshis2013] that thermally-stressful habitats select for investment in tolerance, while organisms from less stressful environments rely on plastically-induced resistance. ...Sara - something about enzyme efficiency tradeoffs???


```{r max_exp_PDF, eval = FALSE, echo = FALSE, warning = FALSE, message = FALSE, fig.cap = "Probability density function of the temperature of maximum expression for all thermally-responsive genes in each species."}
# reshape data
Ap.df$Species <- ifelse(Ap.df$colony == "ApVT", "A. picea", "A. carolinensis")

maxexplot <- ggplot(Ap.df, aes(x=max.val, fill=Species)) + 
  geom_density(alpha=0.2, position="identity") + 
  labs(x = "Temperature of maximum expression", y = "Density")

print(maxexplot)

####################################
### CHANGE TO REACTION NORM FOR HSPs 
####################################

```

Given the differences in the number of thermally-responsive genes between the two species (Fig. 1), it is worth noting the similarity in expression patterns between species. There are 3-4 $\times$ more genes upregulated at cold than high temperatures in both species (Fig. 3). A potential explanation for this pattern is that selection has been stronger on upper than lower thermal limits, thereby reducing genetic variation at high temperatures [@kingsolver2013; @kelly2013]. This explanation is consistent with the observation in *Aphaenogaster rudis* [@warren2013] and other ectotherms [@addo-bediako2000; @hoffmann2013] that critical maximum temperatures vary less among taxa than do critical minimum temperatures. At least among *Drosophila* taxa, these limits to thermal maxima appear to be due to a scarcity of genetic variation in thermal tolerance [@kellermann2012]. In this light, for *A. picea* and *A. carolinensis*, our results suggest that adaptability to increased temperatures is limited, and these species risk replacement by more heat-tolerant competitors [@rodriguez-cabal2012]. 

In contrast to expectations that adaptation occurs via changes from plastically-induced to constitutive expression (e.g. genetic assimilation), we found lower basal gene expression in *A. picea* for genes that were up-regulated in response to cold temperatures (**Low** genes) in *A. carolinensis*. We also observed greater basal expression in *A. carolinensis* for **High** genes in *A. picea* as well as for genes that were **Intermediate** in both species. A potential explanation for the consistently higher basal gene expression in the warmer-environment species, *A. carolinensis*, is that this species has a higher metabolic rate and thus requires greater gene expression.    

Thermal safety margins...

As with all *de novo* transcriptomes, our work has some important caveats. First, given that all sequenced ant genomes to date have about 18,000 genes [@gadau2012], our transcriptome is highly fragmented. Although we took steps to remove contaminants and redundant transcripts, some likely remain, in addition to partially-assembled transcripts. A genome assembly, in progress, will help to reduce fragmentation. Second, the quality of the annotation for a non-model system such as *Aphaenogaster* is not as good as it would be for distantly-related model organisms *Drosophila* and *Apis*. Finally, the mapping of changes in gene expression to organismal fitness is far from direct [@feder2005]. Large differences in patterns of gene expression may only have small effects on organismal fitness. Given these caveats, transcriptomic studies promise to offer many new insights into how organisms respond and will respond to environmental change, especially for long-lived species that are intractable to study in the lab. 

## Conclusions 

In this work, we have brought reaction norms to the genomic era by characterizing the thermal reactionomes of two temperate ant species, *Aphaenogaster picea* and *A. carolinensis*. At least 10% of their transcriptomes are thermally-responsive. For both species, more genes responded to cold than hot temperatures. Together with physiological data showing more variation in critical minimum than maximum temperatures, characterization of the reactionome suggests that these species, and likely other ectotherms, have less adaptive potential in response to hot than cold temperatures. Our results also indicate that these two ant species have different responses to thermal extremes. *A. picea* responds by increasing metabolism and activating heat shock proteins and other protective molecules, whereas *A. carolinensis* shuts down metabolism and must rely on other mechanisms for thermal tolerance. The thermal reactionomes of these two species provide key insights into the genetic basis of thermal tolerance, and a resource for the future study of ecological adaptation in ant species. Finally, the reactionome itself illustrates a new direction for characterizing acclimation and adaptation in a changing climate. 


# Material and Methods

## Samples

...ants...Temperature is a potentially strong selective force for ground-nesting ant populations, which must tolerate seasonally freezing winters and hot summers. On shorter time scales, individual workers can experience extreme thermal environments when they leave the thermally buffered ant nest to forage for food [@diamond2012]. Ants of the genus *Aphaenogaster* are some of the most abundant in eastern North America [@lubertazzi2012], and species as well as populations within species differ in the critical maximum and minimum temperatures [@warren2013]. 

We used colonies of the ant species *Aphaenogaster picea*, collected from Molly Bog, Vermont (University of Vermont Natural Areas; 44.508°N, -72.702°W) and *Aphaenogaster carolinensis*, collected from Durham, North Carolina (36.037° N -78.874° W). The phylogeny of *Aphaeogaster* is currently unresolved [@umphrey1996], but putative species identification was possible based on morphology (Bernice Bacon DeMarco, pers. communication). Colonies of these species were maintained in the lab for 6 months prior to experimentation. 

From each species, we haphazardly collected 12 ants at the same time on 12 days. Each day, the 12 ants were placed in glass tubes into a water bath at one of 12 randomly-assigned temperatures, every 3.5°C between 0° and 38.5°C, for one hour. The minimum and maximum temperatures were selected based on previous work showing that these temperatures are close to the critical minimum and maximum temperatures for *Aphaenogaster* [@warren2013]. At the end of the hour, the ants were flash frozen in liquid nitrogen and stored at -80°C. We extracted mRNA by homogenizing three pooled ants in 500 uL of RNAzol buffer with zirconium silicate beads in a Bullet Blender (Next Advance; Averill Park, NY), followed by RNAzol extraction (Molecular Research Center Inc; Cincinnati, OH) and then an RNeasy micro extraction (Qiagen Inc; Valencia, CA) following the manufacturer's instructions. 

## Sequencing, assembly and annotation

For each species, the 12 samples were barcoded and sequenced in a single lane of 2 x 100bp paired-end reads on an Illumina HiSeq 1500 yielding 200 and 160 million reads for the *A. picea* and *A. carolinensis* samples respectively. Reads were filtered to remove Illumina adapter sequences and low quality bases using the program Trimmomatic [@lohse2012]. 

We assembled the sequenced reads into the full set of mRNA transcripts, the transcriptome, for the combined data set from both species using the Trinity *de novo* transcriptome assembly program [@grabherr2011]. *De novo* transcriptome assembly is prone to falsely identifying alternative transcripts and identifying inaccurate transcripts that are chimeric (e.g. regions of two separate transcripts that assemble into a false, or chimeric, third transcript) [@yang2013]. We removed potentially false transcripts by first running the program `CAP3` [@huang1999] to cluster sequences with greater than 90% similarity and merge transcripts with overlaps longer than 100 bp and 98% similar in length. Second, we ran the program `uclust` which clusters sequences completely contained within longer sequences at greater than 90% similarity (see Supplemental Information for exact parameters). We used liberal values (90% similarity) to merge orthologous transcripts in the two species that may not have assembled together in the initial *de novo* transcriptome assembly. To identify contaminant sequences, we screened our full transcriptome using the program `DeconSeq` [@schmieder2011] with the provided bacteria, virus, archae and human databases of contaminants. 

The Trinity *de novo* transcriptome assembly for both species assembled together included 126,172 genes with a total length of 100 million bp. Filtering to remove redundant or chimeric reads resulted in an assembly with 105,536 genes. The total length was 63 million bp with an N~50~ length of 895 bp and a mean gene size of 593 bp. Of the 105,536 filtered genes, 55,432 had hits to the NCBI-nr database. Of these, 38,711 genes mapped to GO terms, 1,659 genes were identified to an enzyme and 18,935 genes mapped to a domain with >50% coverage. We removed 5,675 genes identified as known contaminants, leaving 99,861 clean transcripts. Of these, 47% have best hits to genes from the class Insecta [(Supplementary Krona plot)](http://johnstantongeddes.org/assets/files/ApTranscriptome_FA.html).

To determine the putative function of the transcripts, we used functional annotation of the transcriptome assembly using the web-based tool `FastAnnotator` [@chen2012] which annotates and classifies genes by Gene Ontology (GO) term assignment, enzyme identification and domain identification. 

We used subsequent analyses at the transcript level due to limitations of working with a system lacking a reference genome, but refer to these transcripts as genes for simplicity.

## Thermally-responsive genes

We quantified expression of each gene using the program `Sailfish` [@patro2014] and use the bias-corrected transcripts per million (TPM) [@wagner2012] as our measure of gene expression. We included the contaminant genes identified by DeconSeq at the quantification stage to avoid incorrectly assigning reads to other genes, but removed these from further analyses. As preliminary examination of the data (see supplemental...) indicated that 7°C samples may have been mis-labeled, we omitted these data from the analysis. The expression values were highly correlated between species at each temperature treatment (r^2^ > 0.98) indicating that assembling the transcriptome with data from both species was justified (Supplemental).

To identify genes that have significant changes in expression across the thermal gradient, we fit the regression model

$$ log(TPM + 1) = \beta_0 + \beta_1(species) + \beta_2(temp) + \beta_3(temp^2) + \beta_4(species * temp) + \beta_5(species * temp^2) + \epsilon $$
    
to each gene. We used $log(TPM + 1)$ as the response to control for skew in the expression data. For a continuous predictor such as temperature, this regression approach is preferred to an ANOVA approach as it can reveal non-linear responses such as hump-shaped or threshold effects even [@cottingham2005]. This method is robust to over-dispersion as we expect errors in the read count distribution [@anders2010] to be independent with respect to temperature. To correct for multiple testing, we calculated adjusted *P* values using the False Discovery Rate (FDR) approach of Benjamini and Hochberg [-@benjamini1995], retaining genes under the 5% FDR threshold.

We identified thermally-responsive genes as the subset under our FDR threshold that had significant $\beta_2(temp)$, $\beta_3(temp^2)$, $\beta_4(species * temp)$ or $\beta_5(species * temp^2)$ terms after step-wise model selection by AIC. All analyses were performed in R [@r_core_team2013] and are fully reproducible (Supplemental file). 

For each thermally-responsive gene, we predicted expression levels across the tested thermal range using the final linear model separately for each species. We used the predicted gene expression values to partition the thermally-responsive genes into five expression categories for each species; genes that had greatest expression at high (> 31°) temperatures (**High**), low (< 10°C) temperatures (**Low**), intermediate (10 - 30°C) temperatures (**Intermediate**), both high and low temperatures (**Bimodal**) or were not thermally-responsive in that species (*NotResp*). For the **Bimodal** group, we required that expression at both low and high temperatures was at least one standard deviation greater than the expression at the mean temperature of 19.5°C.

## Species-level comparisons

We performed comparative analyses between the two species focusing on the genes that had a significant species x temperature interaction to determine if the two species differed in the tempertaure of gene activation and sensitivity of the thermally-responsive genes. For each gene, we predicted expression levels across temperatures for each species using the full linear model, and then grouped genes into the four responsive categories, **High**, **Low**, **Intermediate**, and **Bimodal**, as well as a fifth category **Not Responsive**. To test whether the temperature at which thermally-responsive genes are activated differs between species, we identified the temperature at which there was the greatest change in expression for each gene in each species. We grouped the **High** genes along with the high temperature end of the **Bimodal** genes, and did the same for **Low** genes. We then performed a *t*-test to determine if the mean temperature of gene activation differed between the two species for each group. 

To determine if the sensitivity of genes to thermal extremes differed between the species, we used the subset of **Intermediate** and **Bimodal** genes in each species. For each **Intermediate** gene in each species, we randomly sampled 1,000 temperature values weighted by the expression function (e.g. point of maximum expression had highest probability of being sampled). We then calculated the standard deviation of this random draw, and performed a *t*-test comparing the standard deviations of expression between species. Finding a greater standard deviation of expression would be indicative of less sensitivity to changes in temperature. For the **Bimodal** genes, we performed the same analysis as with the **Intermediate** genes, but with a greater standard deviation being indicative of a more sensitive gene (convex reaction norm) while a low standard deviation being a gene with greater inertia of expression in response to temperature changes (nearly flat).Thus, finding one species had overall greater standard deviation of expression for **Bimodal** genes would be indicative of greater thermal sensitivity for that species. 

To test for genetic assimilation, we compared basal expression levels (expression at 19.5° C) for the subset of genes that were responsive to cold temperatures (**Low** genes) in the southern *A. carolinensis* species, or hot temperatures (**High** genes) in the northern *A. picea* species using non-parametric two-sided Wilcoxon tests.

## Gene set enrichment analysis

To describe the overall molecular toolkit for thermal tolerance in *Aphaenogaster*, we performed gene set enrichment analysis (GSEA) on the intersection of each expression category in the two species using the package `topGO` [@alexa2006; @alexa2010] with the `parentChild` algorithm [@grossmann2007] implemented in R [@r_core_team2013]. Briefly, this approach identifies GO terms that are overrepresented in the significant genes relative to all GO terms in the transcriptome, accounting for dependencies among the GO terms. We clustered enriched GO terms from the `topGO` analysis by semantic similarity using the package `GOSemSim` [@yu2010] to reduce redundancy. 

## Data availability

The raw Illumina reads are available at [...], the script for analysis and version history is available at (http://github.com/johnstantongeddes/ApTranscriptome).  



# Figure Legends




# References

