---
title: Thermal reactionomes reveal divergent responses to thermal extremes in warm
  and cool-climate ant species
author: ' '
csl: bmcgenomics.csl
output:
  pdf_document:
    fig_caption: yes
    includes:
      in_header: header.tex
    keep_tex: yes
    latex_engine: xelatex
    number_sections: no
    toc_depth: 2
  html_document:
    keep_md: yes
  word_document: default
includes: header.tex
bibliography: ApTranscriptome.bib
---

\linenumbers

\newpage

**Authors** 

John Stanton-Geddes \textsuperscript{1}, Andrew Nguyen \textsuperscript{1}, Lacy Chick \textsuperscript{2}, James Vincent \textsuperscript{3}, Mahesh Vangala \textsuperscript{3}, Robert R. Dunn \textsuperscript{4}, Aaron M. Ellison \textsuperscript{5}, Nathan J. Sanders \textsuperscript{2,6}, Nicholas J. Gotelli \textsuperscript{1}, Sara Helms Cahan \textsuperscript{1}
           
\vspace{10 mm}           
           
**Affiliations**

\textsuperscript{1} Department of Biology, University of Vermont, Burlington, Vermont 05405 \newline

\textsuperscript{2} Department of Ecology and Evolutionary Biology, University of Tennessee, Knoxville, Tenessee 37996 \newline

\textsuperscript{3} Vermont Genetics Network, University of Vermont, Burlington, Vermont 05405 \newline

\textsuperscript{4} Department of Biological Sciences, North Carolina State University, Raleigh, North Carolina 27695 \newline

\textsuperscript{5} Harvard Forest, Harvard University, Petersham, Massachusetts 01336 \newline

\textsuperscript{6} Center for Macroecology, Evolution and Climate, University of Copenhagen, Universitetsparken 15, DK-2100 Copenhagen 

\vspace{10 mm}

**Corresponding author**

John Stanton-Geddes, Data Scientist

Dealer.com, 1 Howard St, Burlington, VT 05401 

Phone: 802-656-2922

Email: john.stantongeddes.research@google.com

```{r load, echo=FALSE, results='hide', message=FALSE, warning=FALSE}
#options(scipen = 1, digits = 3)
load("ms_2015-10-22.RData")
library(data.table)
library(ggplot2)
library(reshape2)
library(pander)
library(RColorBrewer)
library(topGO)
library(plyr)

options(digits = 3)
panderOptions('round', 3)
panderOptions('keep.trailing.zeros', TRUE)
```

\newpage

# Abstract 

**Background** The distributions of species and their responses to climate change are in part determined by their thermal tolerances. However, little is known about how thermal tolerance evolves. To test whether evolutionary extension of thermal limits is accomplished through enhanced cellular stress response (*enhanced response*), constitutively elevated expression of protective genes (*genetic assimilation*) or a shift from damage resistance to passive mechanisms of thermal stability (*tolerance*), we conducted an analysis of the *reactionome*: the reaction norm for all genes in an organism's transcriptome measured across an experimental gradient. We characterized thermal reactionomes of two common ant species in the eastern U.S, the northern cool-climate *Aphaenogaster picea* and the southern warm-climate *Aphaenogaster carolinensis*, across 12 temperatures that spanned their entire thermal breadth. 

**Results** We found that at least 2% of all genes changed expression with temperature. The majority of upregulation was specific to exposure to low temperatures. The cool-adapted *A. picea* induced expression of more genes in response to extreme temperatures than did *A. carolinensis*, consistent with the *enhanced response* hypothesis. In contrast, under high temperatures the warm-adapted *A. carolinensis* downregulated many of the genes upregulated in *A. picea*, and required more extreme temperatures to induce down-regulation in gene expression, consistent with the *tolerance* hypothesis. We found no evidence for a trade-off between constitutive and inducible gene expression as predicted by the *genetic assimilation hypothesis*. 

**Conclusions** These results suggest that increases in upper thermal limits may require an evolutionary shift in response mechanism away from damage repair toward tolerance and prevention. 

**Keywords**

*Aphaenogaster*, gene expression, plasticity, reactionome, transcriptome


# Background

Temperature regulates biological activity and shapes diversity from molecular to macroecological scales [@brown2004; @kingsolver2009]. Many species, especially small-bodied arthropods, live at temperatures close to their thermal limits and are at risk from current increases in temperature [@deutsch2008; @kingsolver2013; @sunday2014]. Thermal tolerance, the ability of individuals to maintain function and survive thermal extremes, depends on a complex interplay between the structural integrity of cellular components and activation of physiological response mechanisms to prevent and/or repair damage [@huey1989; @richter2010]. Thermal defense strategies are shaped by the environmental regime organisms experience [@angilletta2003] and thermal limits vary considerably among species and populations [@cowles1939; @deutsch2008; @kingsolver2013; @hoffmann2013]. These differences in thermal tolerance are largely genetic [@krebs1997; @kellermann2012] with a highly polygenic basis [@krebs1998; @williams2012; @morgan2006; @takahashi2011]. Outside of candidate genes [@krebs1998], little is known about the evolution of thermal tolerance or the link between short-term physiological acclimation and longer-term adaptation to novel temperature regimes. This information is critical for understanding the adaptive potential of species to future climates [@hoffmann2008].

To address this gap of knowledge, we need information on the extent to which selection has acted upon the diversity and plasticity of genes involved in thermal tolerance [@hoffmann2008; @somero2011]. In recent years, whole-organism gene expression approaches (e.g. transcriptomics) using high-throughput RNA sequencing (RNAseq) technology have been widely applied to identify genes involved in thermal tolerance [@meyer2011; @teets2012; @barshis2013; @oneil2014] and other traits. Such studies typically use an ANOVA-type experimental or sampling design, with only a few environmental levels, and often find only a few dozen to hundred genes with differential expression in different thermal regimes. However, temperature and other environmental factors vary continuously in nature. As a result, such categorical comparisons (e.g. high vs. low temperatures) are likely to miss key differences that are due not just to whether it is hot, but rather how hot it is. Continuous variation is better characterized with a reaction norm approach, which describes variation in the phenotype of a single genotype across an environmental gradient [@gomulkiewicz1992]. Reaction norms differ not only in mean values, but also in their shapes [@hoffmann2013; @murren2014], and differences in the shape of reaction norms are often larger than differences in mean values at both the species and the population level [@murren2014]. 

In this study, we extend the reaction norm approach to RNAseq analysis and introduce the *reactionome*, which we define as a characterization of the reaction norm for all genes in an organism's transcriptome across an environmental gradient. Although temporal patterns of transcriptional activity (e.g. fast- vs. slow- responding genes) are also important components of an organism's transcriptional response to environmental conditions [@sorensen2005], we focus here on the response of transcripts across conditions at the same time point. 

We used the reactionome method to identify genes that are thermally responsive in two closely-related eastern North American ant species, *Aphaenogaster carolinensis* and *A. picea* [@umphrey1996; @demarco2015]. *Aphaenogaster* are some of the most common ants in eastern North America [@king2013] where they are keystone seed dispersers [@ness2009; @zelikova2011; @rodriguez-cabal2012]. Ants, and ectotherms in general, have little or no thermal safety margin [@sunday2014] and thus are highly susceptible to climate change [@diamond2012; @kingsolver2013], putting at risk important ecosystem services [@toro2012]. Growth chamber studies have demonstrated that reproduction of *Aphaenogaster* will be compromised by increased tempreatures [@pelini2012], while field studies [@diamond2012] and simulations [@bewick2014] indicate that ant species persistence will depend on combinations of physiology and species interactions. *Aphaenogaster carolinensis* experiences a higher mean annual temperature (MAT) (14.6°C) and less seasonal temperature variation (temperature seasonality = 7,678°) than does *A. picea* (MAT = 4.6°C, seasonality = 10,008°; [@hijmans2005]) at their respective collection sites. In controlled laboratory experiments, these warm- and cold-climate species exhibit corresponding differences in their critical maximum (44.7°C for *A. carolinensis* versus 41.3°C for *A. picea*; see Methods) and minimum temperatures (6.1°C for *A. carolinensis* versus -0.1°C for *A. picea*). These differences between species in their thermal environments and physiological tolerances allowed us to investigate adaptation to both lower and upper thermal extremes in this system.


```{r climate, echo=FALSE, results='hide', message=FALSE, warning=FALSE}
library(raster)
w <- getData('worldclim', var='bio', res=10)
#names(w)
# Plot first bioclim dimension (MAT) to check download worked
#plot(w, 1, col=colorRampPalette(c("blue","green","red"))(255))
# sites
sites <- data.frame(site = c("Ac", "Ap"), lon=c(-78.874, -72.702), lat= c(36.037, 44.508))
# extract bioclimatic variables from worldclim data. Note bioclim temperature data is multiplied by 10 (no decimal), so transform by dividing by 10.
dbio <- extract(w, sites[,c("lon","lat")])
# bio1 = MAT
# bio4 = Temperature Seasonality (standard deviation *100)
dbio.vars <- cbind(sites, dbio[,1], dbio[,4])
colnames(dbio.vars)[which(colnames(dbio.vars) == "dbio[, 1]")] <- "MATx10"
dbio.vars$MAT <- dbio.vars$MATx10/10
colnames(dbio.vars)[which(colnames(dbio.vars) == "dbio[, 4]")] <- "Seasonality"
dbio.vars
```


To characterize the thermal reactionome, we measured the reaction norm for each gene using a regression-based statistical approach to identify temperature-dependent patterns of change in gene expression. We used these response patterns to quantitatively test three mechanistic hypotheses of thermal adaptation. First, the *enhanced response hypothesis* [@hofmann1996; @feder1999; @kultz2005] proposes that species extend their thermal limits through a stronger induced response to provide greater protection from more frequently encountered stressors. This hypothesis would predict that the cool-adapted *A. picea* would activate more genes, and induce them more strongly, in response to low temperatures than would the warm-adapted *A. carolinensis*, which would show greater induction in response to high temperatures.

Second, the *tolerance hypothesis* [@cowles1939; @fields2001] proposes that existing inducible stress responses become insufficient or prohibitively costly as environmental stressors increase in frequency, resulting in a shift away from an induced response in favor of structural changes [@lockwood2012] or behavioral adaptations [@gordon2013; @sunday2014]. This hypothesis predicts adaptation to stress should be associated with lower transcriptional responsiveness and less sensitivity to temperature perturbation, as well as a shift to an alternate suite of tolerance genes and pathways [@neelakanta2012; @franssen2011]. 

Finally, the *genetic assimilation hypothesis* [@waddington1953; @sikkink2014] proposes that exposure to more extreme stressors selects for a shift from inducible to constitutive expression of stress-response genes. This hypothesis predicts that transcripts responsive to high temperatures in *A. picea* will have higher constitutive expression in *A. carolinensis*, whereas transcripts responsive to low temperatures in *A. carolinensis* will have higher constitutive expression in *A. picea*. 

To summarise, in this project we generated the transcriptomes of two closely-related temperate ant species, and quantified their gene expression across a wide range of thermal conditions. We then evaluated three non-mutually exclusive hypotheses (enhanced response, tolerance and genetic assimilation) of the evolution of thermal adaptation by comparing the number and expression patterns of transcripts between species in response to extreme low and extreme high temperatures. Finally, we used gene ontology information to determine which gene products and pathways are involved in thermal adaptation in the two species.


# Results

## Reaction norms of thermally-responsive transcripts

The combined *Aphaenogaster* transcriptome assembly contained `r prettyNum(nrow(annotation.table), big.mark =",")` transcripts. About half of these (`r prettyNum(nrow(annotation.table.responsive), big.mark =",")`) transcripts had a signficant BLAST hit, of which 50% (25,797) had a top hit to Insecta and 37% (18,854) had a top hit to Formicidae. We performed a BUSCO analysis [@simao2015] to assess the quality of the transcriptome assembly against the arthropod Benchmarking Universal Single-Copy Orthologs (BUSCOs). This analysis revealed that transcriptome is largely complete, as we recovered 1,426 complete single-copy BUSCOs (62%) and an additional 435 fragmented BUSCOs (16%), which is in line with results of Simao et al. [-@simao2015] for transcriptomes of other non-model species. Moreover, only 8% of the BUSCOs were found to be duplicated in the transcriptome, which indicates that the steps (see Methods) we took to collapse homologs in the combined transcriptome of the two species were successful. 

We quantified gene expression using the program `Sailfish` [@patro2014], and fitted polynomial regression models to the expression values of each transcript to identify those that had a linear or quadratic relationship (Fig. 1). To account for multiple tests, we both applied a False Discovery Rate (FDR) correction, and performed a resampling analysis to determine the number of transcripts that would be expected to have a significant relationship by chance alone. We retained the `r prettyNum(nrow(pre.signif.transcripts) - quantile(D$num_signif, probs = 0.95), big.mark = ",")` (`r round((nrow(pre.signif.transcripts) - quantile(D$num_signif, probs = 0.95)) / nrow(annotation.table), 3) * 100`% of total) transcripts that exceeded the null expectation from the resampling analysis as true positive transcripts for further analyses (Table S1). Of these transcripts, `r round(sigtable[which(sigtable$Coefficient == "Temperature:Colony"), "Number.significant"] / sum(sigtable[sigtable$Coefficient %in% c("Temperature", "Temperature:Colony"), "Number.significant"]), 2) *100`% (`r prettyNum(sigtable[which(sigtable$Coefficient == "Temperature:Colony"), "Number.significant"], big.mark=",")`) had a non-linear relationship with temperature that would likely have been missed with a standard differential expression experiment (e.g. high vs. low temperature). The proportion of responsive transcripts is similar if we focus only on those transcript with a BLAST hit (`r prettyNum(responsive_transcripts_w_BLAST, big.mark = ",")` significant transcripts out of `r prettyNum(nrow(annotation.table.responsive), big.mark = ",")`, `r round(responsive_transcripts_w_BLAST / nrow(annotation.table.responsive), digits = 3)*100`). However, as with all *de novo* transcriptome assemblies, this assembly is fragmented due to partial contigs and alternative transcripts [@vijay2013] so this estimate is likely a lower bound for the true proportion of transcripts that are thermally responsive. 

We used the predicted transcript expression levels to partition transcripts for each species into five expression categories (Fig. 1) which were defined *a priori* to allow us to test predictions derived from three thermal adaptation hypotheses of relative response severity in the two species: **High** transcripts had greatest expression at temperatures > 31°C, **Low** transcripts had greatest expression at temperatures < 10°C, **Intermediate** transcripts had greatest expression between 10 to 30°C, **Bimodal** transcripts had increased expression at both high and low temperatures, while **NotResp** transcripts were those that were not thermally responsive in the focal species but did respond in the other.

## Expression response to thermal extremes differs between species

Although the total number of thermally-responsive transcripts did not differ between species ($\chi^2$~`r chi2$parameter`~ = `r round(chi2$statistic, 2)`, *P* = `r round(chi2$p.value, 2)`), the two species differed in the number of transcripts in each expression category (Table 1, $\chi^2$~`r chi1$parameter`~ = `r chi1$statistic`, *P* < `r max(chi1$p.value, 0.001)`). *Aphaenogaster picea* induced significantly more transcripts in response to both temperature extremes (**Bimodal** transcripts in Table 1; $\chi^2$~`r chi6$parameter`~ = `r chi6$statistic`, *P* < `r max(chi6$p.value, 0.001)`) than did *A. carolinensis*, which downregulated more transcripts under these conditions (**Intermediate** transcripts in Table 1; $\chi^2$~`r chi4$parameter`~ = `r chi4$statistic`, *P* < `r max(chi4$p.value, 0.001)`). Consistent with the *enhanced response* hypothesis, the cool-climate *A. picea* induced `r Ap.type.table["A. picea","Low"] - Ap.type.table["A. carolinensis","Low"]` (~50%) more transcripts in response to low temperatures than the warm-climate *A. carolinensis* (**Low** transcripts in Table 1; $\chi^2$~`r chi3$parameter`~ = `r chi3$statistic`, *P* < `r max(chi3$p.value, 0.001)`). However, there was no difference among species in the number of transcripts upregulated at high temperatures (**High** transcripts in Table 1; $\chi^2$~`r chi5$parameter`~ = `r chi5$statistic`, *P* = `r round(chi5$p.value, 2)`). 

```{r echo=FALSE, results='hide'}
# results of Stuart-Maxwell test
mh.test
```

In addition, we also examined the specific patterns of shifts from one expression category to another between species. As transcripts may change expression between species due to neutral drift alone, we used the Stuart-Maxwell test of marginal homogeneity to test if the number of responsive transcripts in each expression category differed between the species when controlling for overall differences in the number of responsive transcripts. We found that the expression categories of individual transcripts between the two species were not randomly distributed (Stuart-Maxwell test of marginal homogeneity $\chi^2$~4~ = 319, P < 0.001, Fig. S1). Specifically, the two species differed significantly in expression pattern, which captures differences in slope as well as category, for `r prettyNum(sigtable[which(sigtable$Coefficient=="Temperature:Colony"), "Number.significant"], big.mark = ",")` (`r round(sigtable[which(sigtable$Coefficient=="Temperature:Colony"), "Number.significant"] / sigtable[which(sigtable$Coefficient=="Total"), "Number.significant"], 2) * 100`%) of the thermally responsive transcripts.

The *enhanced response* and *tolerance* hypotheses make opposing predictions concerning the overlap in response patterns between the two species (Fig. 2). The *enhanced response hypothesis* posits that temperature adaptation uses existing mechanisms for thermal resistance, which should result in significant overlap in response and fewer transcripts shifting expression categories than expected by chance (Fig. 2, left). In contrast, the *tolerance hypothesis* predicts that transcripts involved in active defense will become non-responsive or shift to other expression categories in the better-adapted species (Fig. 2, right). We tested these predictions by examining if the transcripts upregulated in response to the temperature extreme experienced less frequently by a species (cool temperatures for the warm-climate *A. carolinensis*, and warm temperatures for the cool-climate *A. picea*) displayed the same response profile in the other species that more frequently experiences those conditions.

Transcripts upregulated at low temperatures in *A. carolinensis* (**Low** and **Bimodal** transcripts) were significantly biased toward this same category and away from other expression categories in *A. picea* (Fig. 3A), suggesting shared response pathways as predicted by the *enhanced response* hypothesis. In contrast, transcripts upregulated in response to high temperatures in *A. picea* (**High** and **Bimodal**) shifted expression categories in *A. carolinensis* (Fig. 3B), primarily to the **Intermediate** category (Fig. 3B). These transcripts are less likely to be upregulated in any context, consistent with the *tolerance hypothesis*.

## Molecular processes suggest a generalized stress response mechanism

The gene set enrichment analysis revealed a number of gene groups enriched in each expression category (Table S2). Across both species, there were 9 terms enriched in the **Bimodal** category, including terms involved in stress response (regulation of cellular response to stress, signal transduction by p53 class mediator), cell death (apoptotic signaling pathway) and cellular organization (e.g. protein complex localization). The 6 terms enriched in the **Low** category suggest that proteins undergo structural (e.g protein acylation) and organizational (single-organism organelle organization) changes to tolerate colder temperatures, possibly to maintain membrane fluidity [@ohtsu1998]. The **High** category included only a single enriched GO term, "nicotinamide metabolic process", while the **Intermediate** category had 5 terms including DNA packaging and metabolic process terms. 

## *A. carolinensis* has greater inertia of expression change to increases in temperature than does *A. picea*

As an additional test of the *tolerance hypothesis*, we examined the critical temperature of gene induction in response to increasing and decreasing temperatures. We compared between species the mean temperatures of transcript upregulation, defined as the temperature at which the transcript showed the greatest positive change in expression. In support of the *enhanced response* but not the *tolerance hypothesis*, the temperature of induction at low temperatures was significantly higher for the cool-climate *A. picea* than for *A. carolinensis* (`r round(T_on.low.ttest$estimate[1], 1)`°C) than *A. picea* (`r round(T_on.low.ttest$estimate[2], 1)`°C; *t*~`r prettyNum(T_on.low.ttest$parameter, digits=2)`~ = `r round(T_on.low.ttest$statistic, 1)`, *P* < `r round(max(T_on.low.ttest$p.value, 0.001), 3)`; Fig. 4A), though the temperature of induction did not differ between species for high temperatures (*t*~`r round(T_on.high.ttest$parameter, 0)`~ = `r round(T_on.high.ttest$statistic, 1)`, *P* < `r max(T_on.high.ttest$p.value, 0.001)`). 

Similarly, for down-regulated (**Intermediate**) transcripts, we compared the mean temperatures of downregulation of transcript expression between species at both high (> 20°C) and low (< 20°C) temperatures. Consistent with the *tolerance hypothesis*, *A. carolinensis* had greater inertia of gene expression in response to increasing temperatures. The temperature of downregulation for **Intermediate** transcripts was `r round(T_off.int.high.ttest$estimate[1], 1)`°C for *A. carolinensis* compared to `r round(T_off.int.high.ttest$estimate[2], 1)` for *A. picea* (*t*~`r round(T_off.int.high.ttest$parameter, 0)`~ = `r round(T_off.int.high.ttest$statistic, 1)`, *P* < `r round(max(T_off.int.high.ttest$p.value, 0.001), 3)`). The difference between species was not significant with decreasing temperatures (*t*~`r round(T_off.int.low.ttest$parameter, 0)`~ = `r round(T_off.int.low.ttest$statistic, 1)`, *P* = `r round(max(T_off.int.low.ttest$p.value, 0.001), 3)`, Fig. 4B).


## No evidence for *genetic assimilation*
 
We tested the *genetic assimilation hypothesis* by comparing the log ratios of relative inducibility to relative baseline expression at the rearing temperature (25°C). If stress-response transcripts have shifted between species from inducible to constitutive expression, there should be a negative relationship between the two. We found no evidence of such a relationship for either temperature extreme: transcripts more upregulated at high temperatures in the cool-climate *A. picea* were not expressed at higher baseline levels in the warm-climate *A. carolinensis* (Fig. 5A). Similarly, transcripts more upregulated at low temperatures in *A. carolinensis* did not show higher baseline levels in *A. picea* (Fig. 5B). In fact, for both comparisons we found a weakly positive relationship between relative inducibility and baseline expression between the two species ($\beta_1$ = 0.31, *P* < 0.001 and ($\beta_1$ = 0.21, *P* < 0.001). In addition, the thermally responsive transcripts in *A. carolinensis*, regardless of expression pattern, had higher baseline expression than those in *A. picea*, including those with **Intermediate** expression profiles in both species (Wilcoxon V = `r prettyNum(w3$statistic)`, *P* < `r max(w3$p.value, 0.001)`). An important exception to this pattern is the set of transcripts that had **High** or **Bimodal** expression in *A. picea* but were not thermally responsive in *A. carolinensis* (top-row of Fig. 3B). These transcripts are less likely to be upregulated in any context, consistent with the *tolerance hypothesis*.



# Discussion

The potential for many species to persist in face of climate change will depend in part upon their thermal tolerances. However, for most species little is known about how plasticity or adaptive changes in gene expression underlie thermal tolerance. By using a *reactionome* approach, we were able to quantitatively describe plasticity in transcript expression across a thermal gradient, and identify putative changes in gene expression associated with shifts in thermal tolerance between the ant species *Aphaenogaster picea* and *A. carolinensis*. We found non-linear patterns of gene expression changes in response to temperature, with both quantitative and qualitative differences between species, consistent with different mechanisms of thermal adaptation to low and high temperature extremes. 

Under the *enhanced response* hypothesis, stress-adapted species are hypothesized to induce a stronger and earlier response to extreme conditions. We found evidence for this hypothesis at low temperatures: although the lower thermal limit for *A. picea* is substantially lower than *A. carolinensis*, *A. picea* upregulated responsive transcripts at slightly less extreme temperatures (Fig. 4A). Moreover, the transcripts upregulated in *A. picea* included about half (`r round(tt2["Low", "Low"] / colSums(tt2)["Low"], 2)*100`%) those upregulated in *A. carolinensis* as well as an additional set of `r round(rowSums(tt2)["Low"] - tt2["Low", "Low"], 2)` transcripts (Table 1), enriched for metabolism, organization and translation processes (Table S2). Two non-mutually exclusive hypotheses may explain this pattern. First, surviving prolonged low temperatures, such as would be experienced during overwintering, generally requires advance production of specialized cryoprotectants [@neelakanta2012] and a suite of preparatory physiological modifications [@denlinger2002]. The northern species *A. picea* may induce a greater response to survive the longer winter period. Alternatively, the response to low temperatures may reflect countergradient expression to counteract reduction in enzyme efficiency, and maintain activity as temperature declines [@lockwood2012]. This requirement may be under stronger selection in *A. picea* given the shorter growing season that would necessitate foraging under a broader range of temperatures. 

In contrast to cold tolerance, the enhanced upper thermal limit in *A. carolinensis* is best explained by the *tolerance* hypothesis. High temperatures were associated with significantly fewer upregulated transcripts in *A. carolinensis* (Table 1), and a large proportion (`r round(sum(tt2[c("Intermediate", "NotResp"), "High"]) / Ap.type.table["A. picea", "High"], 2) * 100`%) of the transcripts upregulated at high temperatures in *A. picea* were either downregulated or expressed at negligible levels overall in *A. carolinensis*. These results suggest that mechanisms other than the heat shock response are acting to maintain protein stability in face of temperature increases. Such mechanisms may include novel constitutive defenses [@barshis2013; @meyer2011; @oneil2014], enhanced proteome stability [@ghosh2010] or behavioral quiescence [@sunday2014] to tolerate thermal stress. These differences are in line with expectations that *A. carolinensis*, with a growing season over twice the length of its northern congener, may be better able to afford to restrain from foraging in suboptimal conditions. Indeed, quiescence under stressful conditions by the red harvester ant *Pogonomyrmex barbatus* has been shown to increase colony fitness [@gordon2013].

The one hypothesis that did not receive support was the *genetic assimilation hypothesis*, which predicts that exposure to more frequent stressors will select for a shift from inducible to constitutive expression of stress-response transcripts. This constrasts with other recent studies on adaptation in field populations to thermal stress [@barshis2013]. However, in a short-term selection experiment for heat tolerance, Sikkink et al. [-@sikkink2014] also found no evidence for genetic assimilation at the expression level after 10 generations of selection for heat tolerance in *Caenorhabditis remanei*, even though there was a substantial increase in heat tolerance. Both the genetic assimilation and tolerance routes to increasing thermal limits are functionally similar in that they emphasize damage prevention rather than repair. Whether a particular taxon evolves one strategy over another may be related to availability of alternative mechanisms as well as the intensity, frequency and duration of temperature stress in a given environment. 

Given the differences in the patterns of thermal responsiveness between species (Fig. 3), it is worth noting a number of similarities. In both species, there were 2 -- 3 times more transcripts upregulated at low than high temperatures (Table 1). The degree of upregulation at low temperatures is surprising given previous studies [@teets2012-1; @vesala2012] that found little transcriptional activity at low temperatures. However, these studies exposed organisms to a few extreme (-10 -- 0°C) temperatures. At these extremes, we also found few upregulated transcripts (Fig. 4A), whereas the peak of low-temperature transcriptional activation occurred near 10°C (Fig. 4). A potential explanation for this pattern is that increased gene expression functions to support elevated metabolism at moderately cold temperatures, as suggested by the metabolic cold adaptation hypothesis [@addo-bediako2002]. The observation that more transcripts were upregulated at low than high temperatures could also be due to stronger selection on upper than lower thermal limits, thereby reducing both genetic variation and gene expression plasticity at high temperatures [@kingsolver2013; @kelly2013]. This explanation is consistent with the observation in *Aphaenogaster rudis* [@warren2013] and other ectotherms [@addo-bediako2000; @hoffmann2013] that critical maximum temperatures vary less among taxa than do critical minimum temperatures.

Critical maximum and minimum temperatures are hypothesized to be genetically correlated [@addo-bediako2000; @hoffmann2013], but this was not evident in terms of gene expression in this study. Only ~10% of transcripts upregulated in response to temperature were bimodal, and for both activation and down-regulation, thresholds differed between species at only one temperature extreme (Fig. 4). This suggests that species do not face a fundamental trade-off between these two limits and may be able to shift upper and lower thermal limits independently to match requirements of more seasonally variable environments. 
A major contribution of this study is the construction of a reactionome for gene expression data. Similar approaches have been used in other species [@hodgins-davis2012; @aubin-horth2009], but to our knowledge, none have applied a regression approach to identify a complete list of responsive transcript across an environmental gradient. 
This approach revealed quantitative patterns of temperature response not captured in categorical comparisons. For example, the degree of upregulation at cool (~10°C) but not extreme cold temperatures was missed in previous studies that focused on extreme cold limits, as discussed above. Further, a number of issues have hampered RNA-seq studies to date. Namely, lists of differentially expressed transcripts are prone to false positives [@gonzalez2013], depend on the genetic background of the organism [@sarup2011] and are prone to "storytelling" interpretations [@pavlidis2012]. Our findings are robust to these issues as we focus on the average change in the shape of the reaction norms across many hundreds of responsive transcripts in each species. Although we use gene ontology information to interpret our results, the key findings about differential plasticity of expression between species do not depend on functional annotation.

Moreover, by characterizing responses across thousands of transcripts, the reactionome approach can help to distinguish selection from neutral drift in gene expression [@khaitovich2004; @whitehead2006; @ogasawara2009]. Although we cannot rule out drift as a source of variation for individual transcripts, we would not expect to see systematic differences in expression type categories or critical temperature thresholds as we do here (Fig. 3, Fig. S1). Thus, our method provides an example of how focusing on transcriptome-wide changes in gene expression -- as opposed to identifying lists of differentially-expressed transcripts -- can provide meaningful insight on the process of evolution. It should be noted, however, that although including non-linear relationships between expression and temperature captured a significantly larger range of biologically-relevant responses, it also led to a substantial increase in false positives. Empirical estimation of these rates via randomization tests, combined with robust sampling designs, can help to minimize this bias and focus results on biologically-meaningful gene sets.

A number of caveats do apply to our work. First, species may differ in gene expression along axes which we have not measured here, especially temporal patterns of gene expression [@sorensen2005], which could be studied in further work. Second, the *de novo* transcriptome assembly is highly fragmented, given that all sequenced ant genomes to date have only about 18,000 genes [@gadau2012]. Although we took steps to remove contaminants and redundant transcripts, some likely remain, in addition to partially assembled transcripts. A genome assembly, in progress, will help to reduce fragmentation. Third, the quality of the annotation for a non-model system such as *Aphaenogaster* is not as good as it would be for model arthropods such as *Drosophila* and *Apis*. Finally, the mapping of changes in gene expression to organismal fitness is far from direct [@feder2005], and large differences in patterns of gene expression may have only small effects on fitness. In particular, functional protein levels cannot be expected to be fully linked to mRNA abundance due to post-transcriptional modification, regulation, mRNA fluctuations and protein stability [@feder2005].  

Our results are congruent with evidence from other systems [@barshis2013] that thermally-stressful habitats select for investment in tolerance, whereas organisms from less stressful environments rely on plastically-induced resistance. Although the heat-shock response is one of the most conserved across living organisms [@kultz2005], it is energetically expensive, particularly under chronic stress conditions [@hoekstra2013].  Under such circumstances, it may be advantageous to proactively prevent thermal damage even at the cost of reduced metabolic efficiency, either by maintaining a higher constitutive level of chaperone proteins [@krebs1997] or by increasing the thermal stability of proteins at the expense of catalytic activity [@arnold2001].  Thus, although in the short term increasing temperature stress leads to a quantitatively stronger induced response, adapting to such stress over evolutionary time appears to require a qualitative shift in mechanism of resistance that can alter not only the magnitude, but the sign of gene expression change in response to temperature.  Whether such a shift would be possible in the compressed time frame of projected climate change, particularly for long-lived organisms such as ants, is likely to be critical in determining the capacity of populations to adapt to more frequent and long-lasting stressors. 

## Conclusions

In this work, we have brought reaction norms to the genomic era by characterizing the thermal reactionomes of two temperate ant species, *Aphaenogaster picea* and *A. carolinensis*. At least 2% of their transcriptomes are thermally responsive. Our results indicate that these two ant species have different responses to thermal extremes. *A. picea* responds by increasing expression of transcripts related to metabolism, stress response and other protective molecules, whereas *A. carolinensis* decreases expression of transcripts related to metabolism and likely relies on other mechanisms for thermal tolerance. The thermal reactionomes of these two species provide key insights into the genetic basis of thermal tolerance, and a resource for the future study of ecological adaptation in ant species. Finally, the reactionome itself illustrates a new direction for characterizing acclimation and adaptation in a changing climate. 


# Methods

## Samples

Ants of the genus *Aphaenogaster* are some of the most abundant in eastern North America [@lubertazzi2012], and species as well as populations within species differ in critical maximum and minimum temperatures [@warren2013]. Temperature is a potentially strong selective force for ground-nesting ant populations, which must tolerate seasonally freezing winters and hot summers. On shorter time scales, individual workers can experience extreme thermal environments when they leave the thermally buffered ant nest to forage for food [@diamond2012]. 

In fall 2012, we collected a single colony of *Aphaenogaster picea* from Molly Bog, Vermont (University of Vermont Natural Areas; 44.508° N, -72.702° W) and a single colony of *Aphaenogaster carolinensis*, part of the *A. rudis* species complex [@umphrey1996], from Durham, North Carolina (36.037° N -78.874° W). These sites are centrally located within each species' geographic range. Along the East Coast of the United State, the distribution of *A. picea* ranges from central Maine south to northern Pennsylvania, while *A. carolinensis* is found from Pennsylvania to the Carolinas. Species identity was confirmed with morphological characters (Bernice DeMarco, Michigan State University). Colonies of both species were maintained in common conditions at 25°C for 6 months prior to experimentation. Due to colony size limitations, we were unable to determine the critical thermal limits of these particular colonies. In summer 2013 we collected additional colonies of *Aphaenogaster* from Molly Bog, VT and North Carolina (Duke Forest, 36.036° N, 79.077° W). We tested the upper and lower critical thermal limits for 5 ants from each of these colonies using a ramp of 1° C per minute, starting at 30° C, and recorded the temperature at which the ants were no longer able to right themselves, following the protocol of Warren & Chick [-@warren2013]. 

```{r therm_limits, echo=FALSE, results='hide'}
therm.limits <- read.csv("../data/thermal_limits.csv")
avg.therm.limits <- ddply(therm.limits, .(Species), summarise,
                          CTmax = mean(CTmax),
                          CTmin = mean(CTmin))

therm.limits.table <- data.frame(Species = avg.therm.limits[,"Species"],
                    Lat = therm.limits[c(1,6), "Lat"],
                    Lon = therm.limits[c(1,6), "Lon"],
                    MAT = therm.limits[c(1,6), "MAT"],
                    CTmax = avg.therm.limits["CTmax"],
                    CTmin = avg.therm.limits["CTmin"])

therm.limits.table

#pandoc.table(therm.limits.table, style="rmarkdown", digits = 3, split.table = Inf,
#             caption = "Upper (CTmax) and lower (CTmin) critical thermal limits for 5 ants from colonies of *Aphaenogaster picea* from Molly Bog, Vermont and *Aphaenogaster rudis* from Duke Forest, North Carolina. *Aphaenogaster rudis* and *A. carolinensis* are in the same species-complex. Latitude (Lat), longitude (Long) and mean annual temperature (MAT) of each collection site are also given.")
```

## Common Garden Design

Ideally, genetically-based variation in gene expression profiles would be identified by comparing individuals completely reared under common-garden conditions to eliminate environmental variation experienced either as adults or during development. However, *Aphaenogaster* colonies are long-lived, cannot be bred under laboratory conditions, and do not achieve complete turnover of the workforce for at least a year or longer. Thus, as is commonly done with other long-lived organisms [@whitehead2006; @barshis2013], we exposed both colonies to common-garden rearing conditions for six months to fully acclimate adult workers to common temperatures. Over this time, roughly 1-2 cohorts of new workers are expected to join each colony (~1/3 of the total), such that the workers sampled for thermal traits and gene expression are likely to have included a mix of adult-acclimated and fully lab-reared individuals.  

Unlike ANOVA-based experimental designs, which derive statistical power from replication within each experimental treatment level, regression designs have greater power when sampling additional values across the range of the continuous predictor variable [@gotelli2012]. Ideally, the treatments should be replicated at each level of the predictor variable [@cottingham2005]. However, even with no replication, the regression design is still more powerful than an ANOVA design with comparable replication, and provides an unbiased estimator of the slope [@gotelli2012]. For these reasons, we focused our sequencing efforts on maximizing the number of temperatures at which the transcriptome was profiled, rather than on replication at each temperature. 

To limit differences in gene expression not related to the experimental treatment (e.g. circadian rhythm), on 12 different days we haphazardly collected three ants from each 2012 colony at the same time of day to minimize variation due to circadian oscillations. We measured response to temperature with a one-hour static temperature application, which is ecologically relevant for workers that leave the thermally-buffered nest and are immediately exposed to ambient temperatures while foraging [@lubertazzi2012]. Each day, the ants were placed in glass tubes immersed in a water bath maintained at one of 12 randomly-assigned temperatures (0° to 38.5°C, in 3.5° increments) for one hour. The minimum and maximum temperatures were selected based on previous work showing that these temperatures are close to the critical minimum (~0°C) and maximum (~43°C) temperatures for *Aphaenogaster* [@warren2013], and these treatments did not cause mortality. At the end of the hour, the ants were flash frozen in liquid nitrogen and stored at -80°C. Thus, our reactionome characterized early, but not late, responding genes. We extracted mRNA by homogenizing the three pooled ants in 500 uL of RNAzol buffer with zirconium silicate beads in a Bullet Blender (Next Advance; Averill Park, NY), followed by RNAzol extraction (Molecular Research Center Inc; Cincinnati, OH) and then an RNeasy micro extraction (Qiagen Inc; Valencia, CA) following the manufacturer's instructions. 

## Sequencing, assembly and annotation

For each species, the 12 samples were barcoded and sequenced in a single lane of 2 x 100bp paired-end reads on an Illumina HiSeq 1500 yielding 200 and 160 million reads for the *A. picea* and *A. carolinensis* samples respectively. Reads were filtered to remove Illumina adapter sequences and low quality bases using the program Trimmomatic [@lohse2012]. 

We assembled the sequenced reads into the full set of mRNA transcripts, the transcriptome, for the combined data set from both species using the Trinity *de novo* transcriptome assembly program [@grabherr2011]. *De novo* transcriptome assembly is prone to falsely identifying alternative transcripts and identifying inaccurate transcripts that are chimeric (e.g. regions of two separate transcripts that assemble into a false, or chimeric, third transcript) [@yang2013]. We removed potentially false transcripts by first running the program `CAP3` [@huang1999] to cluster sequences with greater than 90% similarity and merge transcripts with overlaps longer than 100 bp and 98% similar in length. Second, we ran the program `uclust` which clusters sequences completely contained within longer sequences at greater than 90% similarity (see Supplementary Methods). We used liberal values (90% similarity) to merge orthologous transcripts in the two species that may not have assembled together in the initial *de novo* transcriptome assembly. To identify contaminant sequences, we screened our full transcriptome using the program `DeconSeq` [@schmieder2011] with the provided bacteria, virus, archaen and human databases of contaminants. 

The Trinity *de novo* transcriptome assembly for both species assembled together included 126,172 transcripts with a total length of 100 million bp. Filtering to remove redundant or chimeric reads resulted in an assembly with 105,536 transcripts. The total length was 63 million bp with an N~50~ length of 895 bp and a mean transcript size of 593 bp. Of the 105,536 filtered transcripts, 55,432 had hits to the NCBI-nr database. Of these, 38,711 transcripts mapped to GO terms, 1,659 transcripts were identified to an enzyme and 18,935 transcripts mapped to a domain with >50% coverage. We removed 5,675 transcripts identified as known contaminants, leaving 99,861 clean transcripts.

We assessed the quality of the transcriptome assembly using the BUSCO program [@simao2015] available from (http://busco.ezlab.org/). BUSCO asseses transcriptome completeness by measuring the number of near-universal single-copy orthologs selected from OrthoDB, using the Arthropod database.

To determine the putative function of the transcripts, we used functional annotation of the transcriptome assembly using the web-based tool `FastAnnotator` [@chen2012] which annotates and classifies transcripts by Gene Ontology (GO) term assignment, enzyme identification and domain identification. 

## Identification of thermally-responsive transcripts

We quantified expression of each transcript using the program `Sailfish` [@patro2014] and used the bias-corrected transcripts per million (TPM) [@wagner2012] as our measure of transcript expression. We included the contaminant transcripts identified by `DeconSeq` at the quantification stage to avoid incorrectly assigning reads to other transcripts, but removed these from further analyses. Because preliminary examination of the data (Supplementary Methods) indicated that the 7°C samples may have been mis-labeled, we omitted these data from the analysis. The expression values were highly correlated between species at each temperature treatment (r^2^ > 0.98) indicating that assembling the transcriptome with data from both species was justified (Supplementary Methods).

To identify transcripts that had significant changes in expression across the thermal gradient, we fit to each transcript an ordinary least-squares polynomial regression model

$$ log(TPM + 1) = \beta_0 + \beta_1(species) + \beta_2(temperature) + \beta_3(temperature^2) + \beta_4(species * temperature) + \beta_5(species * temperature^2) + \epsilon $$
    
Temperature and species were both fixed effects, with a quadratic term included for temperature. We used $log(TPM + 1)$ as the response to control for skew in the expression data. For a continuous predictor such as temperature, this regression approach is preferred to an ANOVA approach as it can reveal non-linear responses such as hump-shaped or threshold effects [@gotelli2012]. This method is robust to over-dispersion because we expect errors in the read count distribution [@anders2010] to be independent with respect to temperature. 

To evaluate the statistical significance of the patterns, we computed parametric *P*-values for each model and adjusted these *P*-values using the False Discovery Rate (FDR) approach of Benjamini and Hochberg [-@benjamini1995]. As a more stringent filter for false positives, we then randomly re-assigned each transcript within a species to a different temperature, fit the polynomial models as above, and again calculated *P*-values and FDR. Ideally, these randomized data sets should not yield any significant associations. We repeated this resampling approach 100 times, and used the 95th quantile of false significant transcripts as the null expectation for retaining transcripts from the true data. 

Of these overall significant transcripts, we identified thermally-responsive transcripts as the subset that had significant $\beta_2(temp)$, $\beta_3(temp^2)$, $\beta_4(species * temp)$ or $\beta_5(species * temp^2)$ terms after step-wise model selection by AIC. For each thermally-responsive transcript, we predicted expression levels using the final linear model for each species across the tested thermal range. We used the predicted transcript expression levels to partition transcripts for each species into the five *a priori* defined expression categories: **High** transcripts had greatest expression at temperatures > 31°C, **Low** transcripts had greatest expression at temperatures < 10°C, **Intermediate** transcripts had greatest expression between 10 to 30°C, **Bimodal** transcripts had increased expression at both high and low temperatures, while **NotResp** transcripts were those that were not thermally responsive in the focal species but did respond in the other. For the **Bimodal** group, we required that expression at both low and high temperatures was at least one standard deviation greater than the expression at the rearing temperature of 25°C. Because expression category was defined by the temperature of maximal expression, both **Low** and **High** categories were biased toward transcripts up-regulated at that temperature extreme, but also likely included some transcripts down-regulated at the opposing extreme. The two categories which could unambiguously distinguish up- from down-regulation are *Bimodal* (up at both extremes) and *Intermediate* (down at both extremes). 

## Statistical analyses

We used $\chi^2$ tests to determine if the total number of responsive transcripts, and the number of transcripts in each expression category differed between species. To evaluate if shifts from one expression category to another between the two species were randomly distributed, we used the Stuart-Maxwell test of marginal homogeneity from the `coin` package [@hothorn2008] in R [@r_core_team2013] which tests if the row and column marginal proportions are in equity. 

To test whether the temperature at which thermally-responsive transcripts were activated differs between species, we identified the temperature at which there was the greatest change in expression for each transcript in each species, using only the transcripts with a significant species x temperature interaction. For upregulated transcripts, we grouped the **High** transcripts along with the high temperature end of the **Bimodal** transcripts, and did the same for **Low** transcripts. We then performed a *t*-test to determine if the mean temperature of transcript activation differed between the two species for each group. For downregulated transcripts (i.e. **Intermediate**), we identified the greatest change in expression for each transcript in response to both increasing (> 20°C) and decreasing (< 20°C) temperatures, and used a *t*-test to compare the mean temperature of down-regulation between species.  

To test for a tradeoff between induciblity and constitutive baseline expression between species, we fit ordinary least squares regressions with the log ratio of relative constitutive expression as the response variable and the log ratio of relative inducibility as the predictor variable for **High** transcripts in *A. picea* and for **Low** transcripts in *A. carolinensis*. Constitutive expression was defined as predicted expression at 25°C, whereas inducibility of each transcript was defined as *((maximum TPM - minimum TPM) / minimum TPM) x 100*. In addition, we used a Mann-Whitney test to compare the baseline constitutive expression between species for all responsive transcripts.

## Gene set enrichment analysis

To describe the molecular processes involved in thermal adaptation, we performed gene set enrichment analysis (GSEA) using the `parentChild` algorithm [@grossmann2007] from the package `topGO` [@alexa2006] in R [@r_core_team2013]. Briefly, this approach identifies GO terms that are overrepresented in the significant transcripts relative to all GO terms in the transcriptome, after accounting for dependencies among the GO terms. 

All analyses were performed with R 3.2 [@r_core_team2013] and are fully reproducible (Supplementary Methods).


## Availability of supporting data

Table S1 provides the annotation, *P*-value, r^2^, adjusted *P*-value, and expression type for the thermally-responsive transcripts in each species.

Table S2 provides the results of the gene set enrichment analysis, showing the enriched gene ontology terms for each species in each thermal response category.

The Supplementary Methods contain the detailed information on the analysis. The reproducible and version-controlled scripts underlying the analysis are available on GitHub (https://github.com/johnstantongeddes/ApTranscriptome).

The Illumina short-read sequence data supporting the results of this article are available in the NCBI Short Read Archive BioProject repository, PRJNA260626 http://www.ncbi.nlm.nih.gov/bioproject/PRJNA260626/.

The Trinity transcriptome assembly, FastAnnotator annotation file and Sailfish gene expression quantification files supporting the results of this article are available from the LTER data portal, datasets hf113-38, hf113-41, and hf113-42 (http://dx.doi.org/10.6073/pasta/05ea6464df30efa2f1e2c7439366bf47). 

 

## Competing interests

The authors declare they have no competing interests.

## Authors' contributions

JSG, NG and SHC designed research. JSG, ADN and LC performed research. JSG, JV, MV and SCH analyzed data and wrote the paper. ADN and LC performed research. JV and MV analyzed data. JSG, RD, AE, NS, NG and SHC wrote the paper. All authors read and approved of the final manuscript.

# Acknowledgements

Support made possible by NSF DEB Award #1136644 and the Vermont Genetics Network through Grant Number 8P20GM103449 from the INBRE Program of the National Institute of General Medical Sciences (NIGMS) of the National Institutes of Health (NIH). Its contents are solely the responsibility of the authors and do not necessarily represent the official views of NIGMS or NIH.


# Figure Legends

**Figure 1**. Illustration of the patterns against temperature for each of the four expression categories, **Bimodal**, **High**, **Intermediate** and **Low**. The fifth category of **Not Responsive** is not shown.

**Figure 2**. Illustrations of the expected thermal response patterns in the two species under alternative mechanistic hypotheses of temperature adaptation. Although both temperature extremes were investigated in a similar way, for simplicity only the response to low temperatures is illustrated here. Each column indicates the distribution across all response categories in *A. picea*, which has a lower CT~min~ and is therefore better adapted to low temperatures, for the set of transcripts identified as cold-induced (either **High** or **Bimodal** categories) in the species with higher CT~min~, *A. carolinensis*, relative to the null hypothesis of equal marginal frequencies. The dashed boxes highlight cells that would indicate matched responses in the two species, and the color of each cell (blue = excess, orange = deficit) represents the deviation of the observed from expected number of transcripts. The (A) *enhanced response* hypothesis proposes that the increase in cold tolerance in *A. picea* is achieved by amplifying existing molecular mechanisms, and thus there should be an excess of shared response types between species. In contrast, the (B) *tolerance hypothesis* predicts that *A. picea* is less reliant on induced responses to confer cold-tolerance than *A. carolinensis*, leading to an excess of shifts from induction in *A. carolinensis* to the **Not Responsive** or down-regulation categories in *A. picea*. 

**Figure 3**. Results of analysis of thermal response patterns in the two species. The color of each cell  (blue = excess, orange = deficit) represents the deviation of the observed from the expected number of transcripts based on hypothetical equivalence of the marginal frequencies. The units are number of transcripts. For each temperature extreme, the species expected to be less well adapated to that extreme is displayed on the x-axis for the two response categories corresponding to upregulation (**Bimodal** and **Low** for the low temperatures, or **Bimodal** and **High** for high temperatures). The distribution of response categories for those transcripts in the better-adapted species is arrayed along the y-axis. The dashed boxes indicate the matched responses (e.g. **High** - **High**). (A) Low temperature extreme: there is an excess of shared **Low** and **Bimodal** expression types and a bias away from all other categories in *A. picea*, consistent with the *enhanced response* hypothesis (Fig. 2). (B) High temperature extreme: in addition to an excess of matched categories, there is an excess of **High** and **Bimodal** transcripts in *A. picea* that are not upregulated in *A. carolinensis* (**Intermediate** and **Not Responsive**), partially consistent with the *tolerance* hypothesis. The complete set of matched observations is shown in Fig. S1. Expression types are defined in Table 1.

**Figure 4**. Histogram with smooth density estimate of temperature of maximum rate of change in expression for transcripts that have (A) increased expression at **Low** and **High** temperatures and (B) decreased expression at **Low** and **High** temperatures. Red bars and lines are for *A. carolinensis* while blue bars and lines are for *A. picea*.

**Figure 5**. Scatterplots of log ratios of relative inducibility to relative constitutive expression, defined as expression level at the common rearing temperature (25°C) for (A) **High** transcripts in *A. picea* (*P* < 0.001, r^2^ = 0.07) and (B) **Low** transcripts in *A. carolinensis* (*P* < 0.001, r^2^ = 0.1). Blue lines and confidence intervals are from ordinary least squares regressions. 

**Figure S1.** Deviations from expected numbers of transcripts in matched observations of transcript expression type between species (*A. carolinensis* on rows, *A. picea* on columns). The color of each cell represents the deviation of the observed from the expected number of transcripts based on hypothetical equivalence of the marginal frequencies (blue = excess, orange = deficit). The expression types are **Low** transcripts that had greatest expression temperatures < 10°C, **Intermediate** transcripts with greatest expression between 10 and 30°C, **High** transcripts that had greatest expression at temperatures > 31°, **Bimodal** transcripts with increased expression at both high and low temperatures, and **Not Responsive** transcripts that were not thermally responsive in that species.

\newpage

# Tables

```{r type_table, echo=FALSE, results='asis', fig.cap = "Table of the number of thermally responsive transcripts by expression type for *A. carolinensis* and *A. picea*. **Low** are transcripts with increased expression at low temperatuers (< 10°C), **Intermediate** are transcripts with maximum expression between 10 - 30°C, **High** are transcripts with increased expression at high temperatures (> 31°C), **Bimodal** are transcripts with increased expression at both low and high temperatures, while **NotResp** are transcripts that are not thermally responsive in one species but are in the other species."}

pandoc.table(Ap.type.table[,c("Low", "Intermediate", "High", "Bimodal", "NotResp")], style="rmarkdown",
             big.mark = ",", split.tables = Inf, 
             caption = "Table of the number of thermally-responsive transcripts by expression type for *A. carolinensis* and *A. picea*. **Low** are transcripts with increased expression at low temperatures (< 10°C), **Intermediate** are transcripts with maximum expression between 10 - 30°C, **High** are transcripts with increased expression at high temperatures (> 31°C), **Bimodal** are transcripts with increased expression at both low and high temperatures, while **NotResp** are transcripts that are not thermally responsive in one species but are in the other species.")
```


![Fig. 1](../results/expression_types_Fig1.png)

![Fig. 2](../results/predictions_Fig2.png)

![Fig. 3](../results/matched_observations_Fig3.png)

![Fig. 4](../results/crit_temp_regulation_Fig4.png)

![Fig. 5](../results/genetic_assimilation_Fig5.png)


\newpage

# References

