---
title: "Thermal reactionome of a temperate ant species"
author: "John Stanton-Geddes, Andrew Nguyen, Mahesh Vangala, Jim Vincent, Rob Dunn,
  Aaron Ellison, Nate Sanders, Nick Gotelli, Sara Helms-Cahan"
csl: ecology.csl
output:
  word_document: default
  pdf_document:
    fig_caption: yes
    latex_engine: xelatex
  html_document: null
bibliography: ApTranscriptome.bib
---

## Abstract

- When ants are exposed to different temperatures, lots of genes change expression. 
- The number of responsive genes suggest that there is greater adaptability to cold than hot thermal stress. 
- Comparisons between species indicate that *A. carolinensis*, which has greater basal gene expression and physiological thermal tolerance, tolerates thermal stress by shutting down gene expression. *A. picea* tolerates thermal stress by activating protective genes

## Introduction

Temperature plays a prominent role in structuring biological diversity from molecular to macroecological scales [@kingsolver2009]. At the population level, thermal tolerance has been assessed in dozens of studies often demonstrating adaptive differentiation [@hoffmann2013]. In combination with environmental data, this thermal tolerance data has provided the backbone to dozens of studies examining limits to species geographic ranges [@angert2011; @diamond2013; @gouveia2014], and the impacts of climate change on population viability [@deutsch2008; @kingsolver2013]. At the organismal level, studies of thermal tolerance have largely focused on candidate genes such as heat shock proteins [@dahlgaard1998]. However, estimates of the standing genetic variation [@krebs1997; @krebs1998; @williams2012], studies of differential gene expression [@leemans2000; @sorensen2005; @teets2012], quantitative trait loci (QTL) [@morgan2006] and knock-outs [@takahashi2011] in the model system *Drosophila melanogaster* all implicate dozens of additional genes underlying thermal tolerance. Little is known about the diversity of these genes or their expression patterns, especially in non-model species.

The study of whole organism gene expression (e.g. transcriptomics) has rapidly increased due to improved sequencing technology (RNAseq). To date, RNAseq studies have largely focused on differential expression across few environments [@teets2012; @pespeni2013; @barshis2013; @fu2014] or tissues [@li2013] using an ANOVA-type statistical approach. However, many ecologically-relevant aspects of the environment such as temperature vary continuously in nature. Variation in the phenotype of an organism in response to such a continuous predictor (i.e. phenotypic plasticity) is referred to as a reaction norm. A reaction norm can differ not only in mean values, but also in the shape of the curve as is often found for thermal reaction norms [@hoffmann2013; @murren2014]. An analysis of phenotypic reaction norms found that differences in the shape were greater than differences in mean values in comparisons among species and populations [@murren2014]. The authors concluded that many patterns of biologically-relevant plasticity are likely to be missed in studies with few environments. Similarly, gene expression studies with few environments are likely to miss many biologically-relevant changes in gene expression. To overcome this limitation, we use a novel method to characterize the reaction norm of all genes in the transcriptome (e.g. the reactionome) in response to thermal stress. Specifically, we measure gene expression across a thermal gradient and identify thermally-responsive genes using a regression approach.

We apply this approach in a study of two closely related eastern North American ant species, *Aphaenogaster carolinensis* and *A. picea* [@umphrey1996], which have more southern and northern distributions, respectively. Ants are ecologically-important species involved in multiple symbioses¸ and are key contributors to seed dispersal [@lengyel2009; @rodriguezcabal2012], litter processing and decomposition [@lyford1963; @folgarait1998; @terborgh2001; @holec2006]. As ectothermic species, ants experience a wide range of temperature conditions. Throughout the year, ant species persist through freezings and hot summers, and at shorter time scales, individual workers can experience substantial thermal shock as they leave the buffered ant nest to forage for food. Thus, at the organismal level we expect that ants utilize multiple molecular pathways to tolerate thermal stress.

To explore the molecular toolkit used by *Aphaenogaster* to cope with thermal stress, we performed a whole organism RNA sequencing study. Our specific goals were to 1) generate the set of all genes (i.e. transcriptome) expressed by *Aphaenogaster* across physiologically-relevant temperatures, 2) identify genes that have responsive expression across this thermal gradient (i.e. thermal reactionome), 3) explore the functional annotations of genes that are activated or down-regulated in response to cold and high temperatures. We expected to observe candidate genes in the *response to stress* Gene Ontology (GO) category including *heat shock proteins* and ... in the thermally-responsive group.   

While characterizing the reactionome is useful for description of the molecular pathways for thermal tolerance, a greater understanding of the process of evolution can be gained using a comparative approach [@harvey1991]. We compared the thermal reactionomes between two parapatric species of *Aphaenogaster*, *A. carolinensis* from North Carolina, USA and *A. picea* from Vermont, USA, to determine the extent to which geographic differentiation and selection has changed the expression patterns of expression for genes involved in thermal tolerance. Specifically, we examined differences between the species in (i) the number of thermally-responsive genes, (ii) the distribution of responsive genes into functional categories, (iii) basal expression near optimal temperatures for responsive genes and (iv) the thermal sensitivity of the transcriptome. *A. picea*, the northern species, experiences less chronic and acute heat stress than *A. carolinensis* [@diamond2013] we predicted that it would show more plasticity of gene expression in response to heat shock, and lower basal levels of expression for heat-tolerant genes. 


## Methods

### Samples

Colonies of the ant species *Aphaenogaster picea* were collected from Molly Bog, Vermont (University of Vermont Natural Areas; 44.508°N, -72.702°W) and *Aphaenogaster carolinensis* from Durham, North Carolina ( 36.037° N -78.874° W). The phylogeny of *Aphaeogaster* is currently unresolved [@umphrey1996], but putative species identification was possible based on morphology (Bernice Bacon DeMarco, pers. communication). Colonies of these species were maintained in the lab for 6 months prior to experimentation. 

From each species, we haphazardly collected 12 ants at the same time on 12 days. Each day, the 12 ants were placed in glass tubes into a water bath at one of 12 randomly-assigned temperatures, every 3.5°C between 0° and 38.5°C, for one hour. The minimum and maximum temperatures were selected based on previous work showing that these temperatures are close to the critical minimum and maximum temperatures for *Aphaenogaster* [@warren2013]. At the end of the hour, the ants were flash frozen in liquid nitrogen and stored at -80°C. We extracted mRNA by homogenizing three pooled ants in 500 uL of RNAzol buffer with zirconium silicate beads in a Bullet Blender (Next Advance; Averill Park, NY), followed by RNAzol extraction (Molecular Research Center Inc; Cincinnati, OH) and then an RNeasy micro extraction (Qiagen Inc; Valencia, CA) following the manufacturer's instructions. 

### Sequencing, assembly and annotation

For each species, the 12 samples were barcoded and sequenced in a single lane of 2 x 100bp paired-end reads on an Illumina HiSeq 1500 yielding 200 and 160 million reads for the *A. picea* and *A. carolinensis* samples respectively. Reads were filtered to remove Illumina adapter sequences and low quality bases using the program Trimmomatic [@lohse2012]. 

We assembled the sequenced reads into the full set of mRNA transcripts, the transcriptome, for the combined data set from both species using the Trinity *de novo* transcriptome assembly program [@grabherr2011]. *De novo* transcriptome assembly is prone to falsely identifying alternative transcripts and identifying inaccurate transcripts that are chimeric (e.g. regions of two separate transcripts that assemble into a false, or chimeric, third transcript) [@yang2013]. We removed potentially false transcripts by first running the program `CAP3` [@huang1999] to cluster sequences with greater than 90% similarity and merge transcripts with overlaps longer than 100 bp and 98% similar in length. Second, we ran the program `uclust` which clusters sequences completely contained within longer sequences at greater than 90% similarity (see Supplemental Information for exact parameters). We used liberal values (90% similarity) to merge orthologous transcripts in the two species that may not have assembled together in the initial *de novo* transcriptome assembly. To identify contaminant sequences, we screened our full transcriptome using the program `DeconSeq` [@schmeider2011] with the provided bacteria, virus, archae and human databases of contaminants. 

To determine the putative function of the transcripts, we performed functional annotation of the transcriptome assembly using the web-based tool `FastAnnotator` [@chen2012] which annotates and classifies genes by Gene Ontology (GO) term assignment, enzyme identification and domain identification. 

We performed subsequent analyses at the transcript level due to limitations of working with a system lacking a reference genome, but refer to these transcripts as genes for simplicity.

### Thermally-responsive genes

We quantified expression of each gene using the program `Sailfish` [@patro2013] and use the bias-corrected transcripts per million (TPM) [@wagner2012] as our measure of gene expression. We included the contaminant genes identified by DeconSeq at the quantification stage to avoid incorrectly assigning reads to other genes, but removed these from further analyses. As preliminary examination of the data (see supplemental...) indicated that 7°C samples may have been mis-labeled, we omitted these data from the analysis. The expression values were highly correlated between species at each temperature treatment (r^2^ > 0.98) indicating that assembling the transcriptome with data from both species was justified (Supplemental).

To identify genes that have significant changes in expression across the thermal gradient, we fit the regression model

$$ log(TPM + 1) = \beta_0 + \beta_1(species) + \beta_2(temp) + \beta_3(temp^2) + \beta_4(species * temp) + \beta_5(species * temp^2) + \epsilon $$
    
to each gene. We used $log(TPM + 1)$ as the response to control for skew in the expression data. For a continuous predictor such as temperature, this regression approach is preferred to an ANOVA approach as it can reveal non-linear responses such as hump-shaped or threshold effects even [@cottingham2005]. This method is robust to over-dispersion as we expect errors in the read count distribution [@anders2010] to be independent with respect to temperature. To correct for multiple testing, we calculated adjusted *P* values using the False Discovery Rate (FDR) approach of Benjamini and Hochberg [-@benjamini1995], retaining genes under the 5% FDR threshold.

We identified thermally-responsive genes as the subset under our FDR threshold that had significant $\beta_2(temp)$, $\beta_3(temp^2)$, $\beta_4(species * temp)$ or $\beta_5(species * temp^2)$ terms after step-wise model selection by AIC. All analyses were performed in R [@r_core_team2013] and are fully reproducible (Supplemental file). 

For each thermally-responsive gene, we predicted expression levels across the tested thermal range using the final linear model separately for each species. We used the predicted gene expression values to partition the thermally-responsive genes into five expression categories for each species; genes that had greatest expression at high (> 31°) temperatures (*High*), low (< 10°C) temperatures (*Low*), intermediate (10 - 30°C) temperatures (*Intermediate*), both high and low temperatures (*Bimodal*) or were not thermally-responsive in that species (*NotResp*). For the *Bimodal* group, we required that expression at both low and high temperatures was at least one standard deviation greater than the expression at the mean temperature of 19.5°C.

To describe the overall molecular toolkit for thermal tolerance in *Aphaenogaster*, we performed gene set enrichment analysis (GSEA) on the intersection of each expression category in the two species using the program `topGO` [@alexa2006; @alexa2010] with the `parentChild` algorithm [@grossmann2007] implemented in R [@r_core_team2013]. Briefly, this approach identifies GO terms that are overrepresented in the significant genes relative to all GO terms in the transcriptome, accounting for dependencies among the GO terms. In addition, we used the web-tool DAVID for ....


### Species-level comparisons

To gain greater insight on the ecological genetics of thermal tolerance in *Aphaenogaster*, we performed comparative analyses between the two species focusing on the thermally-responsive genes that had a significant species x temperature interaction. For each gene, we predicted expression levels across temperatures for each species using the full linear model, and then grouped genes into the four responsive categories, *High*, *Low*, *Intermediate*, and *Bimodal*, as well as a fifth category *Not Responsive*. To focus on differences in gene expression between the two species, we performed GSEA for the set difference of each category.

With the same set of genes, we explored the extent to which differences in the thermal reactionome between the species were due to changes in the mean or shape of the reaction norm for each individual gene. Following the example of Murren et al [-@murren2014], we tested for overall differences in the (a) mean expression value across temperatures (*M*), (b) slope of expression (*S*), (c) curvature of expression (*C*) and (d) wiggle of expression (*W*), which consists of all higher-order differences in shape not capture by the first three measures (formulas given in Appendix). As genes differed by five orders of magnitude in expression levels, we calculated the mean-standardized difference between the species $\Delta M = (Mean_{A. picea} - Mean_{Ar}) / \bar M$, where $\bar M = (Mean_{A. picea} + Mean_{Ar}) / 2$ yielding $\Delta M$, $\Delta S$, $\Delta C$, and $\Delta W$ for the mean, slope, curvature and wiggle of each gene, respectively. Using the mean-standardized values, we performed one-sample two-sided *t*-tests to determine if the overall mean, slope or curvature were greater in *A. picea* than *A. carolinensis*. 

Further, to quantify the relative contribution of changes in the mean, slope, curvature or wiggle on the overall differences in reaction norms, we defined the *Total* difference in the reaction norm for each gene as $\Delta T = \Delta M + \Delta S + \Delta C + \Delta W$. We then partitioned the contribution of each measure by dividing it by $\Delta T$. 

To test if genes that were responsive to cold temperatures in the southern *A. carolinensis* species or hot temperatures in the northern *A. picea* species had higher basal expression in the contrasting species, we performed non-parametric two-sided Wilcoxon tests comparing expression values near the optimum of 19.5° C.

As critical maximum temperatures are more constrained than critical minimum temperatures [@warren2013; @hoffman2013], we predicted that the more northern *A. picea* species would have a broader thermal range of stable gene expression than *A. carolinensis*. To test this with our data, for each *Intermediate* expressed gene in each species, we randomly sampled 1,000 temperature values weighted by the expression function (e.g. point of maximum expression had highest probability of being sampled). We then calculated the standard deviation of this random draw, and performed a *t*-test comparing the standard deviations of expression between species. Conversely, we predicted that the thermal sensitivity of *Bimodal* expressed genes would be greater in *A. carolinensis* than *A. picea*. We performed the same analysis as with the *Intermediate* transcripts, but with low standard deviation being indicative of a relatively un-sensitive gene (nearly flat), while the maximum standard deviation of expression is for a highly-convex expression pattern. Thus, finding one species had overall greater standard deviation of expression for *Bimodal* genes would be indicative of greater thermal sensitivity. 

Finally, to test whether the temperature at which thremally-responsive genes are activated differs between species, we identified the temperature at which there was the greatest change in expression. We grouped the *High* genes along with the high temperature end of the *Bimodal* genes, and did the same for *Low* genes. We then tested if the temperature at which the *High* or *Low* group differed among the two species. 

### Data availability

The raw Illumina reads are available at [...], the script for analysis and version history is available at (http://github.com/johnstantongeddes/ApTranscriptome).  



## Results

```{r load, echo=FALSE, results='hide', message=FALSE, warning=FALSE}
options(scipen = 1, digits = 3)
load("../ms.RData")
library(data.table)
library(pander)
library(ggplot2)
library(topGO)
```

The Trinity *de novo* transcriptome assembly included 126,172 genes with a total length of 100 million bp. Filtering to remove redundant or chimeric reads resulted in an assembly with 105,536 genes. The total length was 63 million bp with an N~50~ length of 895 bp and a mean gene size of 593 bp. Of the 105,536 filtered genes, 55,432 had hits to the NCBI-nr database. Of these, 38,711 genes mapped to GO terms, 1,659 genes were identified to an enzyme and 18,935 genes mapped to a domain with >50% coverage. DeconSeq identified 5,675 genes as known contaminants from human, virus, bacterial or archael sources, leaving 99,861 clean transcripts. Of these, 47% have best hits to genes from the class Insecta [(Supplementary Krona plot)](http://johnstantongeddes.org/assets/files/ApTranscriptome_FA.html).

We identified `r prettyNum(length(which(RxNpval$pval < 0.05)), big.mark = ",", scientific = FALSE)` genes with overall model fits of *P* < 0.05, retaining `r prettyNum(length(which(RxNpval$padj < 0.05)), big.mark = ",", scientific = FALSE)` after adjusting *P* values for false discovery rate. Of these genes, `r prettyNum(sum(sigtable[sigtable$Coefficient %in% c("Temperature", "Temperature:Colony"), "Number.significant"]), big.mark = ",", scientific = FALSE)` significantly change expression with temperature including `r prettyNum(sum(sigtable[sigtable$Coefficient %in% "Temperature:Colony", "Number.significant"]), big.mark = ",", scientific = FALSE)` that have a species x temperature interaction. The number of transcripts that increased expression with temperature (*High*), decreased expression with temperature (*Low*), increased expression at both high and low (*Bimodal*) temperatures, decreased expression at high and low temperatures (*Intermediate*) or were not thermally-responsive in one species (*NotResp*) differed significantly between the species ($\chi^2$ = `r chi1$statistic`, P < `r max(chi1$p.value, 0.001)`; Table 1). The density function for the temperature of maximum expression (Fig. 1) showed that the majority of responsive genes were in the *Low* category for both species, with *A. picea* having more *Low* genes than *A. carolinensis*. More genes had *Intermediate* expression in *A. carolinensis* than in *A. picea*, while the two species had a similar number of *High* genes (Fig. 1, Table 1).  

```{r expression_type_table, echo=FALSE, results='asis'}
pandoc.table(tt3, style="rmarkdown", split.tables = Inf, caption = "Number of thermally-responsive genes in each expression category for each species. *A. carolinensis* in rows and *A. picea* in columns.")
```

```{r exp_plot, echo=FALSE, warning = FALSE, fig.cap = "Probability density function of peak expression for all thermally-responsive in *A. carolinensis* (blue) and *A. carolinensis* (red) species"}
ggplot(Ap.df, aes(x=max.val, fill=colony, colour = colony)) +
  geom_density(alpha = 0.2, position = "identity") +
  scale_y_continuous(name = "Density") +
  scale_x_continuous(name = "Temperature of maximum expression")
```

The reaction norms of thermally-responsive genes between species differed by mean expression (*t*~`r t.mean$parameter`~ = `r t.mean$statistic`, *P* < `r max(t.mean$p.value, 0.001)`) which was about `r round(exp(t.mean$estimate))` TPM greater in *A. carolinensis* than *A picea*, and mean slope (*t*~`r t.slope$parameter`~ = `r t.slope$statistic`, *P* < `r max(t.slope$p.value, 0.001)`), which was about `r round(exp(t.mean$estimate))` $\Delta$TPM/degree greater in *A. picea*. Curvature (*t*~`r t.curvature$parameter`~ = `r t.curvature$statistic`, *P* < `r round(max(t.curvature$p.value, 0.001), 3)`) and wiggle (*t*~`r t.wiggle$parameter`~ = `r t.wiggle$statistic`, *P* < `r max(t.wiggle$p.value, 0.001)`) did not differ between species. Of the total differences ($\Delta T$) in reaction norms between species for each gene, about `r round(qlmean[2] * 100, 0)`% (95% CI `r paste(round(qlmean[1] * 100, 0), "--", round(qlmean[3] * 100, 0))`%) and `r round(qlslope[2] * 100, 0)`% (95% CI `r paste(round(qlslope[1] * 100, 0), "--", round(qlslope[3] * 100, 0))`%) were due to shifts in mean and slope, respectively. 

We predicted that we would observe that genes up-regulated in response to thermal stress would have increased basal expression in the other species. Instead, we found that for genes expressed at *High* temperatures in *A. picea* (Wilcoxon V = `r prettyNum(w1$statistic)`, *P* < `r max(w1$p.value, 0.001)`, 95% CI location shift `r paste(round(w1$conf.int[1], 3), "--", round(w1$conf.int[2], 3))`), genes expressed at *Low* temperatures in *A. carolinensis* (Wilcoxon V = `r prettyNum(w2$statistic)`, *P* < `r max(w2$p.value, 0.001)`, 95% CI location shift `r paste(round(w2$conf.int[1], 3), "--", round(w2$conf.int[2], 3))`) and *Intermediate* genes in both (Wilcoxon V = `r prettyNum(w3$statistic)`, *P* < `r max(w3$p.value, 0.001)`, 95% CI location shift `r paste(round(w3$conf.int[1], 3), "--", round(w3$conf.int[2], 3))`), basal expression near optimum temperatures of 19.5°C was always higher in *A. carolinensis*. 

Overall greater thermal tolerance of one species would be indicated by finding that gene expression is stable across a greater portion of the thermal gradient. For *Bimodal* expressed genes, we found no difference (*t*~df=1406~ = 0.31, *P* = 0.76) between species in the variance of the reaction norms for thermally-responsive genes. In contrast, *Intermediate* genes from *A. carolinensis* had a broader thermal range of stable gene expression than those from *A. picea* (*t*~df=1406~ = 9.25, *P* < 0.001, 95% CI difference in means 0.41 - 0.64). In addition, the species differed in the temperatures at which thermally-responsive genes were activated. For genes with increased expression at *High* temperatures, we found that *A. carolinensis* genes were activated on average at 33.8°C compared to 32.9°C for *A. picea* (Fig. 2). For genes with increased expression at *Low* temperatures, *A. carolinensis* genes were activated on average at 12.8°C compared to 13°C for a *A. picea* (Fig. 2). 

```{r T_on_plot, echo=FALSE, warning = FALSE, fig.cap = "Histogram with smooth density estimate of temperature of maximum change in expression for genes that have increased expression at *Low* and *High* temperatures. Bars are Probability density function of peak expression for all thermally-responsive in *A. carolinensis* (blue) and *A. carolinensis* (red) species"}
T_on_plot <- ggplot(Ap.T_on, aes(x = T_on, y = ..density.., colour = colony, group = colony, fill = colony)) +
    geom_histogram(position = "dodge", binwidth = 3) +
    geom_density(alpha = 0.2, adjust = 3) + 
    scale_y_continuous(name="Density") +
    scale_x_continuous(name=expression("Temperature of gene activation"))
print(T_on_plot)
```


### Molecular comparisons

As the molecular pathways involved in thermal response to cold and hot temperatures differ, we performed gene set enrichment analysis on each expression category separately. Out of a total of `r length(score(Ap.BP.resultParentChild))` GO terms scored, we found `r nrow(Ap.gsea1[which(Ap.gsea1$Type == "High"), ])` enriched (*P* < 0.01) for *High* transcripts, including categories such as `r Ap.gsea1[which(Ap.gsea1$Type == "High"), "Term"][1]` and `r Ap.gsea1[which(Ap.gsea1$Type == "High"), "Term"][6]` (Table S1). For *Low* genes, `r nrow(Ap.gsea1[which(Ap.gsea1$Type == "Low"), ])` GO terms were enriched including some implicated in stress response (e.g. `r Ap.gsea1[which(Ap.gsea1$Type == "Low"), "Term"][18]`) and cold tolerance (e.g. `r Ap.gsea1[which(Ap.gsea1$Type == "Low"), "Term"][6]`) (Table S1). For *Bimodal* genes, `r nrow(Ap.gsea1[which(Ap.gsea1$Type == "Bimodal"), ])` GO terms were enriched including `r length(grep("regulation", Ap.gsea1[which(Ap.gsea1$Type == "Bimodal"), "Term"]))` *regulation* terms (e.g. regulation of gene expression) indicating that these are general genes involved in activating responses to change (Table S1). Finally, for *Intermediate* genes `r nrow(Ap.gsea1[which(Ap.gsea1$Type == "Intermediate"), ])` GO terms were enriched, primarily metabolic or biosynthetic processes (Table S1).

In addition, we examined the reaction norms for genes in the GO category "response to stress" which we *a priori* expected to have significant thermal responses (Figure 2). Though we found `r resp.stress.responsive.count` genes annotated with this GO term among the thermally-responsive trancripts, this number was not enriched compared to the total number in the transcriptome ($\chi^2$ = `r GO.stress.Xsq$statistic`, P < `r round(GO.stress.Xsq$p.value, 2)`).

```{r rxn_plot, echo = FALSE, warning = FALSE, fig.cap = "Gene expression in transcripts per million (TPM) for thermally-reactive genes in the GO category *response to stress* for each colony"}
rxnplot <- ggplot(resp.TPM.dt.sub[GO0006950.responsive$Transcript], aes(x=val, y=TPM.scaled, group=Transcript)) +
  geom_smooth(method = "lm", formula = y ~ poly(x, 2)) +
  facet_grid(. ~ colony2) +
  scale_y_continuous(name = "Expression (scaled)") +
  scale_x_continuous(name = expression(paste("Temperature", degree, "C")))
print(rxnplot)
```

As these genes represent only a handful of the responsive genes, the expression patterns of all responsive genes can be searched by functional annotation or GO category and viewed interactively at [https://johnsg.shinyapps.io/ApRxN-shinyapp/].


We performed gene set enrichment analysis on the set difference of each expression category to reveal differences among the species response to thermal shock. For *High* genes, there were 4 enriched GO terms in *A. picea* and 12 in *A. carolinensis*. For *Low* genes, there were 40 enriched GO terms in *A. picea* and 23 in *A. carolinensis*. For *Intermediate* genes, there were 3 enriched GO terms in *A. picea* and 54 in *A. carolinensis*. For *Bimodal* genes, there were 12 enriched GO terms in *A. picea* and 10 in *A. carolinensis*. 





## Discussion

- many genes thermally responsive
- thermal reaction norms differ between species
  - *A. carolinensis* limited future potential to adapt to heat stress
- more genes activated in response to cold shock --> greater adaptability to cold
- differences among species 
  - *A. carolinensis* tolerates thermal stress by shutting down biological processes
  - in *A. picea* the *Intermediate* genes from *A. carolinensis* are over-represented in *Low* and *Intermediate* categories
  - potential explanation is that *A. carolinensis* tolerates thermal stress by shutting-down metabolism and relies on other mechanisms (behavior, protein stability) whereas *A. picea* activates protective genes which requires metabolism...
- comparison of gene terms with other species (Sara?)

- regression approach allows us to detect patterns not evident from ANOVA design

-------------------------------------


Thermal tolerance is a key trait for determining the survival of individual organisms, the persistence of populations and the geographic limits of species. While considerable research has focused on a few key genes (e.g. heat shock proteins) involved in thermal stress response [@krebs1997; @gaitainespitia2013], little is known about the functional diversity or transcriptome-wide patterns of thermal responsiveness. In this study, we characterized the thermal reactionome of parapatric ant species, *Aphaenogaster picea* and *Aphaenogaster carolinensis*, found in colder and warmer environments, respectively. We found that nearly 10% of the genes in these species were thermally-responsive, with many more genes upregulated in response to cold than heat stress. The patterns of gene expression between species differed substantially, with more down-regulation of gene expression in *A. carolinensis* and more up-regulation of gene expression in *A. picea* in response to thermal stress (Fig. 2).

Given the differences in the number of thermally-responsive genes between the two species (Table 1), a striking pattern is that 3-4x more genes are upregulated at cold than high temperatures in both species, with greater variation between species at low temperatures (Fig. 1). A potential explanation for this pattern is that selection has been stronger on upper than lower thermal limits, thereby reducing genetic variation at high temperatures. This explanation is consistent with the observation in *Aphaenogaster* [@warren2013] and other ectotherms [@addobediako2000; @hoffmann2013] that critical maximum temperatures are more constrained than critical minimum temperatures. Moreover, in an experiment with 94 species of *Drosophila*, Kellerman et al [-@kellerman2012] showed that limited genetic variation among species (e.g. phylogenetic inertia) rather than similar selection pressures underlie greater similarity of upper than lower thermal limits. Thus, our results indicate that, when coupled with phenotypic data, transcriptomic studies can provide insight on adapatability. For *Aphaenogaster*, our results suggest that adaptability to increased temperatures is limited, and these species risk replacement by more heat-tolerant competitors [@rodriguezcabal]. 

Depending on the frequency with which organisms experience stressful conditions, the reaction norms of protective genes may differ considerably. Organisms that frequently experience stress may have high basal levels of expression with little change, while organisms that less frequently experiences stress would have lower basal levels with large changes in response to environmental stress [@cite...]. For *Aphaenogaster*, we predicted that genes up-regulated to cold or hot temperatures in one species might be expressed at higher levels near optimal growth temperatures in the other species to confer greater basal stress tolerance. This prediction was not supported, as we found greater expression near optimal temperatures in *A. carolinensis* for *High* genes in *A. picea*, as predicted, but also for *Low* genes in *A. carolinensis* and *Intermediate* genes in both species. As *A. carolinensis* is from a warmer environment, a likely explanation for this is that this species simply has a higher metabolic rate and thus greater gene expression across it's thermal range. The observation that there are more *Intermediate* genes, which are shut-down under thermal stress, in *A. carolinensis* supports this explanation. ...cite some coral literature.... 

Of the `r prettyNum(sum(sigtable[sigtable$Coefficient %in% c("Temperature", "Temperature:Colony"), "Number.significant"]), big.mark = ",", scientific = FALSE)` genes that were thermally-responsive, `r round(with(sigtable, sigtable[Coefficient=="Temperature:Colony","Number.significant"]) / with(sigtable, sum(sigtable[Coefficient %in% c("Temperature", "Temperature:Colony"),"Number.significant"])) * 100, 0)`% differed in their responsiveness by colony. The differences between species were due to changes in the mean ($\Delta M$) and slope ($\Delta S$) of the reaction norms for each thermally-responsive gene. This result contrasts with the findings of Murren et al. [-@murren2014], who found that most of the differences in phenotypic reaction norms were due to shape. As genes are not directly analogous to phenotypes, this contradiction is not surprising. Instead, it indicates that small shifts in the mean of many genes underlying a trait can result in non-linear phenotypic changes.  

To date, transcriptomic studies have performed differential expression analysis between a few different conditions or environments (but see [@hodginsdavis2012]). By using a regression-approach, we are able to quantitatively describe changes in gene expression, rather than simply report the numbers of differentially-expressed genes. Thus, we were able to compare the temperature at which genes were activated near the upper and lower thermal limits between species (Fig. 2). For genes activated at cold temperatures, there was little difference among species with *A. picea* genes turning on at 13°C compared to 12.8°C for *A. carolinensis*. For high temperature genes, *A. carolinensis* genes were turned on at 33.8°C compared to 32.9°C for *A. picea*. 


### Molecular toolkit of thermal tolerance

REANALYZING...




### Caveats/limitations 

As with all *de novo* transcriptomes, our work has some important caveats. First, given that all sequenced ant genomes to date have about 18,000 genes [@smith...], our transcriptome is highly fragmented. Though we took steps to remove contaminants and redundant transcripts, some likely remain, in addition to partially-assembled transcripts. A genome assembly, in progress, would considerably resolve this problem. Second, the quality of the annotation for a non-model system such as *Aphaenogaster* is poor, as we have to rely upon resources from distantly related *Drosophila* and *Apis*. Finally, the mapping of changes in gene expression to organismal fitness is far from direct [@feder2005]. Large differences in patterns of gene expression may only have small effects on organismal fitness. Given these caveats, transcriptomic studies promise to offer many new insights on organisms' response to environmental change. Candidate gene studies, while informative, fail to capture the complexity of gene expression changes [@cite...]. In addition, for long-lived species that are intractable to study in the lab, transcriptomics can provide key insights on plasticity and the potential for species to tolerate and adapt to environmental change [@cite...]. 


### Conclusions 

In this work, we have completed the first transcriptome for a temperate ant species, *Aphaenogaster picea* and *A. carolinensis*, to date. We found that a large proportion of the transcriptome is thermally-responsive. Transcriptome-wide changes in gene expression were greater at lower than upper thermal limits, indicating greater adapatabilty to cold than hot temperatures. ...differences among species...*A. carolinensis* greater organismal tolerance and shuts down metabolism under thermal stress...*A. picea* less tolerant and activates protective proteins requiring metabolism... These results provide key insights on the genetic basis of thermal tolerance, and provide a tremendous resource for the future study of ecological adaptation in ant species.


## References

